<!doctype html public "-//IETF//DTD HTML 2.0//EN">
<HTML>
<HEAD>
<title>RasMol V2.7.1.1 Manual</title>
</HEAD>
<BODY BACKGROUND="../html_graphics/rasmolwallpaper.jpg">
<a href="http://www.iucr.org/iucr-top/welcome.html">
<img alt="[IUCr Home Page]" src="../html_graphics/iucrhome.jpg"></a>
<a href="http://www.iucr.org/iucr-top/cif/home.html">
<img alt="[CIF Home Page]" src="../html_graphics/cifhome.jpg"></a>
<A HREF="http://www.OpenRasMol.org"><IMG SRC="../html_graphics/rasmolbutton.jpg"
ALT="[OpenRasMol]"></A>
<hr>
<CENTER>
| <a href="http://www.OpenrasMol.org">OpenRasMol</a> |
<A href="../README.html#Copying">Copying and Distribution</A> |
<A href="../README.html#Contents">Contents</A> |
<A href="../INSTALL.html">Installation Instructions</A> |<BR>
| <A href="../ChangeLog.html">Changes</A> |
<A href="../TODO.html">Things To Do</A> |
<A href="../README.html#Introduction">Introduction</A> |
<A href="../README.html#CodeAndBinaries">Source Code and Binaries</A> |<BR>
| <A href="rasmol.html">RasMol Manual</A> |
<a href="esrasmol27.html">Spanish Translation of RasMol Manual</a>|
<a href="../README.html">Release README</A> |<BR>
</CENTER>


<font color="#0808A0">
<H1 ALIGN=CENTER>Manual<BR>
RasMol 2.7.1.1</H1>
<H3 ALIGN=CENTER>
<IMG  SRC="../html_graphics/rasmollogo.jpg" ALT="RasMol"><BR>
Molecular Graphics Visualisation Tool<BR>
21 January 2001<BR>
</H3>
</font><font color="#000000">
<CENTER>
Based on RasMol 2.6
by
Roger Sayle<BR>
Biomolecular Structures Group<BR>
Glaxo Wellcome Research & Development<BR>
Stevenage, Hertfordshire, UK<BR>
Version 2.6, August 1995, Version 2.6.4, December 1998<BR>
Copyright &#169; Roger Sayle 1992-1999
</CENTER>
<P>
<CENTER>
and Based on Mods by Arne Mueller<BR>
Version 2.6x1, May 1998<BR>
Copyright &#169; Arne Mueller 1998<BR>
</CENTER>
<P>
<CENTER>
Version 2.7.0, 2.7.1, 2.7.1.1 Mods
by
Herbert J. Bernstein<BR>
Bernstein + Sons,
P.O. Box 177, Bellport, NY, USA,
<A HREF="mailto:yaya@bernstein-plus-sons.com">yaya@bernstein-plus-sons.com</A><BR>
2.7.0 March 1999, 2.7.1 June 1999, 2.7.1.1 January 2001<BR>
Copyright &#169; Herbert J. Bernstein 1998-2001
<P>
 Translations of most of the RasMol messages into Spanish contributed by
 Fernando Gabriel Ranea (Laboratorio de Micolog&iacute;a Museo de Farmacobot&aacute;nica,
 Facultad de Farmacia y Bioqu&iacute;mica. Universidad de Buenos Aires
 <A HREF="mailto:davinci@dinamica.com.ar">davinci@dinamica.com.ar</a>).
</CENTER>
<P>
 The original RasMol manual was created by Roger Sayle.  In July 1996,
 Dr. Margaret Wong of the Chemistry Department, Swinburne University
 of Technology, Australia, made extensive revisions to the RasMol 2.5
 manual to accurately reflect the operation of RasMol 2.6.  Eric Martz
 of the University of Massachusetts made further revisions.  In May
 1997, William McClure of Carnegie Mellon University reorganized the
 HTML version of the manual into multiple sections which could be
 downloaded quickly and added use of frames.   Portions of the 2.7.1
 version of the RasMol manual were derived with  permission from
 William McClure's version using Roger Sayle's rasmol.doc for
 version 2.6.4 as the primary source.
<p>
<CENTER>Documentation Last Updated 21 January 2001<BR>
Edited by Herbert J. Bernstein and Frances C. Bernstein</CENTER>
<p>
<H4 Align="CENTER">Translation of RasMol 2.7.1 Manual into Spanish</H4>
<P>Thanks to the efforts of Jos&eacute; Miguel Fern&aacute;ndez
Fern&aacute;ndez (Departamento
de Bioqu&iacute;mica y Biolog&iacute;a Molecular. Universidad de Granada.
Espa&ntilde;a (<A HREF="mailto:jmfernan@ugr.es">jmfernan@ugr.es</A>)) a
<a href="doc/esrasmol27.html">translation of the
Manual for Rasmol version 2.7.1 into Spanish</a> is now available.
<P>La traducci&oacute;n espa&ntilde;ola del manual de la
versi&oacute;n de la Dra. Wong revisada por Eric Martz fue realizada por
Isabel Serv&aacute;n Mart&iacute;nez y Jos&eacute; Miguel Fern&aacute;ndez
Fern&aacute;ndez </P>
<P>La actual traducci&oacute;n del Manual de RasMol 2.7.1
ha sido realizada usando como base la anterior de RasMol 2.6 por
&nbsp;Jos&eacute; Miguel Fern&aacute;ndez Fern&aacute;ndez.
<HR>
<H4 ALIGN=CENTER>Test Version of Next Release Available</H4>
<P>
RasMol 2.7.1.1 is an interim release for testing of the integration
of multilingual capabilities into RasMol.  These same mods
will then be integrated with pending mods, and with the
basic structure of RasMol 2.7.2 to create RasMol 2.7.2.1.
<P>A preliminary release of RasMol 2.7.2 (without the
multilingual capabilities) is available at:
<P align=center>
<a href="http://www.bernstein-plus-sons.com/software/RasMol_2.7.2">
http://www.bernstein-plus-sons.com/software/RasMol_2.7.2</a>
<P Align="CENTER">
<B>
THIS IS A PRELIMINARY RELEASE INVOLVING EXTENSIVE MODIFICATIONS<BR>
*****  USE WITH CAUTION  ******<BR>
</B>
<H2 ALIGN=CENTER>IMPORTANT</H2>
<P>
   This version is based on RasMol version 2.6_CIF.2, on RasMol version
2.6x1 and on RasMol version 2.6.4. Please read the file <A HREF = "NOTICE.html">NOTICE</A> for
important notices which apply to this package.  If you are not going to
make changes to RasMol, you are not only permitted to freely make copies
and distribute them, you are encouraged to do so, provided you do the
following:
<UL>
<LI>  1.  Either include the complete documentation, especially the file
 <A HREF = "NOTICE.html">NOTICE</A>, with what you distribute or
provide a clear indication where people can get a copy of the documentation; and
<LI>  2.  Please give credit where credit is due citing the version and
original authors properly; and
<LI>  3.  Please do not give anyone the impression that the original authors
are providing a warranty of any kind.
</UL>
<P>
If you would like to use major pieces of RasMol in some other program, make
modifications to RasMol, or in some other way make what a lawyer would call
a "derived work", you are not only permitted to do so, you are encouraged
to do so.   In addition to the things we discussed above, please do the
following:
<UL>
<LI>  4.  Please explain in your documentation how what you did differs from
this version of RasMol; and
<LI>  5.  Please make your modified source code available.
</UL>
<P>
This version of RasMol is <b>not</b> in the public domain, but it is given
freely to the community in the hopes of advancing science.  If you make
changes, please make them in a responsible manner, and please offer
us the opportunity to include those changes in future versions of RasMol.
<HR>
<h2>Table of Contents</h2>
<ol>
<li><a href="#chnotice">Notices</a>
<li><a href="#chintro">Introduction</a>
<li><a href="#chgenop">General Operation</a>
<UL>
<LI><A HREF="#go_unix">Running RasMol Under UNIX or VMS</A>
<LI><A HREF="#go_mswin">Running RasMol Under Microsoft Windows</A>
<LI><A HREF="#go_mac">Running RasMol on the Apple Macintosh/PPC</A>
<LI><A HREF="#go_canvas">RasMol's Window</A>
<LI><A HREF="#go_mouse">Mouse Controls</A>
<LI><A HREF="#go_scrollbars">Scroll Bars</A>
<LI><A HREF="#go_picking">Picking</A>
<LI><A HREF="#go_dials">Dials Box</A>
<LI><A HREF="#go_cmdline">Command Line Interface</A>
<LI><A HREF="#go_dimensions">Dimensions within RasMol</A>
<LI><A HREF="#go_ini">Start-up Initialisation Files</A>
<LI><A HREF="#go_interprocess">Inter-Process Communication (IPC)</A>
</UL>
<li><a href="#chcomref">Command Reference</a>
<center>
<table  BORDER CELLPADDING=2>
<tr><td><a href="#backbone">Backbone</a>
<td><a href="#background">Background</a>
<td><a href="#cartoon">Cartoon</a>
<td><a href="#centre">Centre</a>
<td><a href="#clipboard">Clipboard</a>
<td><a href="#colour">Colour</a>
<td><a href="#connect">Connect</a>
<td><a href="#spacefill">CPK</a>
<tr><td><a href="#define">Define</a>
<td><a href="#dots">Dots</a>
<td><a href="#echo">Echo</a>
<td><a href="#english">English</a>
<td><a href="#quit">Exit</a>
<td><a href="#hbonds">HBonds</a>
<td><a href="#help">Help</a>
<td><a href="#label">Label</a>
<tr><td><a href="#load">Load</a>
<td><a href="#monitor">Monitor</a>
<td><a href="#pause">Pause</a>
<td><a href="#print">Print</a>
<td><a href="#quit">Quit</a>
<td><a href="#refresh">Refresh</a>
<td><a href="#renumber">Renumber</a>
<td><a href="#reset">Reset</a>
<tr><td><a href="#restrict">Restrict</a>
<td><a href="#ribbons">Ribbons</a>
<td><a href="#rotate">Rotate</a>
<td><a href="#save">Save</a>
<td><a href="#script">Script</a>
<td><a href="#select">Select</a>
<td><a href="#set">Set</a>
<td><a href="#show">Show</a>
<tr><td><a href="#slab">Slab</a>
<td><a href="#script">Source</a>
<td><a href="#spacefill">Spacefill</a>
<td><a href="#spanish">Spanish</a>
<td><a href="#ssbonds">SSBonds</a>
<td><a href="#star">Star</a>
<td><a href="#stereo">Stereo</a>
<td><a href="#strands">Strands</a>
<tr><td><a href="#structure">Structure</a>
<td><a href="#backbone">Trace</a>
<td><a href="#translate">Translate</a>
<td><a href="#wireframe">Wireframe</a>
<td><a href="#write">Write</a>
<td><a href="#zap">Zap</a>
<td><a href="#zoom">Zoom</a>
<td cellspan=1>
</table>
</center>
<li><a href="#chsetopt">Internal Parameters</a>
<center><table BORDER CELLPADDING=2>
<TR><TD><a href="#setambient">Ambient</a>
<TD><a href="#setaxes">Axes</a>
<TD><a href="#setbackground">Background</a>
<TD><a href="#setbackfade">BackFade</a>
<TD><a href="#setbondmode">BondMode</a>
<TD><a href="#setbonds">Bonds</a>
<TD><a href="#setboundbox">BoundBox</a>
<TD><a href="#setcartoon">Cartoon</a>
<TR><TD><a href="#setcisangle">CisAngle</A>
<TD><a href="#setdisplay">Display</a>
<TD><a href="#setfontsize">FontSize</a>
<TD><a href="#setfontstroke">FontStroke</a>
<TD><a href="#sethbonds">HBonds</a>
<TD><a href="#sethetero">Hetero</a>
<TD><a href="#sethourglass">HourGlass</a>
<TD><a href="#sethydrogen">Hydrogen</a>
<TR><TD><a href="#setkinemage">Kinemage</a>
<TD><a href="#setmenus">Menus</a>
<TD><a href="#setmonitor">Monitor</a>
<TD><a href="#setmouse">Mouse</a>
<TD><a href="#setpicking">Picking</a>
<TD><a href="#setradius">Radius</a>
<TD><a href="#setshadow">Shadow</a>
<TD><a href="#setslabmode">SlabMode</a>
<TR><TD><a href="#setsolvent">Solvent</a>
<TD><a href="#setspecular">Specular</a>
<TD><a href="#setspecpower">SpecPower</a>
<TD><a href="#setstereo">Stereo</a>
<TD><a href="#setssbonds">SSBonds</a>
<TD><a href="#setstrands">Strands</a>
<TD><a href="#settransparent">Transparent</a>
<TD><a href="#setunitcell">UnitCell</a>
<TR><TD><a href="#setvectps">VectPS</a>
<TD><a href="#setwrite">Write</a>
<TD CELLSPAN=6>
</table></center>
<li><a href="#chexprs">Atom Expressions</a>
<li><a href="#predefinedsets">Predefined Sets</a>
<li><a href="#chcolours">Colour Schemes</a>
<li><a href="#chfile">File Formats</a>
<li><a href="#chmacspec">File Machine-Specific Support</a>
<li><a href="#chbib">Bibliography</a>
</ol>
<p><hr>
<P><BR>RasMol Copyright &#169; Roger Sayle 1992-1999
<BR>Version 2.6x1 Mods Copyright &#169; Arne Mueller 1998
<BR>Version 2.7.0, 2.7.1, 2.7.1.1 Mods Copyright &#169; Herbert J. Bernstein 1998-2001
<BR><address><a href="mailto:rasmol@bernstein-plus-sons.com">rasmol@bernstein-plus-sons.com</a></address>
<p>
All rights reserved. Use of copyright notice does not imply publication or
disclosure. The information supplied in this document is believed to be true
but no liability is assumed for its use or for the infringements of the rights
of the others resulting from its use. Information in this document is subject
to change without notice and does not represent a commitment on the part of
the supplier.
<p><hr>
<a name="chnotice"><h2>Notices</h2></a>
<p>
This software has been created from several sources.  Much of the code is
from RasMol 2.6, as created by Roger Sayle.
<P>See:
<A href="ftp://ftp.dcs.ed.ac.uk/pub/rasmol">ftp://ftp.dcs.ed.ac.uk/pub/rasmol</A><P>
The torsion angle code, new POVRAY3 code and other features are derived from
the RasMol2.6x1 revisions by Arne Mueller.
<P>See:
<A href="ftp://nexus.roko.goe.net/pub/rasmol">ftp://nexus.roko.goe.net/pub/rasmol</A><P>
The Ramachandran printer plot code was derived from fisipl created by Frances C.
Bernstein.  See the Protein Data Bank program tape.
<p>
The CIF modifications make use of a library based in part on CBFlib by
Paul J. Ellis and Herbert J. Bernstein.
<P>See:
<A href="http://www.bernstein-plus-sons.com/software/CBF">http://www.bernstein-plus-sons.com/software/CBF</A><P>
Parts of CBFlib is loosely based on the CIFPARSE software package from the NDB
at Rutgers university.
<P>See
<A href="http://ndbserver.rutgers.edu/NDB/mmcif/software">http://ndbserver.rutgers.edu/NDB/mmcif/software</A><P>
Please type the RasMol commands
'<tt><b>help copying</b></tt>',
'<tt><b>help general</b></tt>',
'<tt><b>help IUCR</b></tt>',
'<tt><b>help CBFlib</b></tt>',
 and
'<tt><b>help CIFPARSE</b></tt>'
for applicable notices.  Please type
'<tt><b>help copyright</b></tt>'
for copyright notices.  If you use RasMol V2.6
or an earlier version, type the RasMol command
'<tt><b>help oldnotice</b></tt>'.
<p><hr>
<a name="copyingrasmol"><h3>Copying RasMol</h3></a><p>
This version is based on RasMol version 2.6_CIF.2, RasMol 2.6x1  and
RasMol_2.6.4. If you are not going to make changes to RasMol, you are
not only permitted to freely make copies and distribute them, you are
encouraged to do so, provided you do the following:
<p>
  1.  Either include the complete documentation, especially the file
NOTICE, with what you distribute or provide a clear indication where
people can get a copy of the documentation; and
<p>
  2.  Please give credit where credit is due citing the version and
original authors properly; and
<p>
  3.  Please do not give anyone the impression that the original
authors are providing a warranty of any kind.
<p>
If you would like to use major pieces of RasMol in some other program,
make modifications to RasMol, or in some other way make what a lawyer
would call a &quot;derived work&quot;, you are not only permitted to do so, you
are encouraged to do so. In addition to the things we discussed above,
please do the following:
<p>
  4.  Please explain in your documentation how what you did differs
from this version of RasMol; and
<p>
  5.  Please make your modified source code available.
<p>
This version of RasMol is _not_ in the public domain, but it is given
freely to the community in the hopes of advancing science.  If you make
changes, please make them in a responsible manner, and please offer us
the opportunity to include those changes in future versions of RasMol.
<p><hr>
<a name="generalnotice"><h3>General Notice</h3></a><p>
The following notice applies to this work as a whole and to the works included
within it:
<p>
* Creative endeavors depend on the lively exchange of ideas. There are laws
and customs which establish rights and responsibilities for authors and the
users of what authors create. This notice is not intended to prevent you
from using the software and documents in this package, but to ensure that
there are no misunderstandings about terms and conditions of such use.
<p>
* Please read the following notice carefully. If you do not understand any
portion of this notice, please seek appropriate professional legal advice
before making use of the software and documents included in this software
package. In addition to whatever other steps you may be obliged to take
to respect the intellectual property rights of the various parties
involved, if you do make use of the software and documents in this package,
please give credit where credit is due by citing this package, its authors
and the URL or other source from which you obtained it, or equivalent
primary references in the literature with the same authors.
<p>
* Some of the software and documents included within this software package
are the intellectual property of various parties, and placement in this
package does not in any way imply that any such rights have in any way been
waived or diminished.
<p>
* With respect to any software or documents for which a copyright exists,
ALL RIGHTS ARE RESERVED TO THE OWNERS OF SUCH COPYRIGHT.
<p>
* Even though the authors of the various documents and software found here
have made a good faith effort to ensure that the documents are correct and
that the software performs according to its documentation, and we would
greatly appreciate hearing of any problems you may encounter, the programs
and documents and any files created by the programs are provided **AS IS**
without any warranty as to correctness, merchantability or fitness for any
particular or general use.
<p>
* THE RESPONSIBILITY FOR ANY ADVERSE CONSEQUENCES FROM THE USE OF PROGRAMS
OR DOCUMENTS OR ANY FILE OR FILES CREATED BY USE OF THE PROGRAMS OR
DOCUMENTS LIES SOLELY WITH THE USERS OF THE PROGRAMS OR DOCUMENTS OR FILE
OR FILES AND NOT WITH AUTHORS OF THE PROGRAMS OR DOCUMENTS.
<p>
Subject to your acceptance of the conditions stated above, and your
respect for the terms and conditions stated in the notices below,
if you are not going to make any modifications or create derived works,
you are given permission to freely copy and distribute this package,
provided you do the following:
<p>
1.  Either include the complete documentation, especially the file
NOTICE, with what you distribute or provide a clear indication where
people can get a copy of the documentation; and
<p>
2.  Give credit where credit is due citing the version and
original authors properly; and
<p>
3.  Do not give anyone the impression that the original authors
are providing a warranty of any kind.
<p>
In addition, you may also modify this package and create derived works
provided you do the following:
<p>
4.  Explain in your documentation how what you did differs from
this version of RasMol; and
<p>
5.  Make your modified source code available.
<p><hr>
<a name="rasmolv26notice"><h3>RasMol V2.6 Notice</h3></a><p>
The following notice applies to RasMol V 2.6 and older RasMol versions.
<p>
Information in this document is subject to change without notice and
does not represent a commitment on the part of the supplier. This package
is sold/distributed subject to the condition that it shall not, by way
of trade or otherwise, be lent, re-sold, hired out or otherwise
circulated without the supplier's prior consent, in any form of
packaging or cover other than that in which it was produced. No
part of this manual or accompanying software may be reproduced,
stored in a retrieval system on optical or magnetic disk, tape or
any other medium, or transmitted in any form or by any means,
electronic, mechanical, photocopying, recording or otherwise for
any purpose other than the purchaser's personal use.
<p>
This product is not to be used in the planning, construction,
maintenance, operation or use of any nuclear facility nor the
flight, navigation or communication of aircraft or ground support
equipment. The author shall not be liable, in whole or in part, for
any claims or damages arising from such use, including death,
bankruptcy or outbreak of war.
<p><hr>
<a name="iucrpolicy"><h3>IUCR Policy</h3></a><p>
<tt><b>The IUCr Policy for the Protection and the Promotion</b></tt>
<tt><b>of the STAR File and CIF Standards</b></tt>
<tt><b>for Exchanging and Archiving Electronic Data</b></tt>
<p>
<tt><b>Overview</b></tt>
<p>
The Crystallographic Information File (CIF)[1] is a standard for information
interchange promulgated by the International Union of Crystallography
(IUCr). CIF (Hall, Allen &amp; Brown, 1991) is the recommended method for
submitting publications to Acta Crystallographica Section C and reports of
crystal structure determinations to other sections of Acta Crystallographica
and many other journals. The syntax of a CIF is a subset of the more general
STAR File[2] format. The CIF and STAR File approaches are used increasingly
in the structural sciences for data exchange and archiving, and are having a
significant influence on these activities in other fields.
<p>
<tt><b>Statement of intent</b></tt>
<p>
The IUCr's interest in the STAR File is as a general data interchange
standard for science, and its interest in the CIF, a conformant derivative
of the STAR File, is as a concise data exchange and archival standard for
crystallography and structural science.
<p>
<tt><b>Protection of the standards</b></tt>
<p>
To protect the STAR File and the CIF as standards for interchanging and
archiving electronic data, the IUCr, on behalf of the scientific community,
<p>
   * holds the copyrights on the standards themselves,
<p>
   * owns the associated trademarks and service marks, and
<p>
   * holds a patent on the STAR File.
<p>
These intellectual property rights relate solely to the interchange formats,
not to the data contained therein, nor to the software used in the
generation, access or manipulation of the data.
<p>
<tt><b>Promotion of the standards</b></tt>
<p>
The sole requirement that the IUCr, in its protective role, imposes on
software purporting to process STAR File or CIF data is that the following
conditions be met prior to sale or distribution.
<p>
   * Software claiming to read files written to either the STAR File or the
CIF standard must be able to extract the pertinent data from a file
conformant to the STAR File syntax, or the CIF syntax, respectively.
<p>
   * Software claiming to write files in either the STAR File, or the CIF,
standard must produce files that are conformant to the STAR File
syntax, or the CIF syntax, respectively.
<p>
   * Software claiming to read definitions from a specific data dictionary
approved by the IUCr must be able to extract any pertinent definition
which is conformant to the dictionary definition language (DDL)[3]
associated with that dictionary.
<p>
The IUCr, through its Committee on CIF Standards, will assist any developer
to verify that software meets these conformance conditions.
<p>
<tt><b>Glossary of terms</b></tt>
<p>
[1] CIF:
is a data file conformant to the file syntax defined at
http://www.iucr.org/iucr-top/cif/spec/index.html
<p>
[2] STAR File:
is a data file conformant to the file syntax defined at
http://www.iucr.org/iucr-top/cif/spec/star/index.html
<p>
[3] DDL:
is a language used in a data dictionary to define data items in terms
of &quot;attributes&quot;. Dictionaries currently approved by the IUCr, and the
DDL versions used to construct these dictionaries, are listed at
http://www.iucr.org/iucr-top/cif/spec/ddl/index.html
<p>
Last modified: 30 September 2000
<p>
IUCR Policy Copyright (C) 2000 International Union of Crystallography
<p><hr>
<a name="cbflib"><h3>CBFLIB</h3></a><p>
The following Disclaimer Notice applies to CBFlib V0.1, from which this code
in part is derived.
<p>
* The items furnished herewith were developed under the sponsorship of the
U.S. Government. Neither the U.S., nor the U.S. D.O.E., nor the Leland
Stanford Junior University, nor their employees, makes any warranty,
express or implied, or assumes any liability or responsibility for
accuracy, completeness or usefulness of any information, apparatus, product
or process disclosed, or represents that its use will not infringe
privately-owned rights. Mention of any product, its manufacturer, or
suppliers shall not, nor is it intended to, imply approval, disapproval,
or fitness for any particular use. The U.S. and the University at all times
retain the right to use and disseminate the furnished items for any purpose
whatsoever.
<p>
Notice 91 02 01
<p><hr>
<a name="cifparse"><h3>CIFPARSE</h3></a><p>
Portions of this software are loosely based on the CIFPARSE software package
from the NDB at Rutgers University.  See
<p>
   http://ndbserver.rutgers.edu/NDB/mmcif/software
<p>
CIFPARSE is part of the NDBQUERY application, a program component of the
Nucleic Acid Database Project [ H. M. Berman, W. K. Olson, D. L. Beveridge,
J. K. Westbrook, A. Gelbin, T. Demeny, S. H. Shieh, A. R. Srinivasan, and B.
Schneider. (1992). The Nucleic Acid Database: A Comprehensive Relational
Database of Three-Dimensional Structures of Nucleic Acids. Biophys J., 63,
751-759.], whose cooperation is gratefully acknowledged, especially in the
form of design concepts created by J. Westbrook.
<p>
Please be aware of the following notice in the CIFPARSE API:
<p>
This software is provided WITHOUT WARRANTY OF MERCHANTABILITY OR FITNESS
FOR A PARTICULAR PURPOSE OR ANY OTHER WARRANTY, EXPRESS OR IMPLIED. RUTGERS
MAKE NO REPRESENTATION OR WARRANTY THAT THE SOFTWARE WILL NOT INFRINGE ANY
PATENT, COPYRIGHT OR OTHER PROPRIETARY RIGHT.
<p><hr>
<a name="chintro"><h2>Introduction</h2></a>
RasMol is a molecular graphics program intended for the visualisation of
proteins, nucleic acids and small molecules.
The program is aimed at display, teaching and generation of
publication quality images.   RasMol runs on wide range of architectures
and operating systems including Microsoft Windows, Apple
Macintosh, UNIX and VMS systems. UNIX and VMS versions require an 8, 24 or
32 bit colour X Windows display (X11R4 or later).  The X Windows version of
RasMol provides optional support for a hardware dials box and accelerated
shared memory communication (via the XInput and MIT-SHM extensions)
if available on the current X Server.
<p>
The program reads in a
molecule coordinate file and interactively displays the molecule on the
screen in a variety of colour schemes and molecule representations. Currently
available representations include depth-cued wireframes, 'Dreiding' sticks,
spacefilling (CPK) spheres, ball and stick, solid and strand biomolecular
ribbons, atom labels and dot surfaces.
<p>
The X Windows version of RasMol provides optional support for a
hardware dials box and accelerated shared memory communication (via
the XInput and MIT-SHM extensions) if available on the current X Server.
<p>
The program reads in molecular coordinate files and interactively
displays the molecule on the screen in a variety of representations
and colour schemes. Supported input file formats include Protein
Data Bank (PDB), Tripos Associates' Alchemy and Sybyl Mol2 formats,
Molecular Design Limited's (MDL) Mol file format, Minnesota
Supercomputer Center's (MSC) XYZ (XMol) format, CHARMm format,
CIF format and mmCIF format files. If connectivity information is
not contained in the file this is calculated automatically.  The
loaded molecule can be shown as wireframe bonds, cylinder 'Dreiding'
stick bonds, alpha-carbon trace, space-filling (CPK) spheres,
macromolecular ribbons  (either smooth shaded solid ribbons or parallel
strands), hydrogen bonding and dot surface representations.  Atoms
may also be labelled with arbitrary text strings.  Alternate conformers
and multiple NMR models may be specially coloured and identified in
atom labels. Different parts of the molecule may be represented and
coloured independently of the rest of the molecule or displayed in
several representations simultaneously. The displayed molecule may
be rotated, translated, zoomed and z-clipped (slabbed) interactively
using either the mouse, the scroll bars, the command line or an attached
dial box. RasMol can read a prepared list of commands from a 'script'
file (or via inter-process communication) to allow a given image or
viewpoint to be restored quickly. RasMol can also create a script file
containing the commands required to regenerate the current image. Finally,
the rendered image may be written out in a variety of formats including
either raster or vector PostScript, GIF, PPM, BMP, PICT, Sun rasterfile
or as a MolScript input script or Kinemage.
<p>
The RasMol help facility can be accessed by typing &quot;help &lt;topic&gt;&quot; or &quot;help
&lt;topic&gt; &lt;subtopic&gt;&quot; from the command line. A complete list of RasMol commands
may be displayed by typing &quot;help commands&quot;. A single question mark may also
be used to abbreviate the keyword &quot;help&quot;.  Please type &quot;help notices&quot; for
important notices.
<p><hr>
<A name="chgenop"><H2>General Operation</H2></A>
<UL>
<LI><A HREF="#go_unix">Running RasMol Under UNIX or VMS</A>
<LI><A HREF="#go_mswin">Running RasMol Under Microsoft Windows</A>
<LI><A HREF="#go_mac">Running RasMol on the Apple Macintosh/PPC</A>
<LI><A HREF="#go_canvas">RasMol's Window</A>
<LI><A HREF="#go_mouse">Mouse Controls</A>
<LI><A HREF="#go_scrollbars">Scroll Bars</A>
<LI><A HREF="#go_picking">Picking</A>
<LI><A HREF="#go_dials">Dials Box</A>
<LI><A HREF="#go_cmdline">Command Line Interface</A>
<LI><A HREF="#go_dimensions">Dimensions within RasMol</A>
<LI><A HREF="#go_ini">Start-up Initialisation Files</A>
<LI><A HREF="#go_interprocess">Inter-Process Communication (IPC)</A>
</UL>
<A NAME="go_unix"></A>
<H3>Running RasMol Under UNIX or VMS</H3>
To start RasMol from either the UNIX or VMS prompt, type the command
'rasmol'. This command can be followed by an optional filename.
By default, immediately upon starting, the program displays the
following message to identify the version number and display depth
of the running program.  There will be some variation in this message
depending on your choice of platform:
<PRE>
     RasMol Molecular Renderer
     Roger Sayle, August 1995
     Copyright (C) Roger Sayle 1992-1999
     Version 2.7.1.1 January 2001
     Copyright (C) Herbert J. Bernstein 1998-2001
     *** See "help notice" for further notices ***
     [32-bit version]
</PRE>
Immediately underneath this banner message appears the program's
command line prompt 'RasMol&gt;'. If the program is being executed
under the X Window System, the program determines the type of the
display being used. If the screen has either an 8 bit or 24 bit
colour frame buffer, RasMol creates another window, which is used to
display menu options and the rendered images. If a suitable screen
is not available, RasMol may only be used from the command line.
Commands may be typed to manipulate the model, and to output the
generated image to a raster file. <BR>
<P>
If the program is run under the X Window System environment with
a suitable colour screen, RasMol creates an additional window to
display the rendered molecule interactively, as it is manipulated.
If RasMol is not run under the X Window System, the program displays
the message 'No suitable display detected!'. RasMol may be instructed
not to display a graphics window by using the command line option
'-nodisplay'. This is particularly useful for running RasMol as a
background or batch process. <BR>
<P>
It is possible to specify either a coordinate filename or a script
filename or both on the UNIX/VMS command line. A script file
may be specified by adding the option
'-script &lt;filename&gt;' to the command line.
A molecule coordinate file may be specified by placing
its name on the command line, optionally preceded by a file
format option. If no format option is given, the specified
coordinate file is assumed to be in PDB, CIF or mmCIF format.
Valid format options include '-pdb', '-mdl', '-mol2', '-xyz',
'- alchemy', '-charmm', '-mopac' and '-cif' which
correspond to Protein Data Bank format, Molecular Design Limited's
Mol file format, Tripos's Sybyl Mol2 file format, MSC's XMOL XYZ
file format, Tripos's Alchemy file format, CHARMm file format,
J. P. Stewart's MOPAC file format and IUCr CIF or mmCIF file format,
respectively.  If both a coordinate file and a script file are
specified on the command line, the molecule is loaded first, then
the script commands are applied to it. If either file is not found,
the program displays the error message 'Error: File not found!' and
the user is presented the RasMol prompt.<BR>
<P>
In order to leave RasMol, the user can type the command quit or
exit at the RasMol prompt, and the program will return the user
to the familiar unix prompt. Alternatively, if a prompt other
than the main RasMol prompt is being displayed, the user may
hit control-C (^C) to leave the program. The message '***Quit***'
will be output to the terminal, before the usual unix prompt is
redisplayed. The program may also be terminated by selecting
the Quit menu option, on the bottom of the main menu.
<P>
<A NAME="go_mswin"></A>
<H3>Running RasMol Under Microsoft Windows</H3>
To start RasMol under Microsoft Windows, double click on the
RasMol icon in the program manager. When RasMol first starts,
the program displays a single main window (the display window)
with a black background on the screen and provides the command
line window minimized as a small icon at the bottom of the screen.
The command line or terminal window may be opened by double clicking
on this RasMol icon. <BR>
<P>
It is possible to specify either a coordinate filename or a script
filename or both on the windows command line. A script file
may be specified by adding the option '-script &lt;filename&gt;'
to the command line. A molecule coordinate file may be specified
by placing its name on the command line, optionally preceded by
a file format option. If no format option is given, the specified
coordinate file is assumed to be in PDB, CIF or mmCIF format.
Valid format options include '-pdb', '-mdl', '-mol2', '-xyz',
'- alchemy', '-charmm', '-mopac' and '-cif' which
correspond to Protein Data Bank format, Molecular Design Limited's
Mol file format, Tripos's Sybyl Mol2 file format, MSC's XMOL XYZ
file format, Tripos's Alchemy file format, CHARMm file format,
J. P. Stewart's MOPAC file format and IUCr CIF or mmCIF file format,
respectively.  If both a coordinate file and a script file are
specified on the command line, the molecule is loaded first, then
the script commands are applied to it. If either file is not found,
the program displays the error message 'Error: File not found!' and
the user is presented the RasMol prompt.<BR>
<P>
<A NAME="go_mac"></A>
<H3>Running RasMol on the Apple Macintosh/PPC</H3>
To start RasMol on the Macintosh, double click on the RasMol
icon using Finder. When RasMol first starts, the program displays
two windows, the top window (with the black background) is the
graphics or canvas window and the window underneath it (with the
white background) is the RasMol command line window.
RasMol on the Macintosh may also be started by double clicking
on a file owned/created by the application with the signature
'RSML'. This will start up RasMol and pass the selected file to be
loaded. There is no way of specifying the file format on the command
line with a Macintosh so RasMol attempts to determine the file format
by inspecting the file's type signature. Files with type signature
'RSML' are assumed to be RasMol scripts, files of type 'mMOL' are
assumed to be MDL Mol files and all other types (principally 'TEXT')
are assumed to be in PDB format. Unlike other versions of RasMol
it is impossible to specify both a script and a coordinate file
simultaneously.
<P>
Dragging and dropping multi-file 'movie' scripts onto aliases or
copies of the RasMol application file may fail due to confusion
about which is the correct script folder. Double-clicking on a
script may lead to similar problems if copies are present.
<P>
Note that because on a Macintosh only one 'instance' of an
application may be running at any one time, if you were
double click on another file owned by 'RSML', the running
copy of RasMol would
<A HREF="#zap"><TT><B>zap</B></TT></A>
its molecule and load the newly specified file.
<P>
<A NAME="go_canvas"></A>
<H3>RasMol's Window</H3>
On all platforms RasMol displays two windows, the main <B>graphics</B>
or <B>canvas</B> window with a black background and a <B>command line</B>
or <B>terminal</B> window. At the top of the graphics window (or at the
top of the screen for the Macintosh) is the RasMol menu bar. The contents
of the menu bar change from platform to platform to support the local
user interface guidelines; however, all platforms support the 'File',
'Display', 'Colours', 'Export' and 'Options' pull-down menus.
The Main graphics window also has two scroll bars, one on the
 right and one at the bottom, that may be used to rotate the
 molecule interactively.
<P>
While the mouse pointer is located within the graphics area
of the main display window, the mouse pointer is drawn as a
cross-hair cursor, to enable the
'<A HREF="#go_picking"><TT><B>picking</B></TT></A>'
of objects being displayed; otherwise the mouse pointer is drawn as an
arrowhead. Any characters that are typed at the keyboard while the
display window is in 'focus' (meaning active or foreground) are
redirected to the command line in the terminal window. Hence you
do not need continually to switch focus between the command line
and graphics windows. <BR>
<BR>
The display window may be resized at any point during the session.
This has the effect of simply rescaling the image displayed on
the canvas. RasMol imposes limits on the size of the display window
such that the window must be large enough to display the menu and
scroll bars and yet small enough to fit on a single screen. Attempts
to enlarge the screen may fail owing to insufficient memory on the
host machine, in which case RasMol reports the error message
'Renderer Error: Unable to allocate frame buffer!' or some
similar error. <BR>
<BR>
On eight bit displays, when the number of colours required by
the program exceeds the number of free colours on the screen,
the program uses its own colourmap. This has the effect of
temporarily displaying all windows other than the display
window in false colours while the mouse pointer is within the
display windows. If the mouse pointer is moved outside the
display windows, the original colours of the other windows
return, and the image on the canvas is shown in 'false colour'.
Once the number of colours required by the program drops again,
the presentation of colours returns to normal.
<!--MZ2 go_mouse section added-->
<P>
<A NAME="go_mouse"></A>
<H3>Mouse Controls</H3>
Here is a summary of RasMol's mouse click-and-drag controls.
The '<A HREF="#setmouse"><TT><B>set mouse</B></TT></A>'
command mode defaults to '<B>set mouse rasmol</B>', which gives the controls
summarized below. However, there are also '<B>set mouse
insight</B>' and '<B>set mouse quanta</B>' modes (not shown below). <BR>
<BR>
<CENTER>
<TABLE  BORDER CELLPADDING=2>
<TR><TH>Action</TH> <TH>Windows</TH> <TH>Macintosh</TH></TR>
<TR><TD>Rotate X, Y</TD> <TD>Left</TD> <TD>Unmodified</TD>
</TR> <TR><TD>Translate X, Y</TD> <TD>Right</TD> <TD>Command*</TD> </TR>
<TR> <TD>Rotate Z</TD> <TD>Shift-Right</TD> <TD>Shift-Command*</TD> </TR>
<TR> <TD>Zoom</TD> <TD>Shift-Left</TD> <TD>Shift</TD> </TR>
<TR> <TD>Slab Plane</TD> <TD>Ctrl-Left</TD> <TD>Ctrl</TD> </TR>
<CAPTION ALIGN=BOTTOM>*On some Macs, the Option (Alt)
key has the same effect on RasMol as the Command key.</caption>
</TABLE>
</CENTER>
<P>
<HR>
<P>
<A NAME="go_scrollbars"></A>
<H3>Scroll Bars</H3>
The scroll bar across the bottom of the canvas area is used to rotate
the molecule about the y-axis, <I>i.e.</I> to spin the nearest point
on the molecule left or right; and the scroll bar to the right of
the canvas rotates the molecule about the x-axis, <I>i.e.</I> the
nearest point up or down. Each scroll bar has an 'indicator' to
denote the relative orientation of the molecule, which is initially
 positioned in the centre of the scroll bar. These scroll bars may be
 operated in either of two ways. The first is by clicking any mouse
 button on the dotted scroll bar background to indicate a direct
rotation relative to the current indicator position; the second
is by clicking one of the arrows at either end of the scroll bar
to rotate the molecule in fixed sized increments. Rotating the
molecule by the second method may cause the indicators on the scroll
bars to wrap around from one end of the bar to the other. A complete
 revolution is indicated by the indicator travelling the length of
the scroll bar. The angle rotated by using the arrows depends upon
the current size of the display window.
<A NAME="go_picking"></A>
<H3>Picking</H3>
In order to identify a particular atom or bond being displayed,
RasMol allows the users to 'pick' objects on the screen. The mouse
is used to position the cross-hair cursor over the appropriate
item, and then any of the mouse buttons is depressed.  Provided that
 the pointer is located close enough to a visible object, the program
determines the identity of the nearest atom to the point identified. <BR>
<P>
The program will display, in the terminal window, the atom's type,
serial number, residue name and residue number. If the atom is a
member of a named chain, the chain identifier is also displayed.
Two examples of the output generated by selecting an atom are
displayed below:
<PRE>
     Atom:  CA 349       Group:  SER 70
     Atom:  O  526       Hetero:  HOH 205    Chain:  P</PRE>
The first line describes the alpha carbon of the serine-70 amino
acid in a protein. The unique Protein Data Bank serial number for this
atom is 349. The following line describes the oxygen atom in a
water molecule attached to the P chain of the main molecule. The
word 'Hetero' distinguishes heterogeneous molecules (such as cofactors)
from the residues in the main molecule, noted by 'Group'. [These two
atoms are referred to by the two atom expressions 'SER70.CA'
and 'HOH205:P.O', respectively, when using the RasMol commands
'<TT><B>select</B></TT>' and '<TT><B>restrict</B></TT>'.]
<P>
Clicking the mouse on an atom can be used not only to
<B>identify</B> it, but also to find the <B>coordinates</B>, the
<B>distances</B> between
two atoms (or to display a <B>distance monitor</B>), <B>the
bond angle </B>defined by three atoms, the <B>torsion angle</B>
defined by four atoms, to toggle <B>labels</B> on or off,
or to specify the <B>centre of rotation</B>. See
the '<A HREF="#setpicking"><TT><B>set picking</B></TT></A>'
command for details.
<P>
<A NAME="go_dials"></A>
<H3>Dials Box</H3>
If RasMol detects a 'dials box' attached to the user's
workstation, it also allows the molecule to be manipulated
interactively by the dials. Once RasMol starts up, it labels
the LED displays above each dial, 'ROTATE X', 'ROTATE Y', 'ROTATE Z'
 and 'ZOOM' across the top row from left to right, and
'TRANS X', 'TRANS Y', 'TRANS Z' and 'SLAB' from left to right
across the bottom row. Rotating any of the knobs will
automatically transform and redisplay the molecule interactively.
The dials only have effect while the mouse pointer is within the
display window. If more than one application is using the dials
box at a time, care must be taken to remember the dial labels
assigned by each program, as each application may overwrite the
dial-label LEDS.
<BR>
<P>
The rotation about the X and Y axes automatically updates the
indicators on the appropriate scroll bars. All the rotation
dials rotate the molecule 180 degrees for a complete revolution
of the dial. All the remaining dials clamp their values to
permissible ranges; turning these dials past their limits
has no effect. The centre of rotation of the molecule may be
changed using the '<A HREF="#centre"><TT><B>centre</B></TT></A>'
command on the command line, or the command
'<A HREF="#setpicking"><TT><B>set
picking centre</B></TT></A>' followed by a
mouse click.
<P>
The 'ZOOM' dial allows the interactive zooming of the
molecule between 10% and 200% of the original default
magnification. Rotating the dial clockwise magnifies the
molecule and anticlockwise shrinks it. A complete
revolution of the dial corresponds to a 100% change in scale.
<P>
The 'SLAB' dial, which is only effective when slabbing is
enabled, allows the user to move the front z-clipping plane
from the nearest point on the molecule to the furthest. A
complete rotation of the SLAB dial corresponds to moving
the clipping plane half the distance between the front and
back of the molecule. Turning the SLAB knob clockwise moves
the clipping plane closer to the viewer (increasing the
number of objects displayed), and turning it anticlockwise
moves it further away (preventing more objects from being displayed).
Slabbing mode is enabled by typing the command '<B><TT>slab on</TT></B>'
 on the command line or toggling the slab option on the options menu.
<P>
Translation along the X and Y axis allows the centre of the molecule
to be moved within the canvas area of the screen. Rotation and zooming
are still performed relative to the centre of rotation and the molecule,
respectively, which may often not be at the centre of the canvas. The
TRANS Z dial currently has no effect.
<P>
<A NAME="go_cmdline"></A>
<H3>Command Line Interface</H3>
RasMol allows the execution of interactive commands typed at the
RasMol prompt in the terminal window. Characters typed into
either the terminal or the display window are processed on the
command line. Each command must be given on a separate line
terminated by a newline or carriage return character. Keywords
are case insensitive and may be entered in both lower and upper
case letters. All whitespace (space, tab and formfeed)
characters are ignored, except to separate the keyword and the
arguments of a command. Blank lines (those containing only
whitespace) are ignored. There is an internal restriction that
command lines are limited to a maximum of 256 characters.
Strings may be delimited by matching single or double quotation
marks. Placing a hash '#' character anywhere outside quotes
terminates the line. RasMol will ignore the rest of the line,
which may be used to comment on the command.
<p>
If a syntax error is detected on entering an interactive
command, RasMol indicates the location of the error on the
command line by placing the '^' character under the offending
word or character, and writing an error message on the following
line. If a command is not recognised by RasMol, the program will
generate an 'Unrecognised command!' error and redisplay the main
prompt. If surplus information is given at the end of a command
line, RasMol will execute the recognised command, but issue the
warning message 'Warning: Ignoring rest of command!'. Some
commands may prompt the user for more information. These
commands display a different prompt and are discussed in the
command reference.
<p>
Whenever RasMol outputs diagnostic or error messages to the
screen owing to selecting options from the menu or picking
objects on the screen, the current command line is cleared. The
prompt is redisplayed after any text has been displayed.
<p>
<H3>Command Line Editing</H3>
RasMol allows basic editing of the command line. Pressing
either backspace, delete or ^H (Control-H) will delete the
previous character, and the key ^D may be used to delete the
character under the cursor. Several characters may be used to
move the cursor along the command line. The
characters ^B, ^F, ^A and ^E move the cursor back a
single character, forward a single character, to the
beginning of the line and to the end of the line, respectively.
When the cursor is not at the end of the command line, typed
characters are inserted into the line and do not overwrite
existing characters. After a command line has been edited, a
newline or carriage return will enter the entire line, regardless
of where the cursor is positioned. Because RasMol is unable to
move the cursor up to the previous line, care must be taken
when editing commands that wrap over several lines. In the event
that another process overwrites or corrupts the command line,
the character ^L may be used to redisplay the line on the screen.
<P>
RasMol maintains a history of recently used commands, so that the
user never needs to type the same commands repeatedly.
Typing ^P (Control-P) on the command line will display the
previous command in the history and ^N will display the following
command. These commands may be edited using the features described
below. Moving forward or backward through the command history
 undoes the modifications made to the current line. The number
of commands retained in the history depends upon their length.
RasMol can retain more short command lines and fewer long ones. <BR>
<P>
Users with the Microsoft Windows version or the X windows version
and with 'vt100' or compatible terminals (such as an 'xterm') can
use the cursor control characters on the keyboard to abbreviate the
 control keys. The right and left cursor keys have the same affect
as ^F and ^B, moving the cursor forward and back a single character,
respectively. Similarly, the up and down cursor keys have the same
function as ^P and ^N, producing the previous and next entries in
 the command history, respectively.
<P>
<!-- Added the following paragraph. WM -->
Users with the Macintosh version can use the four 'arrow keys'
to move up and down through previous command line entries; and
back and forth within a single command line statement. Hitting
'return' or 'enter' at any time will result in the execution of
the current, <I>e.g.</I> selected or edited, command line contents.
<P>
<A NAME="go_dimensions"></A>
<H3>Dimensions within RasMol</H3>
All dimensions in RasMol, such as radii and distances, may be
specified in either 'RasMol units' or &Aring;ngstroms (&Aring;). The
RasMol units were first introduced to allow reasonably sized
values to be specified for most of the operations performed in
RasMol. A single RasMol unit corresponds to 1/250th of an
&Aring;ngstrom; therefore the most frequently used values are in
the hundreds. For this reason, if RasMol is given a distance
parameter that does not contain a decimal point, it is assumed
 to be in RasMol units. For example, the command 'spacefill 300'
specifies a sphere radius of 300 RasMol units, or 1.2 &Aring;. <BR>
<BR>
However, dimensions within RasMol can also be specified in
&Aring;ngstroms by placing a decimal point in the number. For
example, 'spacefill 1.2' specifies a sphere radius of 1.2 &Aring;.
This is particularly useful for the cut-off distance parameter in
<B><TT>within</B></TT> expressions.
<P>
<A NAME="go_ini"></A>
<H3>Start-up Initialisation Files</H3>
Each time RasMol is started, it searches for an initialisation
file of commands to run before the command prompt is presented
to the user. The file is called <B>.rasmolrc</B> on UNIX systems,
and <B>RASMOL.INI</B> on VMS and Microsoft Windows Systems. The
format and execution of this file is identical to that of the
 RasMol script command. <BR>
<P>
RasMol first looks for the initialisation file in the current
directory and if it is not found will look for it in the user's
home directory. On all systems the environment variable
<B>HOME</B> may be used to name the user's home
directory. If no personal initialisation file is found the
 program looks for the file <B>rasmolrc</B> (or <B>RASMOLRC</B>)
in the RasMol system directory pointed to by the environment
 variable <B>RASMOLPATH</B>. This directory should also contain
the on-line help file <B>rasmol.hlp</B>. On UNIX systems
RASMOLPATH is typically set to be '/usr/local/lib/rasmol'.
<BR>
<P>
Unlike the command
'<a href="#script"><tt><b>script &quot;.rasmolrc&quot;</b></tt></a>',
the program
will not generate an error message if the file is not found.
The system rasmolrc file is commonly used by system managers to
display information about the local installation and who to
contact for help.  Such system rasmolrc files will contain RasMol
'<tt><b>echo</b></tt>'
commands detailing a
telephone number or e-mail address to be used for contacting
somebody for local assistance.
<P>
<A NAME="go_interprocess"></A>
<H3>Inter-Process Communication</H3>
RasMol supports Inter Process Communication (IPC) in one form or
 another on all platforms. Under Microsoft Windows, IPC is
implemented using Dynamic Data Exchange (DDE), on the
MacIntosh IPC is implemented using Apple Events and on X Windows
systems IPC is implemented using John Ousterhaut's Tcl/Tk
 communication protocol. <BR>
<P>
When RasMol starts up on an <B>X window</B> system it
registers itself with the X window Server as a <B>Tcl
interpreter</B>. From within a Tcl application such as
'wish', you can use the Tcl command 'winfo interps' to
determine the currently register interpreters on that display.
The first instance of RasMol registers itself as 'rasmol', the
 second as 'rasmol #2', the third as 'rasmol #3' and so on. The
Tcl interpreter can easily send a command to rasmol using the
built-in 'send' command. RasMol interprets the string parameter
to the send command not as a Tcl function to execute but as a
RasMol command. Hence, typing 'send {rasmol} {background red}'
into the wish interpreter will cause RasMol's display window to
change colour. Using the same encoding as Microsoft's DDE Execute
protocol, multiple commands may be sent in a single 'send' by
placing the consecutive commands in square brackets. RasMol will
execute all of the commands in a 'send' before refreshing the screen. <BR>
<P>
Under <B>Microsoft Windows</B>, RasMol supports a complete DDE protocol.
 The simplest layers of the protocol may be accessed by sending a DDE
Execute command to application 'RasWin' and any topic. This will start a
DDE conversation
with the most recently launched instance of RasMol.
Although any topic name can be used, the use of
'System' and/or 'RemoteControl' are recommended.
Once again the contents of the execute package consists of a
string for RasMol to execute. If the first non-whitespace
character is an open bracket, the string is interpreted to be a
sequence of consecutive commands enclosed in square brackets;
otherwise the string consists of just a single command. Commands in
square brackets may optionally be separated by whitespace and/or
 semi- colons. RasMol can also act as a 'data server' supporting
 hot, cold and warm links. Currently supported DDE items include
'Name', 'Image', 'Pick', 'Count' which denotes the Molecule name,
 the currently displayed image (in Microsoft DIB format), the atom
 expression of the last picked atom (or an empty string) and the
 number of selected atoms, respectively. Using a hot or warm link
 on the 'Pick' item, for example, allows an application such as
Microsoft Word, Excel or Visual Basic to respond each time the
user clicks on an atom in RasMol. <BR>
<P>
RasMol on the <B>Apple Macintosh</B> supports <B>AppleEvents</B>.
Currently the only supported AppleEvents are the four 'core' events,
Open Application, Open Document, Print Document and Quit. However,
because OpenDocument determines its actions by the file's type
signature this can be used to implement generic IPC. Because RasMol
for the Macintosh treats all files of type 'RSML' as scripts, the
sending application need only place all the commands to be executed
in a temporary file, set the type of the file to 'RSML' and then
send RasMol an OpenDocument AppleEvent with the file as parameter. <BR>
<p><hr>
<a name="chcomref"><h2>Command Reference</h2></a>
RasMol allows the execution of interactive commands typed at the
'<tt><b>RasMol&gt;</b></tt>'
prompt in the terminal window. Each command must be given on
a separate line. Keywords are case insensitive and may be entered in
either upper or lower case letters. All whitespace characters are
ignored except to separate keywords and their arguments.
<p>
The commands/keywords currently recognised by RasMol are given below.
<p>
<center>
<table BORDER CELLPADDING=2>
<tr><td><a href="#backbone">Backbone</a>
<td><a href="#background">Background</a>
<td><a href="#cartoon">Cartoon</a>
<td><a href="#centre">Centre</a>
<td><a href="#clipboard">Clipboard</a>
<td><a href="#colour">Colour</a>
<td><a href="#connect">Connect</a>
<td><a href="#spacefill">CPK</a>
<tr><td><a href="#define">Define</a>
<td><a href="#dots">Dots</a>
<td><a href="#echo">Echo</a>
<td><a href="#english">English</a>
<td><a href="#quit">Exit</a>
<td><a href="#hbonds">HBonds</a>
<td><a href="#help">Help</a>
<td><a href="#label">Label</a>
<tr><td><a href="#load">Load</a>
<td><a href="#monitor">Monitor</a>
<td><a href="#pause">Pause</a>
<td><a href="#print">Print</a>
<td><a href="#quit">Quit</a>
<td><a href="#refresh">Refresh</a>
<td><a href="#renumber">Renumber</a>
<td><a href="#reset">Reset</a>
<tr><td><a href="#restrict">Restrict</a>
<td><a href="#ribbons">Ribbons</a>
<td><a href="#rotate">Rotate</a>
<td><a href="#save">Save</a>
<td><a href="#script">Script</a>
<td><a href="#select">Select</a>
<td><a href="#set">Set</a>
<td><a href="#show">Show</a>
<tr><td><a href="#slab">Slab</a>
<td><a href="#script">Source</a>
<td><a href="#spacefill">Spacefill</a>
<td><a href="#spanish">Spanish</a>
<td><a href="#ssbonds">SSBonds</a>
<td><a href="#star">Star</a>
<td><a href="#stereo">Stereo</a>
<td><a href="#strands">Strands</a>
<tr><td><a href="#structure">Structure</a>
<td><a href="#backbone">Trace</a>
<td><a href="#translate">Translate</a>
<td><a href="#wireframe">Wireframe</a>
<td><a href="#write">Write</a>
<td><a href="#zap">Zap</a>
<td><a href="#zoom">Zoom</a>
<td cellspan=1>
</table>
</center><hr>
<a name="backbone"><h3>Backbone</h3></a><p>
<pre>
Syntax:  backbone {&lt;boolean&gt;}
         backbone &lt;value&gt;
         backbone dash

</pre><p>
The RasMol
'<tt><b>backbone</b></tt>'
command permits the representation of a polypeptide
backbone as a series of bonds connecting the adjacent alpha carbons of
each amino acid in a chain. The display of these backbone 'bonds' is
turned on and off by the command parameter in the same way as with the
'<a href="#wireframe"><tt><b>wireframe</b></tt></a>'
command. The command
'<tt><b>backbone off</b></tt>'
turns off the selected 'bonds', and
'<tt><b>backbone on</b></tt>'
or with a number turns them on. The number can be used
to specify the cylinder radius of the representation in either &Aring;ngstrom
or RasMol units. A parameter value of 500 (2.0 &Aring;ngstroms) or above
results in a &quot;Parameter value too large&quot; error. Backbone objects may be
coloured using the RasMol
'<a href="#colour"><tt><b>colour backbone</b></tt></a>'
command.
<p>
The reserved word backbone is also used as a predefined set (&quot;help sets&quot;)
and as a parameter to the
'<a href="#sethbonds"><tt><b>set hbond</b></tt></a>'
and
'<a href="#setssbonds"><tt><b>set ssbond</b></tt></a>'
commands. The
RasMol command
'<tt><b>trace</b></tt>'
renders a smoothed backbone, in contrast to
'<tt><b>backbone</b></tt>'
which connects alpha carbons with straight lines.
<p>
The backbone may be displayed with dashed lines by use of the
'<tt><b>backbone dash</b></tt>'
command.
<p><hr>
<a name="background"><h3>Background</h3></a><p>
<pre>
Syntax:  background &lt;colour&gt;
</pre><p>
The RasMol
'<tt><b>background</b></tt>'
command is used to set the colour of the &quot;canvas&quot; background. The
colour may be given as either a colour name or a comma separated
triple of Red, Green and Blue (RGB) components enclosed in square
brackets. Typing the command
'<a href="#help"><tt><b>help colours</b></tt></a>'
will give a list of the predefined colour names recognised by RasMol.
When running under X Windows, RasMol also recognises colours in the
X server's colour name database.
<p>
The
'<tt><b>background</b></tt>'
command is synonymous with the RasMol
'<a href="#setbackground"><tt><b>set background</b></tt></a>'
command.
<p><hr>
<a name="cartoon"><h3>Cartoon</h3></a><p>
<pre>
Syntax:  cartoon {&lt;number&gt;}
</pre><p>
The RasMol
'<tt><b>cartoon</b></tt>'
command does a display of a molecule
'<a href="#ribbons"><tt><b>ribbons</b></tt></a>'
as Richardson (MolScript) style protein
'<tt><b>cartoons</b></tt>',
implemented as thick (deep) ribbons. The
easiest way to obtain a cartoon representation of a protein is
to use the
'<tt><b>Cartoons</b></tt>'
option on the
'<tt><b>Display</b></tt>'
menu. The
'<tt><b>cartoon</b></tt>'
command represents the currently selected residues
as a deep ribbon with width specified by the command's argument.
Using the command without a parameter results in the ribbon's
width being taken from the protein's secondary structure,
as described in the
'<a href="#ribbons"><tt><b>ribbons</b></tt></a>'
command. By default, the C-termini of beta-sheets are displayed
as arrow heads. This may be enabled and disabled using the
'<a href="#setcartoon"><tt><b>set cartoons</b></tt></a>'
command.
The depth of the cartoon may be adjusted using the
'<a href="#setcartoon"><tt><b>set cartoons &lt;number&gt;</b></tt></a>'
command. The
'<a href="#setcartoon"><tt><b>set cartoons</b></tt></a>'
command without any parameters returns these two options
to their default values.
<p><hr>
<a name="centre"><h3>Centre</h3></a><p>
<pre>
Syntax:  centre {&lt;expression&gt;}
         center {&lt;expression&gt;}
</pre><p>
The RasMol
'<tt><b>centre</b></tt>'
command defines the point about which the
'<a href="#rotate"><tt><b>rotate</b></tt></a>'
command and the scroll bars rotate the current molecule. Without a
parameter the centre command resets the centre of rotation to be the
centre of gravity of the molecule. If an atom expression is specified,
RasMol rotates the molecule about the centre of gravity of the set of
atoms specified by the expression. Hence, if a single atom is specified
by the expression, that atom will remain 'stationary' during rotations.
<p>
Type
'<a href="#help"><tt><b>help expression</b></tt></a>'
for more information on RasMol atom expressions.
<p>
Alternatively the centring may be given as a comma separated triple of
[CenX, CenY, CenZ] offsets in RasMol units (1/250 of an &Aring;ngstrom) from
the centre of gravity.  The triple must be enclosed in square brackets.
<p><hr>
<a name="clipboard"><h3>Clipboard</h3></a><p>
<pre>
Syntax:  clipboard
</pre><p>
The RasMol
'<tt><b>clipboard</b></tt>'
command places a copy of the currently displayed image on the local
graphics 'clipboard'. Note: this command is not yet supported on
UNIX or VMS machines. It is intended to make transfering images
between applications easier under Microsoft Windows or on an Apple
Macintosh.
<p>
When using RasMol on a UNIX or VMS system this functionality may be
achieved by generating a raster image in a format that can be read
by the receiving program using the RasMol
'<a href="#write"><tt><b>write</b></tt></a>'
command.
<p><hr>
<a name="colour"><h3>Colour</h3></a><p>
<pre>
Syntax:  colour {&lt;object&gt;} &lt;colour&gt;
         color {&lt;object&gt;} &lt;colour&gt;
</pre><p>
Colour the atoms (or other objects) of the selected region. The colour may
be given as either a colour name or a comma separated triple of Red, Green
and Blue (RGB) components enclosed in square brackets. Typing the command
'<a href="#help"><tt><b>help colours</b></tt></a>'
will give a list of all the predefined colour names recognised
by RasMol.
<p>
Allowed objects are
'<tt><b>atoms</b></tt>',
'<tt><b>bonds</b></tt>',
'<a href="#backbone"><tt><b>backbone</b></tt></a>',
'<a href="#ribbons"><tt><b>ribbons</b></tt></a>',
'<a href="#label"><tt><b>labels</b></tt></a>',
'<a href="#dots"><tt><b>dots</b></tt></a>',
'<a href="#hbonds"><tt><b>hbonds</b></tt></a>'
and
'<a href="#ssbonds"><tt><b>ssbonds</b></tt></a>'.
If no object is specified, the default keyword
'<tt><b>atom</b></tt>'
is assumed.
Some colour schemes are defined for certain object types. The colour scheme
'<tt><b>none</b></tt>'
can be applied to all objects except atoms and dots, stating that the selected
objects have no colour of their own, but use the colour of their associated
atoms (<i>i.e.</i> the atoms they connect).
'<tt><b>Atom</b></tt>'
objects can also be coloured by
'<a href="#altcolours"><tt><b>alt</b></tt></a>',
'<a href="#aminocolours"><tt><b>amino</b></tt></a>',
'<a href="#chaincolours"><tt><b>chain</b></tt></a>',
'<a href="#chargecolours"><tt><b>charge</b></tt></a>',
'<a href="#cpkcolours"><tt><b>cpk</b></tt></a>',
'<a href="#groupcolours"><tt><b>group</b></tt></a>',
'<a href="#nmrmodelcolours"><tt><b>model</b></tt></a>',
'<a href="#shapelycolours"><tt><b>shapely</b></tt></a>',
'<a href="#structurecolours"><tt><b>structure</b></tt></a>',
'<a href="#temperaturecolours"><tt><b>temperature</b></tt></a>'
or
'<a href="#usercolours"><tt><b>user</b></tt></a>'.
Hydrogen bonds can also be coloured by
'<a href="#hbondtypecolours"><tt><b>type</b></tt></a>'
and dot surfaces can also be coloured by
'<a href="#potentialcolours"><tt><b>electrostatic potential</b></tt></a>'.
For more information type
'<a href="#chcolours"><tt><b>help colour &lt;colour&gt;</b></tt></a>'.
<p><hr>
<a name="connect"><h3>Connect</h3></a><p>
<pre>
Syntax:  connect {&lt;boolean&gt;}
</pre><p>
The RasMol
'<tt><b>connect</b></tt>'
command is used to force RasMol to (re)calculate the connectivity
of the current molecule.
If the original input file contained connectivity information, this
is discarded. The command
'<tt><b>connect false</b></tt>'
uses a fast heuristic
algorithm that is suitable for determining bonding in large
bio-molecules such as proteins and nucleic acids. The command
&quot;<tt><b>connect true</b></tt>&quot;
uses a slower more accurate algorithm based upon
covalent radii that is more suitable to small molecules containing
inorganic elements or strained rings. If no parameters are given,
RasMol determines which algorithm to use based on the number of atoms
in the input file. Greater than 255 atoms causes RasMol to use the
faster implementation. This is the method used to determine bonding,
if necessary, when a molecule is first read in using the
'<a href="#load"><tt><b>load</b></tt></a>'
command.
<p><hr>
<a name="define"><h3>Define</h3></a><p>
<pre>
Syntax:  define &lt;identifier&gt; &lt;expression&gt;
</pre><p>
The RasMol
'<tt><b>define</b></tt>'
command allows the user to associate an arbitrary set of atoms with a
unique identifier. This allows the definition of user-defined sets. These
sets are declared statically, <i>i.e.</i> once defined the contents of the set
do not change, even if the expression defining them depends on the
current transformation and representation of the molecule.
<p><hr>
<a name="dots"><h3>Dots</h3></a><p>
<pre>
Syntax:  dots {&lt;boolean&gt;}
         dots &lt;value&gt;
</pre><p>
The RasMol
'<tt><b>dots</b></tt>'
command is used to generate a van der Waals' dot surface around the
currently selected atoms. Dot surfaces display regularly spaced points
on a sphere of van der Waals' radius about each selected atom. Dots that
would are 'buried' within the van der Waals' radius of any other atom
(selected or not) are not displayed.
The command
'<tt><b>dots on</b></tt>'
deletes any existing dot surface and generates a dots surface around
the currently selected atom set with a default dot density of 100. The
command
'<tt><b>dots off</b></tt>'
deletes any existing dot surface. The dot density may be
specified by providing a numeric parameter between 1 and 1000. This
value approximately corresponds to the number of dots on the surface
of a medium sized atom.
<p>
By default, the colour of each point on a dot surface is the colour
of its closest atom at the time the surface is generated. The colour
of the whole dot surface may be changed using the
'<a href="#colour"><tt><b>colour dots</b></tt></a>'
command.
<p><hr>
<a name="echo"><h3>Echo</h3></a><p>
<pre>
Syntax:  echo {&lt;string&gt;}
</pre><p>
The RasMol
'<tt><b>echo</b></tt>'
command is used to display a message in the RasMol command/terminal
window. The string parameter may optionally be delimited in double
quote characters. If no parameter is specified, the
'<tt><b>echo</b></tt>'
command displays a blank line. This command is particularly useful
for displaying text from within a RasMol
'<a href="#script"><tt><b>script</b></tt></a>'
file.
<p><hr>
<a name="english"><h3>English</h3></a><p>
<pre>
Syntax:  English
</pre><p>
The RasMol
'<tt><b>English</b></tt>'
command sets the menus and messages to the English versions.
The command
'<a href="#spanish"><tt><b>Spanish</b></tt></a>'
may be used to select Spanish menus and messages.
<p><hr>
<a name="hbonds"><h3>HBonds</h3></a><p>
<pre>
Syntax:  hbonds {&lt;boolean&gt;}
         hbonds &lt;value&gt;
</pre><p>
The RasMol
'<tt><b>hbond</b></tt>'
command is used to represent the hydrogen bonding of the protein
molecule's backbone. This information is useful in assessing the
protein's secondary structure. Hydrogen bonds are represented as
either dotted lines or cylinders between the donor and acceptor
residues. The first time the
'<tt><b>hbond</b></tt>'
command is used, the program searches the structure of the
molecule to find hydrogen bonded residues and reports the number of bonds
to the user. The command
'<tt><b>hbonds on</b></tt>'
displays the selected 'bonds' as dotted lines, and the
'<tt><b>hbonds off</b></tt>'
turns off their display. The colour of hbond objects may be changed
by the
'<a href="#colour"><tt><b>colour hbond</b></tt></a>'
command. Initially, each hydrogen bond has the colours of its connected
atoms.
<p>
By default the dotted lines are drawn between the accepting oxygen and
the donating nitrogen. By using the
'<a href="#sethbonds"><tt><b>set hbonds</b></tt></a>'
command the alpha carbon positions of the appropriate residues may be
used instead. This is especially useful when examining proteins in
backbone representation.
<p><hr>
<a name="help"><h3>Help</h3></a><p>
<pre>
Syntax:  help {&lt;topic&gt; {&lt;subtopic&gt;}}
         ? {&lt;topic&gt; {&lt;subtopic&gt;}}
</pre><p>
The RasMol
'<tt><b>help</b></tt>'
command provides on-line help on the given topic.
<p><hr>
<a name="label"><h3>Label</h3></a><p>
<pre>
Syntax:  label {&lt;string&gt;}
         label &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>label</b></tt>'
command allows an arbitrary formatted text string to be
associated with each currently selected atom.  This string may contain
embedded 'expansion specifiers' which display properties of the atom
being labelled. An expansion specifier consists of a '%' character
followed by a single alphabetic character specifying the property to be
displayed (similar to C's printf syntax).
An actual '%' character may be displayed by using the expansion
specifier '%%'.
<p>
Atom labelling for the currently selected atoms may be turned off with
the command
'<tt><b>label off</b></tt>'.
By default, if no string is given as a parameter, RasMol uses labels
appropriate for the current molecule.
RasMol uses the label '%n%r:%c.%a' if the molecule contains more than
one chain, '%e%i' if the molecule has only a single residue (a small
molecule) and '%n%r.%a' otherwise.
<p>
The colour of each label may be changed using the
'<a href="#colour"><tt><b>colour label</b></tt></a>'
command. By default, each label is drawn in the same colour as the atom
to which it is attached. The size and spacing of the displayed text
may be changed using the
'<a href="#setfontsize"><tt><b>set fontsize</b></tt></a>'
command.  The width of the strokes in the displayed text may be changed
 using the
'<a href="#setfontstroke"><tt><b>set fontstroke</b></tt></a>'
 command.
<p>
The following table lists the current expansion specifiers:
<p>
<pre>
    %a      Atom Name
    %b %t   B-factor/Temperature
    %c %s   Chain Identifier
    %e      Element Atomic Symbol
    %i      Atom Serial Number
    %n      Residue Name
    %r      Residue Number
    %M      NMR Model Number (with leading "/")
    %A      Alternate Conformation Identifier (with leading ";")
</pre>
<p><hr>
<a name="load"><h3>Load</h3></a><p>
<pre>
Syntax:  load {&lt;format&gt;} &lt;filename&gt;
</pre><p>
Load a molecule coordinate file into RasMol. Valid molecule file
formats are
'<tt><b>pdb</b></tt>'
(Protein Data Bank format),
'<tt><b>mdl</b></tt>'
(Molecular Design Limited's MOL file format),
'<tt><b>alchemy</b></tt>'
(Tripos' Alchemy file format),
'<tt><b>mol2</b></tt>'
(Tripos' Sybyl Mol2 file format),
'<tt><b>charmm</b></tt>'
(CHARMm file format),
'<tt><b>xyz</b></tt>'
(MSC's XMol XYZ file format),
'<tt><b>mopac</b></tt>'
(J. P. Stewart's MOPAC file format) or
'<tt><b>cif</b></tt>'
(IUCr CIF or mmCIF file format). If no file format is specified,
'<tt><b>PDB</b></tt>',
'<tt><b>CIF</b></tt>',
or
'<tt><b>mmCIF</b></tt>'
is assumed by default. Only a single molecule may be loaded at a time.
To delete a molecule prior to loading another use the RasMol
'<a href="#zap"><tt><b>zap</b></tt></a>'
command.
<p>
The
'<tt><b>load</b></tt>'
command selects all the atoms in the molecule, centres it on the
screen and renders it as a CPK coloured wireframe model. If the molecule
contains no bonds (<i>i.e.</i> contains only alpha carbons), it is drawn as
an alpha carbon backbone. If the file specifies fewer bonds than atoms,
RasMol determines connectivity using the
'<a href="#connect"><tt><b>connect</b></tt></a>'
command.
<p>
The
'<tt><b>load inline</b></tt>'
command also allows the storing of atom coordinates in scripts
to allow better integration with WWW browsers. A load command
executed inside a script file may specify the keyword
'<tt><b>inline</b></tt>'
instead of a conventional filename. This option specifies that
the coordinates of the molecule to load are stored in the same
file as the currently executing commands.
<p>
Typically this is used in the command
'<tt><b>load pdb inline</b></tt>',
which is followed by a number of RasMol commands terminated by
the command
'<a href="#quit"><tt><b>exit</b></tt></a>'.
The
'<tt><b>exit</b></tt>'
command terminates execution of the current script and returns
control to the command line (or the calling script). This means
any lines following
'<tt><b>exit</b></tt>'
are never interpreted by RasMol. These may be used to store
atomic coordinates in PDB, CIF or mmCIF file format.
One possible use is a standard RasMol script prefix that may be
concatenated with an appropriate PDB file on-the-fly.
<p><hr>
<a name="monitor"><h3>Monitor</h3></a><p>
<pre>
Syntax:  monitor &lt;number&gt;  &lt;number&gt;
         monitor {&lt;boolean&gt;}
</pre><p>
The RasMol
'<tt><b>monitor</b></tt>'
command allows the display of distance monitors. A distance
monitor is a dashed (dotted) line between an arbitrary pair
of atoms, optionally labelled by the distance between them.
The RasMol command
'<tt><b>monitor &lt;number&gt; &lt;number&gt;</b></tt>'
adds such a distance monitor between the two atoms specified by the atom
serial numbers given as parameters
<p>
Distance monitors are turned off with the command
'<tt><b>monitors off</b></tt>'.
By default, monitors display the
distance between its two end points as a label at the centre of
the monitor. These distance labels may be turned off with the
command
'<a href="#setmonitor"><tt><b>set monitors off</b></tt></a>',
and re-enabled with the command
'<a href="#setmonitor"><tt><b>set monitors on</b></tt></a>'.
Like most other representations,
the colour of a monitor is taken from the colour of its end points unless
specified by the
'<a href="#colour"><tt><b>colour monitors</b></tt></a>'
command.
<p>
Distance monitors may also be added to a molecule interactively with
the mouse, using the
'<a href="#setpicking"><tt><b>set picking monitor</b></tt></a>'
command. Clicking on an atom results
in its being identified on the rasmol command line. In addition
every atom picked increments a modulo counter such that, in monitor
mode, every second atom displays the distance between this atom and
the previous one. The shift key may be used to form distance monitors
between a fixed atom and several consecutive positions. A distance
monitor may also be removed (toggled) by selecting the appropriate
pair of atom end points a second time.
<p><hr>
<a name="pause"><h3>Pause</h3></a><p>
<pre>
Syntax:  pause
         wait
</pre><p>
The RasMol
'<tt><b>pause</b></tt>'
command is used in script files to stop the script file for local
manipulation by a mouse, until any key is pushed to restart the
script file.
'<tt><b>Wait</b></tt>'
is synonymous with
'<tt><b>pause</b></tt>'.
This command may be executed in RasMol script files to suspend
the sequential execution of commands and allow the user to examine
the current image.   When RasMol executes a
'<tt><b>pause</b></tt>'
command in a script file, it suspends  execution of the rest
of the file, refreshes the image on the screen and allows the
manipulation of the image using the mouse and scroll  bars, or
resizing of the graphics window.  Once a key is pressed, control
returns to the script file at the line following the
'<tt><b>pause</b></tt>'
command.  While a script is suspended the molecule may be rotated,
translated, scaled, slabbed and picked as usual, but all menu
commands are disabled.
The
'<tt><b>pause</b></tt>'
can probably
be used most effectively with
'<a href="#echo"><tt><b>echo</b></tt></a>'
commands in education pre-scripted demonstrations,
where a description of the current image is presented  to
the user/student. Typically the command before a
'<tt><b>pause</b></tt>'
should  be
'<tt><b>echo Press any key to continue</b></tt>'.
<p>
Execution of a script can be cancelled by pressing Control-D or
Control-Z (on VAX/VMS, Control-C) while standing at a pause.
The command
'<a href="#setpicking"><tt><b>set picking none</b></tt></a>'
disables picking, which avoids the display of spurious messages
whilst a script is suspended at a pause.
<p><hr>
<a name="print"><h3>Print</h3></a><p>
<pre>
Syntax:  print
</pre><p>
The RasMol
'<tt><b>print</b></tt>'
command sends the currently displayed image to the local default printer
using the operating system's native printer driver. Note: this command
is not yet supported under UNIX or VMS. It is intended to take advantage
of Microsoft Windows and Apple Macintosh printer drivers. For example,
this allows images to be printed directly on a dot matrix printer.
<p>
When using RasMol on a UNIX or VMS system this functionality may be
achieved by either generating a PostScript file using the RasMol
'<a href="#write"><tt><b>write ps</b></tt></a>'
or
'<a href="#write"><tt><b>write vectps</b></tt></a>'
commands and printing that or generating a raster image file and using a
utility to dump that to the local printer.
<p><hr>
<a name="quit"><h3>Quit</h3></a><p>
<pre>
Syntax:  quit
         exit
</pre><p>
Exit from the RasMol program. The RasMol commands
'<tt><b>exit</b></tt>'
and
'<tt><b>quit</b></tt>'
are synonymous, except within nested scripts.  In that case,
'<tt><b>exit</b></tt>'
terminates only the current level, while
'<tt><b>quit</b></tt>'
terminates all nested levels of scripts.
<p><hr>
<a name="refresh"><h3>Refresh</h3></a><p>
<pre>
Syntax:  refresh
</pre><p>
The RasMol
'<tt><b>refresh</b></tt>'
command redraws the current image.  This is useful in scripts
to ensure application of a complex list of parameter changes.
<p><hr>
<a name="renumber"><h3>Renumber</h3></a><p>
<pre>
Syntax:  renumber {{-} &lt;value&gt;}
</pre><p>
The RasMol
'<tt><b>renumber</b></tt>'
command sequentially numbers the residues in a macromolecular chain.
The optional parameter specifies the value of the first residue in the
sequence. By default, this value is one. For proteins,
each amino acid is numbered consecutively from the N terminus to the C
terminus. For nucleic acids, each base is numbered from the 5' terminus
to the 3' terminus. All chains in the current database are renumbered and gaps
in the original sequence are ignored. The starting value for numbering may
be negative.
<p><hr>
<a name="reset"><h3>Reset</h3></a><p>
<pre>
Syntax:  reset
</pre><p>
The RasMol
'<tt><b>reset</b></tt>'
command restores the original viewing transformation
and centre of rotation. The scale is set to its default value,
'<a href="#zoom"><tt><b>zoom 100</b></tt></a>',
the centre of rotation is set to the geometric centre of the currently
loaded molecule,
'<a href="#centre"><tt><b>centre all</b></tt></a>',
this centre is translated to the middle of the screen and
the viewpoint set to the default orientation.
<p>
This command should not be mistaken for the RasMol
'<a href="#zap"><tt><b>zap</b></tt></a>'
command which deletes the currently stored molecule, returning the
program to its initial state.
<p><hr>
<a name="restrict"><h3>Restrict</h3></a><p>
<pre>
Syntax:  restrict {&lt;expression&gt;}
</pre><p>
The RasMol
'<tt><b>restrict</b></tt>'
command both defines the currently selected region of the
molecule and disables the representation of (most of) those parts of the
molecule no longer selected.  All subsequent RasMol commands that modify
a molecule's colour or representation affect only the currently selected
region. The parameter of a
'<tt><b>restrict</b></tt>'
command is a RasMol atom expression that is evaluated for every atom
of the current molecule. This command is very similar to the RasMol
'<a href="#select"><tt><b>select</b></tt></a>'
command, except
'<tt><b>restrict</b></tt>'
disables the
'<a href="#wireframe"><tt><b>wireframe</b></tt></a>',
'<a href="#spacefill"><tt><b>spacefill</b></tt></a>'
and
'<a href="#backbone"><tt><b>backbone</b></tt></a>'
representations in the non-selected region.
<p>
Type &quot;help expression&quot; for more information on RasMol atom expressions.
<p><hr>
<a name="ribbons"><h3>Ribbons</h3></a><p>
<pre>
Syntax:  ribbons {&lt;boolean&gt;}
         ribbons &lt;value&gt;
</pre><p>
The RasMol
'<tt><b>ribbons</b></tt>'
command displays the currently loaded protein or nucleic acid as a
smooth solid &quot;ribbon&quot; surface passing along the backbone of the protein.
The ribbon is drawn between each amino acid whose alpha carbon is
currently selected. The colour of the ribbon is changed by the RasMol
'<a href="#colour"><tt><b>colour ribbon</b></tt></a>'
command. If the current ribbon colour is
'<tt><b>none</b></tt>'
(the default), the colour is taken from the alpha carbon at each
position along its length.
<p>
The width of the ribbon at each position is determined by the optional
parameter in the usual RasMol units. By default the width of the ribbon
is taken from the secondary structure of the protein or a constant value
of 720 (2.88 &Aring;ngstroms) for nucleic acids.
The default width of protein alpha helices and beta sheets is 380 (1.52
&Aring;ngstroms) and 100 (0.4 &Aring;ngstroms) for turns and random coil. The
secondary structure assignment is either from the PDB file or calculated
using the DSSP algorithm as used by the
'<a href="#structure"><tt><b>structure</b></tt></a>'
command. This command is similar to the RasMol command
'<a href="#strands"><tt><b>strands</b></tt></a>'
which renders the biomolecular ribbon as parallel depth-cued curves.
<p><hr>
<a name="rotate"><h3>Rotate</h3></a><p>
<pre>
Syntax:  rotate &lt;axis&gt; {-} &lt;value&gt;
</pre><p>
Rotate the molecule about the specified axis.
Permitted values for the axis parameter are
&quot;<tt><b>x</b></tt>&quot;, &quot;<tt><b>y</b></tt>&quot; and &quot;<tt><b>z</b></tt>&quot;.
The integer parameter states the angle in degrees for the structure to
be rotated. For the X and Y axes, positive values move the closest point
up and right, and negative values move it down and left, respectively. For
the Z axis, a positive rotation acts clockwise and a negative angle
anti-clockwise.
<p><hr>
<a name="save"><h3>Save</h3></a><p>
<pre>
Syntax:  save {pdb} &lt;filename&gt;
         save mdl &lt;filename&gt;
         save alchemy &lt;filename&gt;
         save xyz &lt;filename&gt;
</pre><p>
Save the currently selected set of atoms in a Protein
Data Bank (PDB), MDL, Alchemy(tm) or XYZ format file.
The distinction between this command and the RasMol
'<a href="#write"><tt><b>write</b></tt></a>'
command has been dropped. The only difference is that without a format
specifier the
'<tt><b>save</b></tt>'
command generates a
'<tt><b>PDB</b></tt>'
file and the
'<a href="#write"><tt><b>write</b></tt></a>'
command generates a
'<tt><b>GIF</b></tt>'
image.
<p><hr>
<a name="script"><h3>Script</h3></a><p>
<pre>
Syntax:  script &lt;filename&gt;
</pre><p>
The RasMol
'<tt><b>script</b></tt>'
command reads a set of RasMol commands sequentially from a
text file and executes them. This allows sequences of commonly used
commands to be stored and performed by single command. A RasMol script
file may contain a further script command up to a maximum &quot;depth&quot; of 10,
allowing complicated sequences of actions to be executed. RasMol
ignores all characters after the first '#' character on each line
allowing the scripts to be annotated. Script files are often also
annotated using the RasMol
'<a href="#echo"><tt><b>echo</b></tt></a>'
command.
<p>
The most common way to generate a RasMol script file is to use the
'<a href="#write"><tt><b>write script</b></tt></a>'
or
'<a href="#write"><tt><b>write rasmol</b></tt></a>'
commands to output the sequence of commands that are needed to
regenerate the current view, representation and colouring of the
currently displayed molecule.
<p>
The RasMol command
'<tt><b>source</b></tt>'
is synonymous with the
'<tt><b>script</b></tt>'
command.
<p>
Scripts may also be created with a text editor.
<p><hr>
<a name="select"><h3>Select</h3></a><p>
<pre>
Syntax:  select {&lt;expression&gt;}
</pre><p>
Define the currently selected region of the molecule. All subsequent RasMol
commands that manipulate a molecule or modify its colour or representation
only affect the currently selected region. The parameter of a
'<tt><b>select</b></tt>'
command is a RasMol expression that is evaluated for every atom of the
current molecule. The currently selected (active) region of the molecule
are those atoms that cause the expression to evaluate true. To select
the whole molecule use the RasMol command
'<tt><b>select all</b></tt>'.
The behaviour of the
'<tt><b>select</b></tt>'
command without any parameters is determined by the RasMol
'<a href="#sethetero"><tt><b>hetero</b></tt></a>'
and
'<a href="#set"><tt><b>hydrogen</b></tt></a>'
parameters.
<p>
Type &quot;help expression&quot; for more information on RasMol atom expressions.
<p><hr>
<a name="set"><h3>Set</h3></a><p>
<pre>
Syntax:  set &lt;parameter&gt; {&lt;option&gt;}
</pre><p>
The RasMol
'<tt><b>set</b></tt>'
command allows the user to alter various internal program parameters
such as those controlling rendering options. Each parameter has its
own set or permissible parameter options. Typically, omitting the
paramter option resets that parameter to its default value. A list of
valid parameter names is given below.
<p>
<center><table BORDER CELLPADDING=2>
<TR><TD><a href="#setambient">Ambient</a>
<TD><a href="#setaxes">Axes</a>
<TD><a href="#setbackground">Background</a>
<TD><a href="#setbackfade">BackFade</a>
<TD><a href="#setbondmode">BondMode</a>
<TD><a href="#setbonds">Bonds</a>
<TD><a href="#setboundbox">BoundBox</a>
<TD><a href="#setcartoon">Cartoon</a>
<TR><TD><a href="#setcisangle">CisAngle</A>
<TD><a href="#setdisplay">Display</a>
<TD><a href="#setfontsize">FontSize</a>
<TD><a href="#setfontstroke">FontStroke</a>
<TD><a href="#sethbonds">HBonds</a>
<TD><a href="#sethetero">Hetero</a>
<TD><a href="#sethourglass">HourGlass</a>
<TD><a href="#sethydrogen">Hydrogen</a>
<TR><TD><a href="#setkinemage">Kinemage</a>
<TD><a href="#setmenus">Menus</a>
<TD><a href="#setmonitor">Monitor</a>
<TD><a href="#setmouse">Mouse</a>
<TD><a href="#setpicking">Picking</a>
<TD><a href="#setradius">Radius</a>
<TD><a href="#setshadow">Shadow</a>
<TD><a href="#setslabmode">SlabMode</a>
<TR><TD><a href="#setsolvent">Solvent</a>
<TD><a href="#setspecular">Specular</a>
<TD><a href="#setspecpower">SpecPower</a>
<TD><a href="#setstereo">Stereo</a>
<TD><a href="#setssbonds">SSBonds</a>
<TD><a href="#setstrands">Strands</a>
<TD><a href="#settransparent">Transparent</a>
<TD><a href="#setunitcell">UnitCell</a>
<TR><TD><a href="#setvectps">VectPS</a>
<TD><a href="#setwrite">Write</a>
<TD CELLSPAN=6>
</table></center>
<p><hr>
<a name="show"><h3>Show</h3></a><p>
<pre>
Syntax:  show information
         show phipsi
         show RamPrint
         show selected { group | chain | atom }
         show sequence
         show symmetry
</pre><p>
The RasMol
'<tt><b>show</b></tt>'
command display details of the status of the currently
loaded molecule. The command
'<tt><b>show information</b></tt>'
lists the molecule's name,
classification, PDB code and the number of atoms, chains, groups it contains.
If hydrogen bonding, disulphide bridges or secondary structure have been
determined, the number of hbonds, ssbonds, helices, ladders and turns
are also displayed, respectively. The command
'<tt><b>show phipsi</b></tt>'
shows the phi and psi angles of the currently selected residues and
the omega angles of cis peptide bonds. The command
'<tt><b>show RamPrint</b></tt>'
(or 'show RPP' or 'show RamachandranPrinterPlot') shows a simple
Ramachandran printer plot in the style of Frances Bernstein's fisipl
program.  The command
'<tt><b>show selected</b></tt>'
(or 'show selected group' or 'show selected chain' or  'show selected atom' )
shows the groups (default), chains or atoms of the current selection.
The command
'<tt><b>show sequence</b></tt>'
lists the residues that comprise each chain of the molecule.  The command
'<tt><b>show symmetry</b></tt>'
shows the space group and unit cell of the molecule.
<p><hr>
<a name="slab"><h3>Slab</h3></a><p>
<pre>
Syntax:  slab {&lt;boolean&gt;}
         slab &lt;value&gt;
</pre><p>
The RasMol
'<tt><b>slab</b></tt>'
command enables, disables or positions the z-clipping plane of the
molecule. The program only draws those portions of the
molecule that are further from the viewer than the slabbing plane.
Integer values range  from zero at the very back of the molecule to
100 which is completely in front of the molecule. Intermediate values
determine the percentage of the molecule to be drawn.
<p><hr>
<a name="spacefill"><h3>Spacefill</h3></a><p>
<pre>
Syntax:  spacefill {&lt;boolean&gt;}
         spacefill temperature
         spacefill user
         spacefill &lt;value&gt;
</pre><p>
The RasMol
'<tt><b>spacefill</b></tt>'
command is used to represent all of the currently selected atoms as solid
spheres. This command is used to produce both union-of-spheres and
ball-and-stick models of a molecule. The command,
'<tt><b>spacefill true</b></tt>',
the default, represents each atom as a sphere of van der Waals radius.
The command
'<tt><b>spacefill off</b></tt>'
turns off the representation of the selected atom as spheres. A sphere
radius may be specified as an integer in RasMol units (1/250th &Aring;ngstrom)
or a value containing a decimal point. A value of 500 (2.0
&Aring;ngstroms) or greater results in a &quot;Parameter value too large&quot; error.
<p>
The
'<tt><b>temperature</b></tt>'
option sets the radius of each sphere to the value stored in its temperature
field. Zero or negative values have no effect and values greater than
2.0 are truncated to 2.0.  The
'<tt><b>user</b></tt>'
option allows the radius of each sphere to be specified by additional lines
in the molecule's PDB file using Raster 3D's COLOUR record extension.
<p>
The RasMol command
'<tt><b>cpk</b></tt>'
is synonymous with the
'<tt><b>spacefill</b></tt>'
command.
<p><hr>
<a name="spanish"><h3>Spanish</h3></a><p>
<pre>
Syntax:  Spanish
</pre><p>
The RasMol
'<tt><b>Spanish</b></tt>'
command sets the menus and messages to the Spanish versions.
The command
'<a href="#english"><tt><b>English</b></tt></a>'
may be used to select English menus and messages.
<p><hr>
<a name="ssbonds"><h3>SSBonds</h3></a><p>
<pre>
Syntax:  ssbonds {&lt;boolean&gt;}
         ssbonds &lt;value&gt;
</pre><p>
The RasMol
'<tt><b>ssbonds</b></tt>'
command is used to represent the disulphide bridges of the protein
molecule as either dotted lines or cylinders between the connected
cysteines. The first time that the
'<tt><b>ssbonds</b></tt>'
command is used, the program searches the structure of the protein to
find half-cysteine pairs (cysteines whose sulphurs are within 3 &Aring;ngstroms
of each other) and reports the number of bridges to the user. The command
'<tt><b>ssbonds on</b></tt>'
displays the selected &quot;bonds&quot; as dotted lines, and the command
'<tt><b>ssbonds off</b></tt>'
disables the display of ssbonds in the currently selected area. Selection
of disulphide bridges is identical to normal bonds, and may be adjusted
using the RasMol
'<a href="#setbondmode"><tt><b>set bondmode</b></tt></a>'
command. The colour of disulphide bonds may be changed using the
'<a href="#colour"><tt><b>colour ssbonds</b></tt></a>'
command. By default, each disulphide bond has the colours of its connected
atoms.
<p>
By default disulphide bonds are drawn between the sulphur atoms within
the cysteine groups. By using the
'<a href="#setssbonds"><tt><b>set ssbonds</b></tt></a>'
command the position of the cysteine's alpha carbons may be used instead.
<p><hr>
<a name="star"><h3>Star</h3></a><p>
<pre>
Syntax:  star {&lt;boolean&gt;}
         star temperature
         star user
         star &lt;value&gt;
</pre><p>
The RasMol
'<tt><b>star</b></tt>'
command is used to represent all of the currently selected atoms as
stars (six strokes, one each in the x, -x, y, -y, z and -z directions).
The commands
'<tt><b>select not bonded</b></tt>'
followed by
'<tt><b>star 75</b></tt>'
are useful to mark unbonded atoms in a
'<a href="#wireframe"><tt><b>wireframe</b></tt></a>'
display with less overhead than provided by
'<a href="#spacefill"><tt><b>spacefill 75</b></tt></a>'.
This can be done automatically for all subsequent wireframe
displays with the command
'<a href="#setbondmode"><tt><b>set bondmode not bonded</b></tt></a>'.
<p>
The command
'<tt><b>star true</b></tt>',
the default, represents each atom as a star with strokes
length equal to van der Waals radius.
The command
'<tt><b>star off</b></tt>'
turns off the representation of the selected atom as stars. A star
stroke length may be specified as an integer in RasMol units
(1/250th &Aring;ngstrom)
or a value containing a decimal point. A value of 500 (2.0
&Aring;ngstroms) or greater results in a &quot;Parameter value too large&quot; error.
<p>
The
'<tt><b>temperature</b></tt>'
option sets the stroke length of each star to the value stored
in its temperature
field. Zero or negative values have no effect and values greater than
2.0 are truncated to 2.0.  The
'<tt><b>user</b></tt>'
option allows the stroke length of each star to be specified by
additional lines
in the molecule's PDB file using Raster 3D's COLOUR record extension.
<p>
The RasMol
'<a href="#spacefill"><tt><b>spacefill</b></tt></a>'
command can be used for more artistic rendering of atoms as spheres.
<p><hr>
<a name="stereo"><h3>Stereo</h3></a><p>
<pre>
Syntax:  stereo on
         stereo &lt;number&gt;
         stereo off
</pre><p>
The RasMol
'<tt><b>stereo</b></tt>'
command provides side-by-side stereo display of images. Stereo
viewing of a molecule may be turned on (and off) either by
selecting
'<tt><b>Stereo</b></tt>'
from the
'<tt><b>Options</b></tt>'
menu, or by
typing the commands
'<tt><b>stereo on</b></tt>'
or
'<tt><b>stereo off</b></tt>'.
The separation angle between
the two views may be adjusted with the
'<a href="#setstereo"><tt><b>set stereo [-] &lt;number&gt;</b></tt></a>'
command, where positive values result in crossed eye
viewing and negative values in relaxed (wall-eyed) viewing.
The inclusion of
'<tt><b>[-] &lt;number&gt;</b></tt>'
in the
'<tt><b>stereo</b></tt>'
command, as for example in
'<tt><b>stereo 3</b></tt>'
or
'<tt><b>stereo -5</b></tt>',
also controls angle and direction.
<p>
The stereo command is only partially
implemented. When stereo is turned on, the image is not properly
recentred. (This can be done with a
'<a href="#translate"><tt><b>translate x -&lt;number&gt;</b></tt></a>'
 command.)
It is not supported in vector PostScript output files, is not
saved by the
'<tt><b>write script</b></tt>'
command, and in
general is not yet properly interfaced with several other
features of the program.
<p><hr>
<a name="strands"><h3>Strands</h3></a><p>
<pre>
Syntax:  strands {&lt;boolean&gt;}
         strands &lt;value&gt;
</pre><p>
The RasMol
'<tt><b>strands</b></tt>'
command displays the currently loaded protein or nucleic acid as a
smooth &quot;ribbon&quot; of depth-cued curves passing along the backbone of the
protein. The ribbon is composed of a number of strands that run parallel
to one another along the peptide plane of each residue. The ribbon is
drawn between each amino acid whose alpha carbon is currently selected.
The colour of the ribbon is changed by the RasMol
'<a href="#colour"><tt><b>colour ribbon</b></tt></a>'
command. If the current ribbon colour is
'<tt><b>none</b></tt>'
(the default), the colour is taken from the alpha carbon at each
position along its length. The central and outermost
strands may be coloured independently using the
'<a href="#colour"><tt><b>colour ribbon1</b></tt></a>'
and
'<a href="#colour"><tt><b>colour ribbon2</b></tt></a>'
commands, respectively. The number of strands in the ribbon may be
altered using the RasMol
'<a href="#setstrands"><tt><b>set strands</b></tt></a>'
command.
<p>
The width of the ribbon at each position is determined by the optional
parameter in the usual RasMol units. By default the width of the ribbon
is taken from the secondary structure of the protein or a constant value
of 720 for nucleic acids (which produces a ribbon 2.88 &Aring;ngstroms wide).
The default width of protein alpha helices and beta sheets is 380 (1.52
&Aring;ngstroms) and 100 (0.4 &Aring;ngstroms) for turns and random coil. The
secondary structure assignment is either from the PDB file or calculated
using the DSSP algorithm as used by the
'<a href="#structure"><tt><b>structure</b></tt></a>'
command. This command is similar to the RasMol command
'<a href="#ribbons"><tt><b>ribbons</b></tt></a>'
which renders the biomolecular ribbon as a smooth shaded surface.
<p><hr>
<a name="structure"><h3>Structure</h3></a><p>
<pre>
Syntax:  structure
</pre><p>
The RasMol
'<tt><b>structure</b></tt>'
command calculates secondary structure assignments
for the currently loaded protein. If the original PDB file contained
structural assignment records (HELIX, SHEET and TURN) these are discarded.
Initially, the hydrogen bonds of the current molecule are found, if this
hasn't been done already. The secondary structure is then determined using
Kabsch and Sander's DSSP algorithm. Once finished the program reports the
number of helices, strands and turns found.
<p><hr>
<a name="trace"><h3>Trace</h3></a><p>
<pre>
Syntax:  trace {&lt;boolean&gt;}
         trace <value>
         trace temperature
</pre><p>
The RasMol
'<tt><b>trace</b></tt>'
command displays a smooth spline between  consecutive alpha
carbon positions.  This spline does not pass exactly through
the alpha carbon position of each residue, but  follows the
same path as
'<a href="#ribbons"><tt><b>ribbons</b></tt></a>',
'<a href="#strands"><tt><b>strands</b></tt></a>'
and
'<a href="#cartoon"><tt><b>cartoons</b></tt></a>'.
Note that residues may be displayed as
'<a href="#ribbons"><tt><b>ribbons</b></tt></a>',
'<a href="#strands"><tt><b>strands</b></tt></a>',
'<a href="#cartoon"><tt><b>cartoons</b></tt></a>'
or as a
'<tt><b>trace</b></tt>'.
Enabling one of these
representations disables the others. However, a residue
may be displayed simultaneously as backbone and as one of
the above representations.  This may change in future
versions of RasMol.  Prior to version 2.6,
'<tt><b>trace</b></tt>'
was synonymous with
'<tt><b>backbone</b></tt>'.
<p>
'<tt><b>Trace temperature</b></tt>'
displays the backbone as a wider cylinder
at high temperature factors and thinner at lower.  This
representation is useful to X-ray crystallographers and NMR spectroscopists.
<p><hr>
<a name="translate"><h3>Translate</h3></a><p>
<pre>
Syntax:  translate &lt;axis&gt; {-} &lt;value&gt;
</pre><p>
The RasMol
'<tt><b>translate</b></tt>'
command moves the position of the centre of the molecule on the
screen. The axis parameter specifies along which axis the molecule
is to be moved and the integer parameter specifies the absolute
position of the molecule centre from the middle of the screen.
Permitted values for the axis parameter are
&quot;<tt><b>x</b></tt>&quot;, &quot;<tt><b>y</b></tt>&quot; and &quot;<tt><b>z</b></tt>&quot;.
Displacement values must be between -100 and 100 which correspond to
moving the current molecule just off the screen. A positive
&quot;<tt><b>x</b></tt>&quot;
displacement moves the molecule to the right, and a positive
&quot;<tt><b>y</b></tt>&quot;
displacement moves the molecule down the screen. The pair of commands
'<tt><b>translate x 0</b></tt>'
and
'<tt><b>translate y 0</b></tt>'
centres the molecule on the screen.
<p><hr>
<a name="wireframe"><h3>Wireframe</h3></a><p>
<pre>
Syntax:  wireframe {&lt;boolean&gt;}
         wireframe &lt;value&gt;
</pre><p>
The RasMol
'<tt><b>wireframe</b></tt>'
command represents each bond within the selected region of the molecule
as a cylinder, a line or a depth-cued vector. The display of bonds
as depth-cued vectors (drawn darker the further away from the viewer)
is turned on by the command
'<tt><b>wireframe</b></tt>'
or
'<tt><b>wireframe on</b></tt>'.
The selected bonds are displayed as cylinders by specifying a radius
either as an integer in RasMol units or containing a decimal point as
a value in &Aring;ngstroms.  A parameter value of 500 (2.0 &Aring;ngstroms) or
above results in an &quot;Parameter value too large&quot; error. Bonds may be
coloured using the
'<a href="#colour"><tt><b>colour bonds</b></tt></a>'
command.
<p>
Non-bonded atoms, which could become invisible in an
ordinary
'<tt><b>wireframe</b></tt>'
display can be marked by a preceding
'<a href="#setbondmode"><tt><b>set bondmode not bonded</b></tt></a>'
command.  If nearly co-linear bonds to atoms cause them to be
difficult to see in a wireframe display, the
'<a href="#setbondmode"><tt><b>set bondmode all</b></tt></a>'
command will add markers for
'<tt><b>all</b></tt>'
atoms in subsequent
'<tt><b>wireframe</b></tt>'
command executions.
<p><hr>
<a name="write"><h3>Write</h3></a><p>
<pre>
Syntax:  write {&lt;format&gt;} &lt;filename&gt;
</pre><p>
Write the current image to a file in a standard format. Currently
supported image file formats include
'<tt><b>bmp</b></tt>'
(Microsoft bitmap) and
'<tt><b>gif</b></tt>'
(Compuserve GIF),
'<tt><b>iris</b></tt>'
(IRIS RGB),
'<tt><b>ppm</b></tt>'
(Portable Pixmap),
'<tt><b>ras</b></tt>'
(Sun rasterfile),
'<tt><b>ps</b></tt>'
and
'<tt><b>epsf</b></tt>'
(Encapsulated PostScript),
'<tt><b>monops</b></tt>'
(Monochrome Encapsulated PostScript),
'<tt><b>pict</b></tt>'
(Apple PICT),
'<tt><b>vectps</b></tt>'
(Vector Postscript).  The
'<tt><b>write</b></tt>'
command may also be used to generate command scripts for other graphics
programs. The format
'<tt><b>script</b></tt>'
writes out a file containing the RasMol
'<a href="#script"><tt><b>script</b></tt></a>'
commands to reproduce the current image. The format
'<tt><b>molscript</b></tt>'
writes out the commands required to render the current view of the
molecule as ribbons in Per Kraulis' Molscript program and the format
'<tt><b>kinemage</b></tt>'
the commands for David Richardson's program Mage.  The following
formats are useful for further processing:
'<tt><b>povray</b></tt>'
(POVRay 2),
'<tt><b>povray3</b></tt>'
(POVRay 3 -- under development),
'<tt><b>vrml</b></tt>'
(VRML file).
Finally, several
formats are provided to provide phi-psi data for listing or for
'<tt><b>phipsi</b></tt>'
(phi-psi data as an annotated list with cis omegas),
'<tt><b>ramachan</b></tt>'
and
'<tt><b>RDF</b></tt>'
and
'<tt><b>RamachandranDataFile</b></tt>'
(phi-psi data as columns of numbers for gnuplot),
'<tt><b>RPP</b></tt>'
and
'<tt><b>RamachandranPrinterPlot</b></tt>'
(phi-psi data as a printer plot).
<p>
The distinction between this command and the RasMol
'<a href="#save"><tt><b>save</b></tt></a>'
command has been dropped. The only difference is that without a format
specifier the
'<a href="#save"><tt><b>save</b></tt></a>'
command generates a
'<tt><b>PDB</b></tt>'
file and the
'<tt><b>write</b></tt>'
command generates a
'<tt><b>GIF</b></tt>'
image.
<p><hr>
<a name="zap"><h3>Zap</h3></a><p>
<pre>
Syntax:  zap
</pre><p>
Deletes the contents of the current database and resets parameter
variables to their initial default state.
<p><hr>
<a name="zoom"><h3>Zoom</h3></a><p>
<pre>
Syntax:  zoom {&lt;boolean&gt;}
         zoom &lt;value&gt;
</pre><p>
Change the magnification of the currently displayed image. Boolean
parameters either magnify or reset the scale of current molecule. An
integer parameter specifies the desired magnification as a percentage
of the default scale. The minimum parameter value is 10; the maximum
parameter value is dependent upon the size of the molecule being
displayed. For medium sized proteins this is about 500.
<p><hr>
<a name="chsetopt"><h2>Internal Parameters</h2></a>
RasMol has a number of internal parameters that may be modified using the
'<a href="#set"><tt><b>set</b></tt></a>'
command. These parameters control a number of program options such as
rendering options and mouse button mappings.
<p>
A complete list of internal parameter names is given below.
<p>
<center><table BORDER CELLPADDING=2>
<TR><TD><a href="#setambient">Ambient</a>
<TD><a href="#setaxes">Axes</a>
<TD><a href="#setbackground">Background</a>
<TD><a href="#setbackfade">BackFade</a>
<TD><a href="#setbondmode">BondMode</a>
<TD><a href="#setbonds">Bonds</a>
<TD><a href="#setboundbox">BoundBox</a>
<TD><a href="#setcartoon">Cartoon</a>
<TR><TD><a href="#setcisangle">CisAngle</A>
<TD><a href="#setdisplay">Display</a>
<TD><a href="#setfontsize">FontSize</a>
<TD><a href="#setfontstroke">FontStroke</a>
<TD><a href="#sethbonds">HBonds</a>
<TD><a href="#sethetero">Hetero</a>
<TD><a href="#sethourglass">HourGlass</a>
<TD><a href="#sethydrogen">Hydrogen</a>
<TR><TD><a href="#setkinemage">Kinemage</a>
<TD><a href="#setmenus">Menus</a>
<TD><a href="#setmonitor">Monitor</a>
<TD><a href="#setmouse">Mouse</a>
<TD><a href="#setpicking">Picking</a>
<TD><a href="#setradius">Radius</a>
<TD><a href="#setshadow">Shadow</a>
<TD><a href="#setslabmode">SlabMode</a>
<TR><TD><a href="#setsolvent">Solvent</a>
<TD><a href="#setspecular">Specular</a>
<TD><a href="#setspecpower">SpecPower</a>
<TD><a href="#setstereo">Stereo</a>
<TD><a href="#setssbonds">SSBonds</a>
<TD><a href="#setstrands">Strands</a>
<TD><a href="#settransparent">Transparent</a>
<TD><a href="#setunitcell">UnitCell</a>
<TR><TD><a href="#setvectps">VectPS</a>
<TD><a href="#setwrite">Write</a>
<TD CELLSPAN=6>
</table></center>
<p><hr>
<a name="setambient"><h3>Set Ambient</h3></a><p>
<pre>
Syntax:  set ambient {&lt;value&gt;}
</pre><p>
The RasMol
'<tt><b>ambient</b></tt>'
parameter is used to control the amount of ambient (or surrounding)
light in the scene. The
'<tt><b>ambient</b></tt>'
value must be between 0 and 100. It controls the percentage intensity
of the darkest shade of an object. For a solid object, this is the
intensity of surfaces facing away from the light source or in shadow.
For depth-cued objects this is the intensity of objects furthest from
the viewer.
<p>
This parameter is commonly used to correct for monitors with different
&quot;gamma values&quot; (brightness), to change how light or dark a hardcopy
image appears when printed or to alter the feeling of depth for
wireframe or ribbon representations.
<p><hr>
<a name="setaxes"><h3>Set Axes</h3></a><p>
<pre>
Syntax:  set axes &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>axes</b></tt>'
parameter controls the display of orthogonal coordinate axes on
the current display. The coordinate axes are those used in the
molecule data file, and the origin is the centre of the molecule's
bounding box. The
'<tt><b>set axes</b></tt>'
command is similar to the commands
'<a href="#setboundbox"><tt><b>set boundbox</b></tt></a>'
and
'<a href="#setunitcell"><tt><b>set unitcell</b></tt></a>'
that display the bounding box and the crystallographic unit cell,
respectively.
<p><hr>
<a name="setbackfade"><h3>Set Backfade</h3></a><p>
<pre>
Syntax:  set backfade &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>backfade</b></tt>'
parameter is used to control backfade to the specified background
colour, rather than black.  This is controlled by the commands
'<tt><b>set backfade on</b></tt>'
and
'<tt><b>set backfade off</b></tt>'.
For example, this may be used to generate depth-cued images that
fade to white, rather than black.
<p><hr>
<a name="setbackground"><h3>Set Background</h3></a><p>
<pre>
Syntax:  set background {&lt;colour&gt;}
</pre><p>
The RasMol
'<tt><b>background</b></tt>'
parameter is used to set the colour of the &quot;canvas&quot; background. The
colour may be given as either a colour name or a comma separated
triple of Red, Green, Blue (RGB) components enclosed in square
brackets. Typing the command
'<a href="#help"><tt><b>help colours</b></tt></a>'
will give a list of the predefined colour names recognised by RasMol.
When running under X Windows, RasMol also recognises colours in the
X server's colour name database.
<p>
The command
'<tt><b>set background</b></tt>'
is synonymous with the RasMol command
'<a href="#background"><tt><b>background</b></tt></a>'.
<p><hr>
<a name="setbondmode"><h3>Set BondMode</h3></a><p>
<pre>
Syntax:  set bondmode and
         set bondmode or
         set bondmode all
         set bondmode none
         set bondmode not bonded
</pre><p>
The RasMol
'<tt><b>set bondmode</b></tt>'
command controls the mechanism used to select individual bonds
and modifies the display of bonded and non-bonded atoms by subsequent
'<a href="#wireframe"><tt><b>wireframe</b></tt></a>'
commands.
<p>
When using the
'<a href="#select"><tt><b>select</b></tt></a>'
and
'<a href="#restrict"><tt><b>restrict</b></tt></a>'
commands, a given bond will be selected if i) the bondmode is
'<tt><b>or</b></tt>'
and either of the connected atoms is selected, or ii) the bondmode is
'<tt><b>and</b></tt>'
and both atoms connected by the bond are selected. Hence an individual
bond may be uniquely identified by using the command
'<tt><b>set bondmode and</b></tt>'
and then uniquely selecting the atoms at both ends.
<p>
The
'<tt><b>bondmode [all | none | not bonded]</b></tt>'
commands add
'<a href="#star"><tt><b>star 75</b></tt></a>'
or
'<a href="#spacefill"><tt><b>spacefill 75</b></tt></a>'
markers for the designated atoms to
'<a href="#wireframe"><tt><b>wireframe</b></tt></a>'
displays.  Stars are used when the specified wireframe radius is zero.
<p><hr>
<a name="setbonds"><h3>Set Bonds</h3></a><p>
<pre>
Syntax:  set bonds &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>bonds</b></tt>'
parameter is used to control display of double and triple bonds as
multiple  lines or cylinders.  Currently bond orders are only read
from  MDL Mol files, Sybyl Mol2 format files, Tripos Alchemy format
files, CIF and mmCIF,  and suitable PDB files.  Double (and triple)  bonds
are specified in some  PDB files by specifying a given bond twice  (and
three times) in CONECT records.  The  command
'<tt><b>set bonds on</b></tt>'
enables the display of bond orders, and  the command
'<tt><b>set bonds off</b></tt>'
disables them.
<p><hr>
<a name="setboundbox"><h3>Set BoundBox</h3></a><p>
<pre>
Syntax:  set boundbox &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>boundbox</b></tt>'
parameter controls the display of the current molecule's bounding box
on the display. The bounding box is orthogonal to the data file's
original coordinate axes. The
'<tt><b>set boundbox</b></tt>'
command is similar to the commands
'<a href="#setaxes"><tt><b>set axes</b></tt></a>'
and
'<a href="#setboundbox"><tt><b>set unitcell</b></tt></a>'
that display orthogonal coordinate axes and the bounding box,
respectively.
<p><hr>
<a name="setcartoon"><h3>Set Cartoon</h3></a><p>
<pre>
Syntax:  set cartoon {&lt;boolean&gt;}
         set cartoon {&lt;number&gt;}
</pre><p>
The RasMol
'<tt><b>cartoon</b></tt>'
parameter is used to control display of the cartoon version of the
'<a href="#ribbons"><tt><b>ribbons</b></tt></a>'
display.  By default, the C-termini of beta-sheets are displayed as
arrow heads. This may be enabled and disabled using the
'<tt><b>set cartoons &lt;boolean&gt;</b></tt>'
command. The depth of the cartoon may be adjusted using the
'<tt><b>cartoons &lt;number&gt;</b></tt>'
command. The
'<tt><b>set cartoons</b></tt>'
command without any parameters returns these two options to
 their default values.
<p><hr>
<a name="setcisangle"><h3>Set CisAngle</h3></a><p>
<pre>
Syntax:  set cisangle {&lt;value&gt;}
</pre><p>
The RasMol
'<tt><b>cisangle</b></tt>'
parameter controls the cutoff angle for identifying cis peptide
 bonds.  If no value is given, the cutoff is set to 90 degrees.
<p><hr>
<a name="setdisplay"><h3>Set Display</h3></a><p>
<pre>
Syntax:  set display selected
         set display normal
</pre><p>
This command controls the display mode within RasMol. By default,
'<tt><b>set display normal</b></tt>',
RasMol displays the molecule in the representation specified by the
user. The command
'<tt><b>set display selected</b></tt>'
changes the display mode such that the molecule is temporarily drawn
so as to indicate currently selected portion of the molecule. The
user specified colour scheme and representation remains unchanged.
In this representation all selected atoms are shown in yellow and
all non selected atoms are shown in blue. The colour of the background
is also changed to a dark grey to indicate the change of display mode.
This command is typically only used by external Graphical User
Interfaces (GUIs).
<p><hr>
<a name="setfontsize"><h3>Set FontSize</h3></a><p>
<pre>
Syntax:  set fontsize {&lt;value&gt;} { FS | PS }
</pre><p>
The RasMol
'<tt><b>set fontsize</b></tt>'
command is used to control the size of the characters that
form atom labels. This value corresponds to the height of
the displayed character in pixels. The maximum value of
'<tt><b>fontsize</b></tt>'
is 48 pixels, and the default value is 8 pixels high.
Fixed or proportional spacing may be selected by appending the
&quot;FS&quot; or &quot;PS&quot; modifiers, respectively.  The default is &quot;FS&quot;.
To display atom labels on the screen use the RasMol
'<a href="#label"><tt><b>label</b></tt></a>'
command and to change the colour of displayed labels, use
the
'<a href="#colour"><tt><b>colour labels</b></tt></a>'
command.
<p><hr>
<a name="setfontstroke"><h3>Set FontStroke</h3></a><p>
<pre>
Syntax:  set fontstroke {&lt;value&gt;}
</pre><p>
The RasMol
'<tt><b>set fontstroke</b></tt>'
command is used to control the size of the stroke width of the
characters that form atom labels.  This value is the radius in
pixels of cylinders used to form the strokes.  The special value
of &quot;0&quot; is the default used for the normal single pixel stroke width,
which allows for rapid drawing and rotation of the image.
Non-zero values are provided to allow for more artistic atom
labels for publication at the expense of extra time in rendering
the image.
<p>
When wider strokes are used, a larger font size is recommend, e.g.
by using the RasMol
'<a href="#setfontsize"><tt><b>set fontsize 24 PS</b></tt></a>'
command, followed by
'<tt><b>set fontstroke 2</b></tt>'
<p>
The character sets used by RasMol rendered with fixed spacing with
single-pixel-width strokes and with proportional spacing with
2-pixel-radius cylinder strokes are shown in the following sample:
<p>
<CENTER><IMG SRC="../html_graphics/fontsample.jpg" ALT="[font sample]">
</CENTER>
<p>
To display atom labels on the screen use the RasMol
'<a href="#label"><tt><b>label</b></tt></a>'
command, and to change the colour of displayed labels use
the
'<a href="#colour"><tt><b>colour labels</b></tt></a>'
command.
<p><hr>
<a name="sethbonds"><h3>Set HBonds</h3></a><p>
<pre>
Syntax:  set hbonds backbone
         set hbonds sidechain
</pre><p>
The RasMol
'<tt><b>hbonds</b></tt>'
parameter determines whether hydrogen bonds are drawn between
the donor and acceptor atoms of the hydrogen bond,
'<tt><b>set hbonds sidechain</b></tt>'
or between the alpha carbon atoms of the protein backbone and between
the phosphorous atoms of the nucleic acid backbone,
'<tt><b>set hbonds backbone</b></tt>'.
The actual display of hydrogen bonds is controlled by the
'<a href="#hbonds"><tt><b>hbonds</b></tt></a>'
command. Drawing hydrogen bonds between protein alpha carbons or
nucleic acid phosphorous atoms is useful when the rest of the molecule
is shown in only a schematic representation such as
'<a href="#backbone"><tt><b>backbone</b></tt></a>',
'<a href="#ribbons"><tt><b>ribbons</b></tt></a>'
or
'<a href="#strands"><tt><b>strands</b></tt></a>'.
This parameter is similar to the RasMol
'<a href="#setssbonds"><tt><b>ssbonds</b></tt></a>'
parameter.
<p><hr>
<a name="sethetero"><h3>Set Hetero</h3></a><p>
<pre>
Syntax:  set hetero &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>hetero</b></tt>'
parameter is used to modify the 'default' behaviour of the RasMol
'<a href="#select"><tt><b>select</b></tt></a>'
command, <i>i.e.</i> the behaviour of
'<tt><b>select</b></tt>'
without any parameters. When this value is
'<tt><b>false</b></tt>',
the default
'<a href="#select"><tt><b>select</b></tt></a>'
region does not include any heterogeneous atoms (refer to the
predefined set
'<a href="#heteroset"><tt><b>hetero</b></tt></a>'
). When this value is
'<tt><b>true</b></tt>',
the default
'<a href="#select"><tt><b>select</b></tt></a>'
region may contain hetero atoms. This parameter is similar to
the RasMol
'<a href="#sethydrogen"><tt><b>hydrogen</b></tt></a>'
parameter which determines whether hydrogen atoms should be
included in the default set. If both
'<tt><b>hetero</b></tt>'
and
'<a href="#sethydrogen"><tt><b>hydrogen</b></tt></a>'
are
'<tt><b>true</b></tt>',
'<a href="#select"><tt><b>select</b></tt></a>'
without any parameters is equivalent to
'<a href="#select"><tt><b>select all</b></tt></a>'.
<p><hr>
<a name="sethourglass"><h3>Set HourGlass</h3></a><p>
<pre>
Syntax:  set hourglass &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>hourglass</b></tt>'
parameter allows the user to enable and disable the use of the 'hour
glass' cursor used by RasMol to indicate that the program is currently
busy drawing the next frame. The command
'<tt><b>set hourglass on</b></tt>'
enables the indicator, whilst
'<tt><b>set hourglass off</b></tt>'
prevents RasMol from changing the cursor. This is useful when spinning
the molecule, running a sequence of commands from a script file or
using interprocess communication to execute complex sequences of
commands. In these cases a 'flashing' cursor may be distracting.
<p><hr>
<a name="sethydrogen"><h3>Set Hydrogen</h3></a><p>
<pre>
Syntax:  set hydrogen &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>hydrogen</b></tt>'
parameter is used to modify the &quot;default&quot; behaviour of the RasMol
'<a href="#select"><tt><b>select</b></tt></a>'
command, <i>i.e.</i> the behaviour of
'<tt><b>select</b></tt>'
without any parameters. When this value is
'<tt><b>false</b></tt>',
the default
'<a href="#select"><tt><b>select</b></tt></a>'
region does not include any hydrogen, deuterium or tritium atoms (refer
to the predefined set
'<a href="#hydrogenset"><tt><b>hydrogen</b></tt></a>'
). When this value is
'<tt><b>true</b></tt>',
the default
'<a href="#select"><tt><b>select</b></tt></a>'
region may contain hydrogen atoms. This parameter is similar to
the RasMol
'<a href="#sethetero"><tt><b>hetero</b></tt></a>'
parameter which determines whether heterogeneous atoms should be
included in the default set. If both
'<tt><b>hydrogen</b></tt>'
and
'<a href="#sethetero"><tt><b>hetero</b></tt></a>'
are
'<tt><b>true</b></tt>',
'<a href="#select"><tt><b>select</b></tt></a>'
without any parameters is equivalent to
'<a href="#select"><tt><b>select all</b></tt></a>'.
<p><hr>
<a name="setkinemage"><h3>Set Kinemage</h3></a><p>
<pre>
Syntax:  set kinemage &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>set kinemage</b></tt>'
command controls the amount of detail stored in a Kinemage output
file generated by the RasMol
'<a href="#write"><tt><b>write kinemage</b></tt></a>'
command. The output kinemage files are intended to be displayed by
David Richardson's Mage program.
'<tt><b>set kinemage false</b></tt>',
the default, only stores the currently displayed representation in
the generated output file. The command
'<tt><b>set kinemage true</b></tt>',
generates a more complex Kinemage that contains both the wireframe
and backbone representations as well as the coordinate axes,
bounding box and crystal unit cell.
<p><hr>
<a name="setmenus"><h3>Set Menus</h3></a><p>
<pre>
Syntax:  set menus &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>set menus</b></tt>'
command enables the canvas window's menu buttons or menu bar. This
command is typically only used by graphical user interfaces or to
create as large an image as possible when using Microsoft Windows.
<p><hr>
<a name="setmonitor"><h3>Set Monitor</h3></a><p>
<pre>
Syntax:  set monitor &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>set monitor</b></tt>'
command enables
'<a href="#monitor"><tt><b>monitors</b></tt></a>'.
The distance monitor labels may be turned off with the command
'<tt><b>set monitor off</b></tt>',
and re-enabled with the command
'<tt><b>set monitor on</b></tt>'.

<p><hr>
<a name="setmouse"><h3>Set Mouse</h3></a><p>
<pre>
Syntax:  set mouse rasmol
         set mouse insight
         set mouse quanta
</pre><p>
The RasMol
'<tt><b>set mouse</b></tt>'
command sets the rotation, translation, scaling and zooming mouse
bindings. The default value is
'<tt><b>rasmol</b></tt>'
which is suitable for two button mice (for three button mice the
second and third buttons are synonymous); X-Y rotation is controlled
by the first button, and X-Y translation by the second. Additional
functions are controlled by holding a modifier key on the keyboard.
[Shift] and the first button performs scaling, [shift] and the second
button performs Z-rotation, and [control] and the first mouse button
controls the clipping plane. The
'<tt><b>insight</b></tt>'
and
'<tt><b>quanta</b></tt>'
options provide the same mouse bindings as other packages for experienced
users.
<p><hr>
<a name="setpicking"><h3>Set Picking</h3></a><p>
<pre>
Syntax:  set picking &lt;boolean&gt;
         set picking off
         set picking none
         set picking ident
         set picking distance
         set picking monitor
         set picking angle
         set picking torsion
         set picking label
         set picking centre
         set picking center
         set picking coord
</pre><p>
The RasMol
'<tt><b>set picking</b></tt>'
series of commands affects how a user may interact with a
molecule displayed on the screen in RasMol.
<p>
<tt><b>Enabling/Disabling Atom Picking</b></tt>:
Clicking on an atom with the mouse results in identification and
the display of its residue name, residue number, atom name, atom serial
number and chain in the command window. This behavior may be disabled
with the command
'<tt><b>set picking none</b></tt>'
and restored with the command
'<tt><b>set picking ident</b></tt>'.
The command
'<tt><b>set picking coord</b></tt>'
adds the atomic coordinates of the atom to the display.
<p>
Disabling picking, by using
'<tt><b>set picking off</b></tt>'
is useful when executing the
'<a href="#pause"><tt><b>pause</b></tt></a>'
command in RasMol scripts as it prevents the display of
spurious message on the command line while the script is suspended.
<p>
<tt><b>Measuring Distances, Angles and Torsions</b></tt>:
Interactive measurement of distances, angles and torsions
is achieved using the commands:
'<tt><b>set picking distance</b></tt>',
'<tt><b>set picking monitor</b></tt>',
'<tt><b>set picking angle</b></tt>'
and
'<tt><b>set picking torsion</b></tt>',
respectively. In these modes, clicking on an atom results in it
being identified on the rasmol command line. In addition every
atom picked increments a modulo counter such that in distance
mode, every second atom displays the distance (or distance monitor)
between this atom and the previous one. In angle mode, every
third atom displays the angle between the previous three atoms
and in torsion mode every fourth atom displays the torsion between
the last four atoms. By holding down the shift key while picking an
atom, this modulo counter is not incremented and allows, for
example, the distances of consecutive atoms from a fixed atom to
be displayed. See the
'<a href="#monitor"><tt><b>monitor</b></tt></a>'
command for how to control the display of distance monitor lines and labels.
<p>
<tt><b>Labelling Atoms with the Mouse</b></tt>:
The mouse may also be used to toggle the display of an atom label
on a given atom. The RasMol command
'<tt><b>set picking label</b></tt>'
removes a label from a picked atom if it already has one or
displays a concise label at that atom position otherwise.
<p>
<tt><b>Centring Rotation with the Mouse</b></tt>:
A molecule may be centred on a specified atom position using the
RasMol commands
'<tt><b>set picking centre</b></tt>'
or
'<tt><b>set picking center</b></tt>'.
In this mode, picking an atom causes all futher rotations to be
about that point.
<p><hr>
<a name="setradius"><h3>Set Radius</h3></a><p>
<pre>
Syntax:  set radius {&lt;value&gt;}
</pre><p>
The RasMol
'<tt><b>set radius</b></tt>'
command is used to alter the behaviour of the RasMol
'<a href="#dots"><tt><b>dots</b></tt></a>'
command depending upon the value of the
'<a href="#setsolvent"><tt><b>solvent</b></tt></a>'
parameter.
When
'<a href="#setsolvent"><tt><b>solvent</b></tt></a>'
is
'<tt><b>true</b></tt>',
the
'<tt><b>radius</b></tt>'
parameter controls whether a true van der Waals' surface
is generated by the
'<a href="#dots"><tt><b>dots</b></tt></a>'
command. If the value of
'<tt><b>radius</b></tt>'
is anything other than zero, that value is used as the
radius of each atom instead of its true vdW value. When
the value of
'<a href="#setsolvent"><tt><b>solvent</b></tt></a>'
is
'<tt><b>true</b></tt>',
this parameter determines the 'probe sphere' (solvent) radius.
The parameter may be given as an integer in rasmol units or
containing a decimal point in &Aring;ngstroms. The default value of
this parameter is determined by the value of
'<a href="#setsolvent"><tt><b>solvent</b></tt></a>'
and changing
'<a href="#setsolvent"><tt><b>solvent</b></tt></a>'
resets
'<tt><b>radius</b></tt>'
to its new default value.
<p><hr>
<a name="setshadow"><h3>Set Shadow</h3></a><p>
<pre>
Syntax:  set shadow &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>set shadow</b></tt>'
command enables and disables ray-tracing of the currently rendered image.
Currently only the spacefilling representation is shadowed or can cast
shadows. Enabling shadowing will automatically disable the Z-clipping
(slabbing) plane using the command
'<a href="#slab"><tt><b>slab off</b></tt></a>'.
Ray-tracing typically takes about several seconds for a moderately sized protein.
It is recommended that shadowing be normally disabled whilst the
molecule is being transformed or manipulated, and only enabled once
an appropiate viewpoint is selected, to provide a greater impression
of depth.
<p><hr>
<a name="setslabmode"><h3>Set SlabMode</h3></a><p>
<pre>
Syntax:  set slabmode &lt;slabmode&gt;
</pre><p>
The RasMol
'<tt><b>slabmode</b></tt>'
parameter controls the rendering method of objects cut by the
slabbing (z-clipping) plane. Valid slabmode parameters are
&quot;<tt><b>reject</b></tt>&quot;, &quot;<tt><b>half</b></tt>&quot;,
&quot;<tt><b>hollow</b></tt>&quot;, &quot;<tt><b>solid</b></tt>&quot; and
&quot;<tt><b>section</b></tt>&quot;.
<p><hr>
<a name="setsolvent"><h3>Set Solvent</h3></a><p>
<pre>
Syntax:  set solvent &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>set solvent</b></tt>'
command is used to control the behaviour of the RasMol
'<a href="#dots"><tt><b>dots</b></tt></a>'
command. Depending upon the value of the
'<tt><b>solvent</b></tt>'
parameter, the
'<a href="#dots"><tt><b>dots</b></tt></a>'
command either generates a van der Waals' or a solvent
accessible surface around the currently selected set of
atoms. Changing this parameter automatically resets the
value of the RasMol
'<a href="#setradius"><tt><b>radius</b></tt></a>'
parameter.
The command
'<tt><b>set solvent false</b></tt>',
the default value, indicates that a van der Waals' surface
should be generated and resets the value of
'<a href="#setradius"><tt><b>radius</b></tt></a>'
to zero. The command
'<tt><b>set solvent true</b></tt>'
indicates that a 'Connolly' or 'Richards' solvent
accessible surface should be drawn and sets the
'<a href="#setradius"><tt><b>radius</b></tt></a>'
parameter, the solvent radius, to 1.2 &Aring;ngstroms (or 300
RasMol units).
<p><hr>
<a name="setspecular"><h3>Set Specular</h3></a><p>
<pre>
Syntax:  set specular &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>set specular</b></tt>'
command enables and disables the display of specular highlights on
solid objects drawn by RasMol. Specular highlights appear as white
reflections of the light source on the surface of the object. The
current RasMol implementation uses an approximation function to
generate this highlight.
<p>
The specular highlights on the surfaces of solid objects may be
altered by using the specular reflection coefficient, which is
altered using the RasMol
'<a href="#setspecpower"><tt><b>set specpower</b></tt></a>'
command.
<p><hr>
<a name="setspecpower"><h3>Set SpecPower</h3></a><p>
<pre>
Syntax:  set specpower {&lt;value&gt;}
</pre><p>
The
'<tt><b>specpower</b></tt>'
parameter determines the shininess of solid objects rendered by
RasMol. This value between 0 and 100 adjusts the reflection
coefficient used in specular highlight calculations. The specular
highlights are enabled and disabled by the RasMol
'<a href="#setspecular"><tt><b>set specular</b></tt></a>'
command. Values around 20 or 30 produce plastic looking surfaces.
High values represent more shiny surfaces such as metals, while
lower values produce more diffuse/dull surfaces.
<p><hr>
<a name="setssbonds"><h3>Set SSBonds</h3></a><p>
<pre>
Syntax:  set ssbonds backbone
         set ssbonds sidechain
</pre><p>
The RasMol
'<tt><b>ssbonds</b></tt>'
parameter determines whether disulphide bridges are drawn between
the sulphur atoms in the sidechain (the default) or between the alpha
carbon atoms in the backbone of the cysteines residues. The actual
display of disulphide bridges is controlled by the
'<a href="#ssbonds"><tt><b>ssbonds</b></tt></a>'
command. Drawing disulphide bridges between alpha carbons is useful
when the rest of the protein is shown in only a schematic
representation such as
'<a href="#backbone"><tt><b>backbone</b></tt></a>',
'<a href="#ribbons"><tt><b>ribbons</b></tt></a>'
or
'<a href="#strands"><tt><b>strands</b></tt></a>'.
This parameter is similar to the RasMol
'<a href="#sethbonds"><tt><b>hbonds</b></tt></a>'
parameter.
<p><hr>
<a name="setstereo"><h3>Set Stereo</h3></a><p>
<PRE>
Syntax:  set stereo &lt;boolean&gt;
         set stereo [-] &lt;number&gt
</PRE><P>
The RasMol
'<tt><b>set stereo</b></tt>'
parameter controls the separation between the left and
right images. Turning stereo on and off doesn't reposition
the centre of the molecule.
<p>
Stereo viewing of a molecule may be turned on (and off) either
by selecting
'<tt><b>Stereo</b></tt>'
from the
'<tt><b>Options</b></tt>'
menu, or by typing the commands
'<tt><b>stereo on</b></tt>'
or
'<tt><b>stereo off</b></tt>'.
<p>
The separation angle between the two views may be adjusted with the
'<tt><b>set stereo [-] &lt;number&gt;</b></tt>'
command, where positive values result in crossed eye viewing and
negative values in relaxed (wall-eyed) viewing. Currently, stereo viewing
is not supported in
'<a href="#setvectps"><tt><b>vector PostScript</b></tt></a>'
output files.
<p><hr>
<a name="setstrands"><h3>Set Strands</h3></a><p>
<pre>
Syntax:  set strands {&lt;value&gt;}
</pre><p>
The RasMol
'<tt><b>strands</b></tt>'
parameter controls the number of parallel strands that are displayed
in the ribbon representations of proteins. The permissible values for
this parameter are 1, 2, 3, 4, 5 and 9. The default value is 5. The
number of strands is constant for all ribbons being displayed.
However, the ribbon width (the separation between strands) may be
controlled on a residue by residue basis using the RasMol
'<a href="#ribbons"><tt><b>ribbons</b></tt></a>'
command.
<p><hr>
<a name="settransparent"><h3>Set Transparent</h3></a><p>
<pre>
Syntax:  set transparent &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>transparent</b></tt>'
parameter controls the writing of transparent GIFs by the
'<a href="#write"><tt><b>write gif &lt;filename&gt;</b></tt></a>'
command.  This may be controlled by the
'<tt><b>set transparent on</b></tt>'
and
'<tt><b>set transparent off</b></tt>'
commands.
<p><hr>
<a name="setunitcell"><h3>Set UnitCell</h3></a><p>
<pre>
Syntax:  set unitcell &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>unitcell</b></tt>'
parameter controls the display of the crystallographic unit cell on
the current display. The crystal cell is only enabled if the appropriate
crystal symmetry information is contained in the PDB, CIF or mmCIF data
file. The
RasMol command
'<a href="#show"><tt><b>show symmetry</b></tt></a>'
display details of the crystal's space group and unit cell axes. The
'<tt><b>set unitcell</b></tt>'
command is similar to the commands
'<a href="#setaxes"><tt><b>set axes</b></tt></a>'
and
'<a href="#setboundbox"><tt><b>set boundbox</b></tt></a>'
that display orthogonal coordinate axes and the bounding box,
respectively.
<p><hr>
<a name="setvectps"><h3>Set VectPS</h3></a><p>
<pre>
Syntax:  set vectps &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>vectps</b></tt>'
parameter is use to control the way in which the RasMol
'<a href="#write"><tt><b>write</b></tt></a>'
command generates vector PostScript output files. The command
'<tt><b>set vectps on</b></tt>'
enables the use of black outlines around spheres and cylinder bonds
producing &quot;cartoon-like&quot; high resolution output. However, the current
implementation of RasMol incorrectly cartoons spheres that are intersected
by more than one other sphere. Hence &quot;ball and stick&quot; models are rendered
correctly but not large spacefilling spheres models. Cartoon outlines
can be disabled, the default, by the command
'<tt><b>set vectps off</b></tt>'.
<p><hr>
<a name="setwrite"><h3>Set Write</h3></a><p>
<pre>
Syntax:  set write &lt;boolean&gt;
</pre><p>
The RasMol
'<tt><b>write</b></tt>'
parameter controls the use of the
'<a href="#save"><tt><b>save</b></tt></a>'
and
'<a href="#write"><tt><b>write</b></tt></a>'
commands within scripts, but it may only be executed from the
command line.  By default, this value is
'<tt><b>false</b></tt>',
prohibiting the generation of files in any scripts executed at
start-up (such as those launched from a WWW browser). However,
animators may start up RasMol interactively: type
'<tt><b>set write on</b></tt>'
and then execute a script to generate each frame using the
source command.
<p><hr>
<a name="chexprs"><h2>Atom Expressions</h2></a>
RasMol atom expressions uniquely identify an arbitrary group of atoms
within a molecule. Atom expressions are composed of either primitive
expressions,
predefined sets,
comparison operators,
'<tt><b>within</b></tt>'
expressions,
or logical (boolean) combinations of the above expression types.
<p>
The logical operators allow complex queries to be constructed out of
simpler ones using the standard boolean connectives
'<tt><b>and</b></tt>',
'<tt><b>or</b></tt>'
and
'<tt><b>not</b></tt>'.
These may be abbreviated by the symbols
&quot;<tt>&amp;</tt>&quot;, &quot;<tt>|</tt>&quot; and &quot;<tt>!</tt>&quot;,
respectively. Parentheses (brackets) may be used to alter the
precedence of the operators. For convenience, a comma may also
be used for boolean disjunction.
<p>
The atom expression is evaluated for each atom, hence
'<tt><b>protein and backbone</b></tt>'
selects protein backbone atoms, not the protein and [nucleic] acid
backbone atoms!
<p>
<pre>
Examples:    <b>backbone and not helix</b>
             <b>within( 8.0, ser70 )</b>
             <b>not (hydrogen or hetero)</b>
             <b>not *.FE and hetero</b>
             <b>8, 12, 16, 20-28</b>
             <b>arg, his, lys</b>
</pre>
<p>
<ul>
<li><a href="#primitiveexpressions">Primitive Expressions</a>
<li><a href="#predefinedsets">Predefined Sets</a>
<li><a href="#comparisonoperators">Comparison Operators</a>
<li><a href="#withinexpressions">Within Expressions</a>
<li><a href="#exampleexpressions">Examples</a>
</ul>
<p><hr>
<a name="exampleexpressions"><h3>Example Expressions</h3></a><p>
The following table gives some useful examples of RasMol
atom expressions.
<p>
<pre>
    <b>Expression</b>      <b>Interpretation</b>

    *               All atoms
    cys             Atoms in cysteines
    hoh             Atoms in heterogeneous water molecules
    as?             Atoms in either asparagine or aspartic acid
    *120            Atoms at residue 120 of all chains
    *p              Atoms in chain P
    *.n?            Nitrogen atoms
    cys.sg          Sulphur atoms in cysteine residues
    ser70.c?        Carbon atoms in serine-70
    hem*p.fe        Iron atoms in the Heme groups of chain P
    *.*;A           All atoms in alternate conformation A
    */4             All atoms in model 4
</pre>
<p><hr>
<a name="primitiveexpressions"><h3>Primitive Expressions</h3></a><p>
RasMol primitive expressions are the fundamental building blocks
of atom expressions. There are two types of primitive expression.
The first type is used to identify a given residue number or range
of residue numbers. A single residue is identified by its number
(position in the sequence), and a range is specified by lower and
upper bounds separated by a hyphen character. For example
'<tt><b>select 5,6,7,8</b></tt>'
is also
'<tt><b>select 5-8</b></tt>'.
Note that this selects the given residue numbers in all macromolecule
chains.
<p>
The second type of primitive expression specifies a sequence of fields
that must match for a given atom. The first part specifies a residue
(or group of residues) and an optional second part specifies the atoms
within those residues. The first part consists of a residue name,
optionally followed by a residue number and/or chain identifier.
<p>
A residue name typically consists of up to three alphabetic characters,
which are case insensitive. Hence the primitive expressions
'<tt><b>SER</b></tt>'
and
'<tt><b>ser</b></tt>'
are equivalent, identifying all serine residues.
Residue names that contain non-alphabetic characters, such as
sulphate groups, may be delimited using square brackets, <i>i.e.</i>
'<tt><b>[SO4]</b></tt>'.
<p>
The residue number is intended to be the residue's position in
the macromolecule sequence, but negative sequence numbers, gaps
in numbering, or even reverse numbering are permitted in the PDB format.
Care must be taken when specifying both residue name and number.
If the group at the specified position isn't the specified residue
then no atoms are selected.
<p>
The chain identifier is typically a single case-insensitive
alphabetic or numeric character. Numeric chain identifiers must
be distinguished or separated from residue numbers by a colon
character. For example,
&quot;<tt><b>SER70A</b></tt>&quot;
for the alphabetic chain identifier, &quot;A&quot;, or
&quot;<tt><b>SER70:1</b></tt>&quot;
for the numeric chain identifier, &quot;1&quot;.
<p>
The second part consists of a period character followed by an atom
name.  An atom name may be up to four alphabetic or numeric characters.
An optional semicolon followed by an alternate conformation
identifier may be appended.  An optional slash followed by a
model number may also be appended.
<p>
An atom name may be up to four alphabetic or numeric characters.
<p>
An asterisk may be used as a wild card for a whole field and a
question mark as a single character wildcard.
<p><hr>
<a name="comparisonoperators"><h3>Comparison Operators</h3></a><p>
Parts of a molecule may also be distinguished using equality,
inequality and ordering operators on their properties. The format
of such comparison expression is a property name, followed by a
comparison operator and then an integer value.
<p>
The atom properties that may be used in RasMol are
'<tt><b>atomno</b></tt>'
for the atom serial number,
'<tt><b>elemno</b></tt>'
for the atom's atomic number (element),
'<tt><b>resno</b></tt>'
for the residue number,
'<tt><b>radius</b></tt>'
for the spacefill radius in RasMol units (or zero if not represented
as a sphere) and
'<tt><b>temperature</b></tt>'
for the PDB isotropic temperature value.
<p>
The equality operator is denoted either
&quot;<tt><b>=</b></tt>&quot; or &quot;<tt><b>==</b></tt>&quot;.
The inequality operator as either
&quot;<tt><b>&lt;&gt;</b></tt>&quot;, &quot;<tt><b>!=</b></tt>&quot; or
&quot;<tt><b>/=</b></tt>&quot;.
The ordering operators are
&quot;<tt><b>&lt;</b></tt>&quot;
for less than,
&quot;<tt><b>&lt;=</b></tt>&quot;
for less than or equal to,
&quot;<tt><b>&gt;</b></tt>&quot;
for greater than, and
&quot;<tt><b>&gt;=</b></tt>&quot;
for greater than or equal to.
<p>
<pre>
Examples:    resno &lt; 23
             temperature &gt;= 900
             atomno == 487
</pre>
<p><hr>
<a name="withinexpressions"><h3>Within Expressions</h3></a><p>
A RasMol
'<tt><b>within</b></tt>'
expression allows atoms to be selected on their proximity to
another set of atoms. A
'<tt><b>within</b></tt>'
expression takes two parameters separated by a comma and surrounded
by parentheses. The first argument is an integer value called the
&quot;cut-off&quot; distance of the within expression and the second argument
is any  valid atom expression. The cut-off distance is expressed in
either integer RasMol units or &Aring;ngstroms containing a decimal point.
An atom is selected if it is within the cut-off distance of any of
the atoms defined by the second argument. This allows complex
expressions to be constructed containing nested
'<tt><b>within</b></tt>'
expressions.
<p>
For example, the command
'<tt><b>select within(3.2,backbone)</b></tt>'
selects any atom within a 3.2 &Aring;ngstrom radius of any atom in a
protein or nucleic acid backbone.
'<tt><b>Within</b></tt>'
expressions are particularly useful for selecting the atoms
around an active site.
<p><hr>
<a name="predefinedsets"><h3>Predefined Sets</h3></a><p>
RasMol atom expressions may contain predefined sets. These sets
are single keywords that represent portions of a molecule of interest.
Predefined sets are often abbreviations of primitive atom expressions.
In some cases the use of predefined sets allows selection of areas of
a molecule that could not otherwise be distinguished.
A list of the currently predefined sets
is given below.
In addition to the sets listed here, RasMol also treats element names
(and their plurals) as predefined sets containing all atoms of that
element type, <i>i.e.</i> the command
'<a href="#select"><tt><b>select oxygen</b></tt></a>'
is equivalent to the command
'<a href="#select"><tt><b>select elemno=8</b></tt></a>'.
<p>
<pre>
    <a href="#atset">AT</a>              <a href="#acidicset">Acidic</a>          <a href="#acyclicset">Acyclic</a>
    <a href="#aliphaticset">Aliphatic</a>       <a href="#alphaset">Alpha</a>           <a href="#aminoset">Amino</a>
    <a href="#aromaticset">Aromatic</a>        <a href="#backboneset">Backbone</a>        <a href="#basicset">Basic</a>
    <a href="#bondedset">Bonded</a>          <a href="#buriedset">Buried</a>          <a href="#cgset">CG</a>
    <a href="#chargedset">Charged</a>         <a href="#cyclicset">Cyclic</a>          <a href="#cystineset">Cystine</a>
    <a href="#helixset">Helix</a>           <a href="#heteroset">Hetero</a>          <a href="#hydrogenset">Hydrogen</a>
    <a href="#hydrophobicset">Hydrophobic</a>     <a href="#ionsset">Ions</a>            <a href="#largeset">Large</a>
    <a href="#ligandset">Ligand</a>          <a href="#mediumset">Medium</a>          <a href="#neutralset">Neutral</a>
    <a href="#nucleicset">Nucleic</a>         <a href="#polarset">Polar</a>           <a href="#proteinset">Protein</a>
    <a href="#purineset">Purine</a>          <a href="#pyrimidineset">Pyrimidine</a>      <a href="#selectedset">Selected</a>
    <a href="#sheetset">Sheet</a>           <a href="#sidechainset">Sidechain</a>       <a href="#smallset">Small</a>
    <a href="#solventset">Solvent</a>         <a href="#surfaceset">Surface</a>         <a href="#turnset">Turn</a>
    <a href="#waterset">Water</a>
</pre>
<p><hr>
<a name="atset"><h3>AT Set</h3></a><p>
This set contains the atoms in the complementary nucleotides
adenosine and thymidine (A and T, respectively). All nucleotides
are classified as either the set
'<tt><b>at</b></tt>'
or the set
'<a href="#cgset"><tt><b>cg</b></tt></a>'
This set is equivalent to the RasMol atom expressions
&quot;<tt><b>a,t</b></tt>&quot;,
and
&quot;<tt><b>nucleic and not cg</b></tt>&quot;.
<p><hr>
<a name="acidicset"><h3>Acidic Set</h3></a><p>
The set of acidic amino acids.
These are the residue types Asp and Glu.
All amino acids are classified as either
'<tt><b>acidic</b></tt>',
'<a href="#basicset"><tt><b>basic</b></tt></a>'
'<tt><b>or</b></tt>'
'<a href="#neutralset"><tt><b>neutral</b></tt></a>'.
This set is equivalent to the RasMol atom expressions
&quot;<tt><b>asp, glu</b></tt>&quot;
and
&quot;<tt><b>amino and not (basic or neutral)</b></tt>&quot;.
<p><hr>
<a name="acyclicset"><h3>Acyclic Set</h3></a><p>
The set of atoms in amino acids not containing a cycle or
ring. All amino acids are classified as either
'<a href="#cyclicset"><tt><b>cyclic</b></tt></a>'
or
'<tt><b>acyclic</b></tt>'.
This set is equivalent to the RasMol atom expression
&quot;<tt><b>amino and not cyclic</b></tt>&quot;.
<p><hr>
<a name="aliphaticset"><h3>Aliphatic Set</h3></a><p>
This set contains the aliphatic amino acids.
These are the amino acids Ala, Gly, Ile, Leu and Val.
This set is equivalent to the RasMol atom expression
&quot;<tt><b>ala, gly, ile, leu, val</b></tt>&quot;.
<p><hr>
<a name="alphaset"><h3>Alpha Set</h3></a><p>
The set of alpha carbons in the protein molecule. This set is
approximately equivalent to the RasMol atom expression
&quot;<tt><b>*.CA</b></tt>&quot;.
This command should not be confused with the predefined set
'<a href="#helixset"><tt><b>helix</b></tt></a>'
which contains the atoms in the amino acids of the protein's
alpha helices.
<p><hr>
<a name="aminoset"><h3>Amino Set</h3></a><p>
This set contains all the atoms contained in amino acid residues.
This is useful for distinguishing the protein from the nucleic
acid and heterogeneous atoms in the current molecule database.
<p><hr>
<a name="aromaticset"><h3>Aromatic Set</h3></a><p>
The set of atoms in amino acids containing aromatic rings.
These are the amino acids His, Phe, Trp and Tyr.
Because they contain aromatic rings all members of this
set are member of the predefined set
'<a href="#cyclicset"><tt><b>cyclic</b></tt></a>'.
This set is equivalent to the RasMol atom expressions
&quot;<tt><b>his, phe, trp, tyr</b></tt>&quot;
and
&quot;<tt><b>cyclic and not pro</b></tt>&quot;.
<p><hr>
<a name="backboneset"><h3>Backbone Set</h3></a><p>
This set contains the four atoms of each amino acid that form the
polypeptide N-C-C-O backbone of proteins, and the atoms of the sugar
phosphate backbone of nucleic acids.
Use the RasMol predefined sets
'<tt><b>protein</b></tt>'
and
'<tt><b>nucleic</b></tt>'
to distinguish between the two forms of backbone.
Atoms in nucleic acids and proteins are either
'<tt><b>backbone</b></tt>'
or
'<a href="#sidechainset"><tt><b>sidechain</b></tt></a>'.
This set is equivalent to the RasMol expression
&quot;<tt><b>(protein or nucleic) and not sidechain</b></tt>&quot;.
<p>
The predefined set
'<tt><b>mainchain</b></tt>'
is synonymous with the set
'<tt><b>backbone</b></tt>'.
<p><hr>
<a name="basicset"><h3>Basic Set</h3></a><p>
The set of basic amino acids.
These are the residue types Arg, His and Lys.
All amino acids are classified as either
'<a href="#acidicset"><tt><b>acidic</b></tt></a>',
'<tt><b>basic</b></tt>'
or
'<a href="#neutralset"><tt><b>neutral</b></tt></a>'.
This set is equivalent to the RasMol atom expressions
&quot;<tt><b>arg, his, lys</b></tt>&quot;
and
&quot;<tt><b>amino and not (acidic or neutral)</b></tt>&quot;.
<p><hr>
<a name="bondedset"><h3>Bonded Set</h3></a><p>
This set contain all the atoms in the current molecule database that
are bonded to at least one other atom.
<p><hr>
<a name="buriedset"><h3>Buried Set</h3></a><p>
This set contains the atoms in those amino acids that tend
(prefer) to be buried inside protein, away from contact with
solvent molecules. This set refers to the amino acids
preference and not the actual solvent accessibility for
the current protein.
All amino acids are classified as either
'<a href="#surfaceset"><tt><b>surface</b></tt></a>'
or
'<tt><b>buried</b></tt>'.
This set is equivalent to the RasMol atom expression
&quot;<tt><b>amino and not surface</b></tt>&quot;.
<p><hr>
<a name="cgset"><h3>CG Set</h3></a><p>
This set contains the atoms in the complementary nucleotides
cytidine and guanosine (C and G, respectively). All nucleotides
are classified as either the set
'<a href="#atset"><tt><b>at</b></tt></a>'
or the set
'<tt><b>cg</b></tt>'
This set is equivalent to the RasMol atom expressions
&quot;<tt><b>c,g</b></tt>&quot;
and
&quot;<tt><b>nucleic and not at</b></tt>&quot;.
<p><hr>
<a name="chargedset"><h3>Charged Set</h3></a><p>
This set contains the charged amino acids. These are the amino
acids that are either
'<a href="#acidicset"><tt><b>acidic</b></tt></a>'
or
'<a href="#basicset"><tt><b>basic</b></tt></a>'.
Amino acids are classified as being either
'<tt><b>charged</b></tt>'
or
'<a href="#neutralset"><tt><b>neutral</b></tt></a>'.
This set is equivalent to the RasMol atom expressions
&quot;<tt><b>acidic or basic</b></tt>&quot;
and
&quot;<tt><b>amino and not neutral</b></tt>&quot;.
<p><hr>
<a name="cyclicset"><h3>Cyclic Set</h3></a><p>
The set of atoms in amino acids containing a cycle or rings.
All amino acids are classified as either
'<tt><b>cyclic</b></tt>'
or
'<a href="#acyclicset"><tt><b>acyclic</b></tt></a>'.
This set consists of the amino acids His, Phe, Pro, Trp and Tyr.
The members of the predefined set
'<a href="#aromaticset"><tt><b>aromatic</b></tt></a>'
are members of this set.
The only cyclic but non-aromatic amino acid is proline.
This set is equivalent to the RasMol atom expressions
&quot;<tt><b>his, phe, pro, trp, tyr</b></tt>&quot;
and
&quot;<tt><b>aromatic or pro</b></tt>&quot;
and
&quot;<tt><b>amino and not acyclic</b></tt>&quot;.
<p><hr>
<a name="cystineset"><h3>Cystine Set</h3></a><p>
This set contains the atoms of cysteine residues that form part
of a disulphide bridge, <i>i.e.</i> half cystines. RasMol automatically
determines disulphide bridges, if neither the predefined set
'<tt><b>cystine</b></tt>'
nor the RasMol
'<a href="#ssbonds"><tt><b>ssbonds</b></tt></a>'
command have been used since the molecule was loaded. The set of
free cysteines may be determined using the RasMol atom expression
&quot;<tt><b>cys and not cystine</b></tt>&quot;.
<p><hr>
<a name="helixset"><h3>Helix Set</h3></a><p>
This set contains all atoms that form part of a protein alpha
helix as determined by either the PDB file author or Kabsch and
Sander's DSSP algorithm. By default, RasMol uses the secondary
structure determination given in the PDB file if it exists.
Otherwise, it uses the DSSP algorithm as used by the RasMol
'<a href="#structure"><tt><b>structure</b></tt></a>'
command.
<p>
This predefined set should not be confused with the predefined set
'<a href="#alphaset"><tt><b>alpha</b></tt></a>'
which contains the alpha carbon atoms of a protein.
<p><hr>
<a name="heteroset"><h3>Hetero Set</h3></a><p>
This set contains all the heterogeneous atoms in the molecule. These
are the atoms described by HETATM entries in the PDB file. These
typically contain water, cofactors and other solvents and ligands. All
'<tt><b>hetero</b></tt>'
atoms are classified as either
'<a href="#ligandset"><tt><b>ligand</b></tt></a>'
or
'<a href="#solventset"><tt><b>solvent</b></tt></a>'
atoms. These heterogeneous
'<a href="#solventset"><tt><b>solvent</b></tt></a>'
atoms are further classified as either
'<a href="#waterset"><tt><b>water</b></tt></a>'
or
'<a href="#ionsset"><tt><b>ions</b></tt></a>'.
<p><hr>
<a name="hydrogenset"><h3>Hydrogen Set</h3></a><p>
This predefined set contains all the hydrogen, deuterium and tritium atoms
of the current molecule. This predefined set is equivalent to the
RasMol atom expression
&quot;<tt><b>elemno=1</b></tt>&quot;.
<p><hr>
<a name="hydrophobicset"><h3>Hydrophobic Set</h3></a><p>
This set contains all the hydrophobic amino acids.
These are the amino acids Ala, Leu, Val, Ile, Pro, Phe, Met and Trp.
All amino acids are classified as either
'<tt><b>hydrophobic</b></tt>'
or
'<a href="#polarset"><tt><b>polar</b></tt></a>'.
This set is equivalent to the RasMol atom expressions
&quot;<tt><b>ala, leu, val, ile, pro, phe, met, trp</b></tt>&quot;
and
&quot;<tt><b>amino and not polar</b></tt>&quot;.
<p><hr>
<a name="ionsset"><h3>Ions Set</h3></a><p>
This set contains all the heterogeneous phosphate and sulphate ions in
the current molecule data file. A large number of these ions are
sometimes associated with protein and nucleic acid structures determined
by X-ray crystallography. These atoms tend to clutter an image. All
'<a href="#heteroset"><tt><b>hetero</b></tt></a>'
atoms are classified as either
'<a href="#ligandset"><tt><b>ligand</b></tt></a>'
or
'<a href="#solventset"><tt><b>solvent</b></tt></a>'
atoms. All
'<a href="#solventset"><tt><b>solvent</b></tt></a>'
atoms are classified as either
'<a href="#waterset"><tt><b>water</b></tt></a>'
or
'<tt><b>ions</b></tt>'.
<p><hr>
<a name="largeset"><h3>Large Set</h3></a><p>
All amino acids are classified as either
'<a href="#smallset"><tt><b>small</b></tt></a>',
'<a href="#mediumset"><tt><b>medium</b></tt></a>'
or
'<tt><b>large</b></tt>'.
This set is equivalent to the RasMol atom expression
&quot;<tt><b>amino and not (small or medium)</b></tt>&quot;.
<p><hr>
<a name="ligandset"><h3>Ligand Set</h3></a><p>
This set contains all the heterogeneous cofactor and ligand moieties that
are contained in the current molecule data file.  This set is defined
to be all
'<a href="#heteroset"><tt><b>hetero</b></tt></a>'
atoms that are not
'<a href="#solventset"><tt><b>solvent</b></tt></a>'
atoms. Hence this set is equivalent to the RasMol atom expression
&quot;<tt><b>hetero and not solvent</b></tt>&quot;.
<p><hr>
<a name="mediumset"><h3>Medium Set</h3></a><p>
All amino acids are classified as either
'<a href="#smallset"><tt><b>small</b></tt></a>',
'<tt><b>medium</b></tt>'
or
'<a href="#largeset"><tt><b>large</b></tt></a>'.
This set is equivalent to the RasMol atom expression
&quot;<tt><b>amino and not (large or small)</b></tt>&quot;.
<p><hr>
<a name="neutralset"><h3>Neutral Set</h3></a><p>
The set of neutral amino acids.
All amino acids are classified as either
'<a href="#acidicset"><tt><b>acidic</b></tt></a>',
'<a href="#basicset"><tt><b>basic</b></tt></a>'
or
'<tt><b>neutral</b></tt>'.
This set is equivalent to the RasMol atom expression
&quot;<tt><b>amino and not (acidic or basic)</b></tt>&quot;.
<p><hr>
<a name="nucleicset"><h3>Nucleic Set</h3></a><p>
The set of all atoms in nucleic acids, which consists of the four
nucleotide bases adenosine, cytidine, guanosine and thymidine (A,
C, G and T, respectively). All neucleotides are classified as either
'<a href="#purineset"><tt><b>purine</b></tt></a>'
or
'<a href="#pyrimidineset"><tt><b>pyrimidine</b></tt></a>'.
This set is equivalent to the RasMol atom expressions
&quot;<tt><b>a,c,g,t</b></tt>&quot;
and
&quot;<tt><b>purine or pyrimidine</b></tt>&quot;.
The symbols for RNA nucleotides (U, +U, I, 1MA, 5MC, OMC,
1MG, 2MG, M2G, 7MG, OMG, YG, H2U, 5MU, and PSU) are also
recognized as members of this set.
<p><hr>
<a name="polarset"><h3>Polar Set</h3></a><p>
This set contains the polar amino acids.
All amino acids are classified as either
'<a href="#hydrophobicset"><tt><b>hydrophobic</b></tt></a>'
or
'<tt><b>polar</b></tt>'.
This set is equivalent to the RasMol atom expression
&quot;<tt><b>amino and not hydrophobic</b></tt>&quot;.
<p><hr>
<a name="proteinset"><h3>Protein Set</h3></a><p>
The set of all atoms in proteins. This consists of the RasMol
predefined set
'<a href="#aminoset"><tt><b>amino</b></tt></a>'
and common post-translation modifications.
<p><hr>
<a name="purineset"><h3>Purine Set</h3></a><p>
The set of purine nucleotides.
These are the bases adenosine and guanosine (A and G, respectively).
All nucleotides are either
'<tt><b>purines</b></tt>'
or
'<a href="#pyrimidineset"><tt><b>pyrimidines</b></tt></a>'.
This set is equivalent to the RasMol atom expressions
&quot;<tt><b>a,g</b></tt>&quot;
and
&quot;<tt><b>nucleic and not pyrimidine</b></tt>&quot;.
<p><hr>
<a name="pyrimidineset"><h3>Pyrimidine Set</h3></a><p>
The set of pyrimidine nucleotides.
These are the bases cytidine and thymidine (C and T, respectively).
All nucleotides are either
'<a href="#purineset"><tt><b>purines</b></tt></a>'
or
'<tt><b>pyrimidines</b></tt>'.
This set is equivalent to the RasMol atom expressions
&quot;<tt><b>c,t</b></tt>&quot;
and
&quot;<tt><b>nucleic and not purine</b></tt>&quot;.
<p><hr>
<a name="selectedset"><h3>Selected Set</h3></a><p>
This set contains the set of atoms in the currently selected
region. The currently selected region is defined by the preceding
'<a href="#select"><tt><b>select</b></tt></a>'
or
'<a href="#restrict"><tt><b>restrict</b></tt></a>'
command and not the atom expression containing the
'<tt><b>selected</b></tt>'
keyword.
<p><hr>
<a name="sheetset"><h3>Sheet Set</h3></a><p>
This set contains all atoms that form part of a protein beta
sheet as determined by either the PDB file author or Kabsch and
Sander's DSSP algorithm. By default, RasMol uses the secondary
structure determination given in the PDB file if it exists.
Otherwise, it uses the DSSP algorithm as used by the RasMol
'<a href="#structure"><tt><b>structure</b></tt></a>'
command.
<p><hr>
<a name="sidechainset"><h3>Sidechain Set</h3></a><p>
This set contains the functional sidechains of any amino acids
and the base of each nucleotide. These are the atoms not part of
the polypeptide N-C-C-O backbone of proteins or the sugar
phosphate backbone of nucleic acids.
Use the RasMol predefined sets
'<tt><b>protein</b></tt>'
and
'<tt><b>nucleic</b></tt>'
to distinguish between the two forms of sidechain.
Atoms in nucleic acids and proteins are either
'<a href="#backboneset"><tt><b>backbone</b></tt></a>'
or
'<tt><b>sidechain</b></tt>'.
This set is equivalent to the RasMol expression
&quot;<tt><b>(protein or nucleic) and not backbone</b></tt>&quot;.
<p><hr>
<a name="smallset"><h3>Small Set</h3></a><p>
All amino acids are classified as either
'<tt><b>small</b></tt>',
'<a href="#mediumset"><tt><b>medium</b></tt></a>'
or
'<a href="#largeset"><tt><b>large</b></tt></a>'.
This set is equivalent to the RasMol atom expression
&quot;<tt><b>amino and not (medium or large)</b></tt>&quot;.
<p><hr>
<a name="solventset"><h3>Solvent Set</h3></a><p>
This set contains the solvent atoms in the molecule coordinate file.
These are the heterogeneous water molecules, phosphate and sulphate
ions. All
'<a href="#heteroset"><tt><b>hetero</b></tt></a>'
atoms are classified as either
'<a href="#ligandset"><tt><b>ligand</b></tt></a>'
or
'<tt><b>solvent</b></tt>'
atoms. All
'<tt><b>solvent</b></tt>'
atoms are classified as either
'<a href="#waterset"><tt><b>water</b></tt></a>'
or
'<a href="#ionsset"><tt><b>ions</b></tt></a>'.
This set is equivalent to the RasMol atom expressions
&quot;<tt><b>hetero and not ligand</b></tt>&quot;
and
&quot;<tt><b>water or ions</b></tt>&quot;.
<p><hr>
<a name="surfaceset"><h3>Surface Set</h3></a><p>
This set contains the atoms in those amino acids that tend
(prefer) to be on the surface of proteins, in contact with
solvent molecules. This set refers to the amino acids
preference and not the actual solvent accessibility for
the current protein.
All amino acids are classified as either
'<tt><b>surface</b></tt>'
or
'<a href="#buriedset"><tt><b>buried</b></tt></a>'.
This set is equivalent to the RasMol atom expression
&quot;<tt><b>amino and not buried</b></tt>&quot;.
<p><hr>
<a name="turnset"><h3>Turn Set</h3></a><p>
This set contains all atoms that form part of a protein turns
as determined by either the PDB file author or Kabsch and
Sander's DSSP algorithm. By default, RasMol uses the secondary
structure determination given in the PDB file if it exists.
Otherwise, it uses the DSSP algorithm as used by the RasMol
'<a href="#structure"><tt><b>structure</b></tt></a>'
command.
<p><hr>
<a name="waterset"><h3>Water Set</h3></a><p>
This set contains all the heterogeneous water molecules in the current
database. A large number of water molecules are sometimes associated
with protein and nucleic acid structures determined by X-ray
crystallography. These atoms tend to clutter an image.
All
'<a href="#heteroset"><tt><b>hetero</b></tt></a>'
atoms are classified as either
'<a href="#ligandset"><tt><b>ligand</b></tt></a>'
or
'<a href="#solventset"><tt><b>solvent</b></tt></a>'
atoms. The
'<a href="#solventset"><tt><b>solvent</b></tt></a>'
atoms are further classified as either
'<tt><b>water</b></tt>'
or
'<a href="#ionsset"><tt><b>ions</b></tt></a>'.
<p><hr>
<a name="setsummary"><h3>Set Summary</h3></a><p>
The table below summarises RasMol's classification of the common amino acids.
<p>
<CENTER>
<TABLE BORDER CELLPADDING=2>
<TR><TH>Residues:<TH>ala<TH>arg<TH>asn<TH>asp<TH>cys<TH>glu<TH>gln<TH>gly<TH>his<TH>ile<TH>leu<TH>lys<TH>met<TH>phe<TH>pro<TH>ser<TH>thr<TH>trp<TH>tyr<TH>val
<TR><TH><TH>A<TH>R<TH>N<TH>D<TH>C<TH>E<TH>Q<TH>G<TH>H<TH>I<TH>L<TH>K<TH>M<TH>F<TH>P<TH>S<TH>T<TH>W<TH>Y<TH>V
<TR><TH>Predefined Set<TH CELLSPAN=20 ALIGN=CENTER>
<TR><TH ALIGN=LEFT>acidic     <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>
<TR><TH ALIGN=LEFT>acyclic    <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*
<TR><TH ALIGN=LEFT>aliphatic  <TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*
<TR><TH ALIGN=LEFT>aromatic   <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER>
<TR><TH ALIGN=LEFT>basic      <TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>
<TR><TH ALIGN=LEFT>buried     <TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>
<TD ALIGN=CENTER>*
<TR><TH ALIGN=LEFT>charged    <TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>
<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>
<TR><TH ALIGN=LEFT>cyclic     <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER>
<TR><TH ALIGN=LEFT>hydrophobic<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER>*
<TR><TH ALIGN=LEFT>large      <TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER>
<TR><TH ALIGN=LEFT>medium     <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>*
<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>*
<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*
<TR><TH ALIGN=LEFT>negative   <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>
<TR><TH ALIGN=LEFT>neutral    <TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>*
<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>*
<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER>*
<TR><TH ALIGN=LEFT>polar      <TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>
<TR><TH ALIGN=LEFT>positive   <TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>
<TR><TH ALIGN=LEFT>small      <TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>
<TR><TH ALIGN=LEFT>surface    <TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>*<TD ALIGN=CENTER>*
<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>
<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER> <TD ALIGN=CENTER>*
<TD ALIGN=CENTER>*<TD ALIGN=CENTER>*<TD ALIGN=CENTER> <TD ALIGN=CENTER>*
<TD ALIGN=CENTER>
</TABLE></CENTER>
<p><hr>
<a name="chcolours"><h2>Colour Schemes</h2></a>
The RasMol
'<a href="#colour"><tt><b>colour</b></tt></a>'
command allows different objects (such as atoms, bonds and ribbon segments)
to be given a specified colour. Typically this colour is either a RasMol
predefined colour name or an RGB triple. Additionally RasMol also supports
'<a href="#altcolours"><tt><b>alt</b></tt></a>',
'<a href="#aminocolours"><tt><b>amino</b></tt></a>',
'<a href="#chaincolours"><tt><b>chain</b></tt></a>',
'<a href="#chargecolours"><tt><b>charge</b></tt></a>',
'<a href="#cpkcolours"><tt><b>cpk</b></tt></a>',
'<a href="#groupcolours"><tt><b>group</b></tt></a>',
'<a href="#nmrmodelcolours"><tt><b>model</b></tt></a>',
'<a href="#shapelycolours"><tt><b>shapely</b></tt></a>',
'<a href="#structurecolours"><tt><b>structure</b></tt></a>',
'<a href="#temperaturecolours"><tt><b>temperature</b></tt></a>'
or
'<a href="#usercolours"><tt><b>user</b></tt></a>'
colour schemes for atoms, and
'<a href="#hbondtypecolours"><tt><b>hbond type</b></tt></a>'
colour scheme for hydrogen bonds and
'<a href="#potentialcolours"><tt><b>electrostatic potential</b></tt></a>'
colour scheme for dot surfaces.
The 24 currently predefined colour
names are
listed below with their corresponding RGB triplet and hexadecimal value.
<p>
<CENTER>
<TABLE BORDER CELLPADDING=2>
<TR><TH height="20">Predefined colour</TH>
<TH HEIGHT="20">Sample</TH>
<TH HEIGHT="20">RGB Values</TH>
<TH HEIGHT="20">Hexadecimal</TH></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Black</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#000000">&nbsp;
<TD HEIGHT="18"><TT>[  0,  0,  0]</TT></TD>
<TD HEIGHT="18"><TT>000000</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD height="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Blue</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#0000FF">&nbsp;
<TD HEIGHT="18"><TT>[  0,  0,255]</TT></TD>
<TD HEIGHT="18"><TT>0000FF</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD height="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>BlueTint</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#AFD7FF">&nbsp;
<TD HEIGHT="18"><TT>[175,214,255]</TT></TD>
<TD HEIGHT="18"><TT>AFD7FF</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD height="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Brown</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#AF7559">&nbsp;
<TD HEIGHT="18"><TT>[175,117,89]</TT></TD>
<TD HEIGHT="18"><TT>AF7559</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Cyan</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#00FFFF">&nbsp;
<TD HEIGHT="18"><TT>[  0,255,255]</TT></TD>
<TD HEIGHT="18"><TT>00FFFF</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Gold</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#FC9C00">&nbsp;
<TD HEIGHT="18"><TT>[255,156,  0]</TT></TD>
<TD HEIGHT="18"><TT>FC9C00</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Grey</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#7D7D7D">&nbsp;
<TD HEIGHT="18"><TT>[125,125,125]</TT></TD>
<TD HEIGHT="18"><TT>7D7D7D</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Green</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#00FF00">&nbsp;
<TD HEIGHT="18"><TT>[  0,255,  0]</TT></TD>
<TD HEIGHT="18"><TT>00FF00</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>GreenBlue</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#2E8B57">&nbsp;
<TD HEIGHT="18"><TT>[ 46,139,87]</TT></TD>
<TD HEIGHT="18"><TT>2E8B57</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>GreenTint</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#98FFB3">&nbsp;
<TD HEIGHT="18"><TT>[152,255,179]</TT></TD>
<TD HEIGHT="18"><TT>98FFB3</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>HotPink</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#FF0065">&nbsp;
<TD HEIGHT="18"><TT>[255,  0,101]</TT></TD>
<TD HEIGHT="18"><TT>FF0065</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Magenta</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#FF00FF">&nbsp;
<TD HEIGHT="18"><TT>[0,255,0]</TT></TD>
<TD HEIGHT="18"><TT>FF00FF</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Orange</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#FFA500">&nbsp;
<TD HEIGHT="18"><TT>[255,165,  0]</TT></TD>
<TD HEIGHT="18"><TT>FFA500</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Pink</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#FF6575">&nbsp;
<TD HEIGHT="18"><TT>[255,101,117]</TT></TD>
<TD HEIGHT="18"><TT>FF6575</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>PinkTint</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#FFABBB">&nbsp;
<TD HEIGHT="18"><TT>[255,171,187]</TT></TD>
<TD HEIGHT="18"><TT>FFABBB</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Purple</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#A020F0">&nbsp;
<TD HEIGHT="18"><TT>[160, 32,240]</TT></TD>
<TD HEIGHT="18"><TT>A020F0</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Red</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#FF0000">&nbsp;
<TD HEIGHT="18"><TT>[255,  0,  0]</TT></TD>
<TD HEIGHT="18"><TT>FF0000</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>RedOrange</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#FF4500">&nbsp;
<TD HEIGHT="18"><TT>[255, 69,  0]</TT></TD>
<TD HEIGHT="18"><TT>FF4500</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>SeaGreen</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#00FA6D">&nbsp;
<TD HEIGHT="18"><TT>[  0,250,109]</TT></TD>
<TD HEIGHT="18"><TT>00FA6D</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>SkyBlue</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#3A90FF">&nbsp;
<TD HEIGHT="18"><TT>[ 58,144,255]</TT></TD>
<TD HEIGHT="18"><TT>3A90FF</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Violet</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#EE82EE">&nbsp;
<TD HEIGHT="18"><TT>[238,130,238]</TT></TD>
<TD HEIGHT="18"><TT>EE82EE</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>White</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#FFFFFF">&nbsp;
<TD HEIGHT="18"><TT>[255,255,255]</TT></TD>
<TD HEIGHT="18"><TT>FFFFFF</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Yellow</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#FFFF00">&nbsp;
<TD HEIGHT="18"><TT>[255,255,  0]</TT></TD>
<TD HEIGHT="18"><TT>FFFF00</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="18" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>YellowTint</B></TT></FONT></TD>
<TD HEIGHT="18" BGCOLOR="#F6F675">&nbsp;
<TD HEIGHT="18"><TT>[246,246,117]</TT></TD>
<TD HEIGHT="18"><TT>F6F675</TT></TD></TR>
</TABLE>
</CENTER>
<P>
Note that the rendering of the hexadecimal-equivalent colors shown
here will depend on many factors.
Thus, they only approximate how RasMol will render the RGB colors on
your computer.
<p>
If you frequently wish to use a colour not predefined, you can write
a one-line script. For example, if you make the file
'<tt><b>grey.col</b></tt>'
containing the line,
'<a href="#colour"><tt><b>colour [180,180,180] #grey</b></tt></a>',
then the command
'<a href="#script"><tt><b>script grey.col</b></tt></a>'
colours the currently selected atom set grey.
<p><hr>
<a name="altcolours"><h3>Alt Colours</h3></a><p>
The RasMol
'<tt><b>alt</b></tt>'
(Alternate Conformer)
colour scheme codes the base
structure with one colour and applies a limited number of colours to each
alternate conformer.  In a RasMol built for 8-bit color systems, 4 colours
are allowed for alternate conformers. Otherwise, 8 colours are available.
<p><hr>
<a name="aminocolours"><h3>Amino Colours</h3></a><p>
The RasMol
'<tt><b>amino</b></tt>'
colour scheme colours amino acids according to traditional amino acid
properties. The purpose of colouring is to identify amino acids in an
unusual or surprising environment. The outer parts of a protein that are
polar are visible (bright) colours and non-polar residues darker. Most
colours are hallowed by tradition. This colour scheme is similar to the
'<a href="#shapelycolours"><tt><b>shapely</b></tt></a>'
scheme.
<p>
<CENTER>
<TABLE BORDER CELLPADDING=2>

<TR><TH HEIGHT="20">Amino Acids</TH>
<TH HEIGHT="20">colour Name</TH>
<TH HEIGHT="20">Sample</TH>
<TH HEIGHT="20">RGB Values</TH>
<TH HEIGHT="20">Hexadecimal</TH></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>ASP, GLU</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Bright Red</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#E60A0A"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[230,230, 10]</TT></TD>
<TD HEIGHT="20"><TT>E60A0A</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>CYS, MET</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Yellow</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#E6E600"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[230,230,  0]</TT></TD>
<TD HEIGHT="20"><TT>E6E600</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>LYS, ARG</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Blue</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#145AFF"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[ 20, 90,255]</TT></TD>
<TD HEIGHT="20"><TT>145AFF</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>SER, THR</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Orange</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FA9600"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[250,150,  0]</TT></TD>
<TD HEIGHT="20"><TT>FA9600</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>PHE, TYR</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Mid Blue</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#3232AA"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[ 50, 50,170]</TT></TD>
<TD HEIGHT="20"><TT>3232AA</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>ASN, GLN</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Cyan</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#00DCDC"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[  0,220,220]</TT></TD>
<TD HEIGHT="20"><TT>00DCDC</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>GLY</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Light Grey</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#EBEBEB"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[235,235,235]</TT></TD>
<TD HEIGHT="20"><TT>EBEBEB</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>LEU, VAL, ILE</TT></B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Green</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#0F820F"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[ 15,130, 15]</TT></TD>
<TD HEIGHT="20"><TT>0F820F</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>ALA</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Dark Grey</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#C8C8C8"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[200,200,200]</TT></TD>
<TD HEIGHT="20"><TT>C8C8C8</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>TRP</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Purple</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#B45AB4"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[180, 90,180]</TT></TD>
<TD HEIGHT="20"><TT>B45AB4</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>HIS</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Pale Blue</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#8282D2"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[130,130,210]</TT></TD>
<TD HEIGHT="20"><TT>8282D2</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>PRO</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Flesh</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#DC9682"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[220,150,130]</TT></TD>
<TD HEIGHT="20"><TT>DC9682</TT></TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TT><TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Others</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Tan</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#BEA06E"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[190,160,110]</TT></TD>
<TD HEIGHT="20"><TT>BEA06E</TT></TD></TR>
</TABLE>
</CENTER>
<P>
<p><hr>
<a name="chaincolours"><h3>Chain Colours</h3></a><p>
The RasMol
'<tt><b>chain</b></tt>'
colour scheme assigns each macromolecular chain a unique colour. This
colour scheme is particularly useful for distinguishing the parts of
multimeric structure or the individual 'strands' of a DNA chain.
'<tt><b>Chain</b></tt>'
can be selected from the RasMol
'<tt><b>Colours</b></tt>'
menu.
<p><hr>
<a name="chargecolours"><h3>Charge Colours</h3></a><p>
The RasMol
'<tt><b>charge</b></tt>'
colour scheme colour codes each atom according to the charge value
stored in the input file (or beta factor field of PDB files). High
values are coloured in blue (positive) and lower values coloured in
red (negative). Rather than use a fixed scale this scheme determines
the maximum and minimum values of the charge/temperature field and
interpolates from red to blue appropriately. Hence, green cannot be
assumed to be 'no net charge' charge.
<p>
The difference between the
'<a href="#chargecolours"><tt><b>charge</b></tt></a>'
and
'<tt><b>temperature</b></tt>'
colour schemes is that increasing temperature values proceed from blue
to red, whereas increasing charge values go from red to blue.
<p>
If the charge/temperature field stores reasonable values it is possible
to use the RasMol
'<a href="#potentialcolours"><tt><b>colour dots potential</b></tt></a>'
command to colour code a dot surface (generated by the
'<a href="#dots"><tt><b>dots</b></tt></a>'
command) by electrostatic potential.
<p><hr>
<a name="cpkcolours"><h3>CPK Colours</h3></a><p>
The RasMol
'<tt><b>cpk</b></tt>'
colour scheme is based upon the colours of the popular plastic
spacefilling models which were developed by Corey, Pauling and later
improved by Kultun. This colour scheme colours 'atom' objects by the
atom (element) type. This is the scheme conventionally used by chemists.
The assignment of the most commonly used element types to colours is
given below.
<p>
<CENTER>
<TABLE BORDER CELLPADDING=2>
<TR><TH HEIGHT="20">Element</TH>
<TH HEIGHT="20">Colour Name</TH>
<TH HEIGHT="20">Sample</TH>
<TH HEIGHT="20">RGB Values</TH>
<TH HEIGHT="20">Hexadecimal</TH></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#C8C8C8">
<B>Carbon</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#C8C8C8"><B>light grey</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#C8C8C8">&nbsp;</TD>
<TD HEIGHT="20"><TT>[200,200,200]</TT></TD>
<TD HEIGHT="20">C8C8C8</TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#F00000">
<B>Oxygen</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#F00000"><B>red</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#F00000">&nbsp;</TD>
<TD HEIGHT="20"><TT>[240,0,0]</TT></TD>
<TD HEIGHT="20">F00000</TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#FFFFFF">
<B>Hydrogen</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#FFFFFF"><TT><B>white</TT></B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF">&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,255,255]</TT></TD>
<TD HEIGHT="20">FFFFFF</TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#8F8FFF">
<B>Nitrogen</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#8F8FFF"><B>sky blue</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#8F8FFF">&nbsp;</TD>
<TD HEIGHT="20"><TT>[143,143,255]</TT></TD>
<TD HEIGHT="20">8F8FFF</TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#FFC832">
<B>Sulfur</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#FFC832"><B>yellow</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFC832">&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,200,50]</TT></TD>
<TD HEIGHT="20">FFC832</TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#FFA500">
<B>Phosphorus</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#FFA500"><TT><B>orange</TT></B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFA500">&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,165,0]</TT></TD>
<TD HEIGHT="20">FFA500</TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#00FF00">
<B>Chlorine</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#00FF00"><TT><B>green</TT></B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#00FF00">&nbsp;</TD>
<TD HEIGHT="20"><TT>[0,255,0]</TT></TD>
<TD HEIGHT="20">00FF00</TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#A52A2A">
<B>Bromine, Zinc</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#A52A2A"><B>brown</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#A52A2A">&nbsp;</TD>
<TD HEIGHT="20"><TT>[165,42,42]</TT></TD>
<TD HEIGHT="20">A52A2A</TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#0000FF">
<B>Sodium</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#0000FF"><TT><B>blue</TT></B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#0000FF">&nbsp;</TD>
<TD HEIGHT="20"><TT>[0,0,255]</TT></TD>
<TD HEIGHT="20">0000FF</TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#FFA500">
<B>Iron</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#FFA500"><TT><B>orange</TT></B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFA500">&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,165,0]</TT></TD>
<TD HEIGHT="20">FFA500</TD></TR>
<!-- 5/26/97 Changed Iron from purple to orange. WM -->
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#228B22">
<B>Magnesium</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#228B22"><B>forest green</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#228B22">&nbsp;</TD>
<TD HEIGHT="20"><TT>[34,139,34]</TT></TD>
<TD HEIGHT="20">228B22</TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#808090">
<B>Calcium</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#808090"><B>dark grey</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#808090">&nbsp;</TD>
<TD HEIGHT="20"><TT>[128,128,144]</TT></TD>
<TD HEIGHT="20">808090</TD></TR>
<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#FF1493">
<B>Unknown</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#000000"><FONT COLOR="#FF1493"><B>deep pink</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FF1493">&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,20,147]</TT></TD>
<TD HEIGHT="20">FF1493</TD></TR>
</TABLE>
</CENTER>
<P>
<p>
Note that except for <TT><B>green</B></TT>, <TT><B>white</B></TT>, <TT><B>blue</B></TT>,
and <TT><B>orange</B></TT>, these colour names are not the ones specified as
&quot;<B>Predefined colours</B>&quot; in RasMol; thus, they can only be specified
on the command line as RGB triplets.<BR>
<P>
In the CPK colouring scheme, RasMol will attempt to assign a colour to
each element from the periodic table from a list of 16 colours (the
colour codes listed are to help in understanding the mapping and are
not used by RasMol):
<P>
<CENTER>
<TABLE BORDER CELLPADDING=2>

<TR><TH HEIGHT="20">Code</TH>
<TH HEIGHT="20">colour Name</TH>
<TH HEIGHT="20">Sample</TH>
<TH HEIGHT="20">RGB Values</TH>
<TH HEIGHT="20">Hexadecimal</TH></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>LG</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Light Grey</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#C8C8C8"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[200,200,200]</TT></TD>
<TD HEIGHT="20"><TT>C8C8C8</TT></TD></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>SB</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Sky Blue</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#8F8FFF"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[143,143,255]</TT></TD>
<TD HEIGHT="20"><TT>8F8FFF</TT></TD></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>R</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Red</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#F00000"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[240,  0,  0]</TT></TD>
<TD HEIGHT="20"><TT>F00000</TT></TD></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>Y</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Yellow</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFC832"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,200, 50]</TT></TD>
<TD HEIGHT="20"><TT>FFC832</TT></TD></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>W</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>White</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,255,255]</TT></TD>
<TD HEIGHT="20"><TT>FFFFFF</TT></TD></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>Pk</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Pink</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFC0CB"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,192,203]</TT></TD>
<TD HEIGHT="20"><TT>FFC0CB</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>Go</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Golden Rod</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#DAA520"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[218,165, 32]</TT></TD>
<TD HEIGHT="20"><TT>DAA520</TT></TD></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>Bl</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Blue</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#0000FF"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[  0,  0,255]</TT></TD>
<TD HEIGHT="20"><TT>0000FF</TT></TD></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>Or</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Orange</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFA500"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,165,  0]</TT></TD>
<TD HEIGHT="20"><TT>FFA500</TT></TD></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>DG</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Dark Grey</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#808090"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[128,128,144]</TT></TD>
<TD HEIGHT="20"><TT>808090</TT></TD></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>Br</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Brown</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#A52A2A"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[165, 42, 42]</TT></TD>
<TD HEIGHT="20"><TT>A52A2A</TT></TD></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>P</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Purple</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#A020F0"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[160, 32,240]</TT></TD>
<TD HEIGHT="20"><TT>A020F0</TT></TD></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>DP</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Deep Pink</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FF1493"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[255, 20,147]</TT></TD>
<TD HEIGHT="20"><TT>FF1493</TT></TD></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>G</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Green</TT></B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#00FF00"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[  0,255,  0]</TT></TD>
<TD HEIGHT="20"><TT>00FF00</TT></TD></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>FB</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Fire Brick</TT></B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#B22222"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[178, 34, 34]</TT></TD>
<TD HEIGHT="20"><TT>B22222</TT></TD></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<B>FG</B></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Forest Green</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#228B22"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[ 34,139, 34]</TT></TD>
<TD HEIGHT="20"><TT>228B22</TT></TD></TR>
</TABLE>
</CENTER>

<p>
<CENTER>

<TABLE BORDER CELLPADDING=2>
<TR ALIGN=center>
 <TH>1a</TH><TH>2a</TH><TH>3b</TH><TH>4b</TH><TH>5b</TH><TH>6b</TH><TH>7b</TH>
 <TH COLSPAN=3>8</TH>
 <TH>1b</TH><TH>2b</TH><TH>3a</TH><TH>4a</TH><TH>5a</TH><TH>6a</TH><TH>7a</TH>
 <TH>0</TH>
</TR>

<TR ALIGN=center>
 <TD BGCOLOR="#FFFFFF"><FONT COLOR="#000000">H<BR>1 W</FONT></TD>
 <TD COLSPAN=16></TD>
 <TD BGCOLOR="#FFC0CB"><FONT  COLOR="#000000">He<BR>2 Pk</FONT></TD>
</TR>

<TR ALIGN=center>
 <TD BGCOLOR="#B22222"><FONT COLOR="#00FFFF">Li<BR>3 FB</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Be<BR>4 DP</FONT></TD>
 <TD COLSPAN=10></TD>
 <TD BGCOLOR="#00FF00"><FONT COLOR="#000000">B<BR>5 G</FONT></TD>
 <TD BGCOLOR="#C8C8C8"><FONT COLOR="#000000">C<BR>6 LG</FONT></TD>
 <TD BGCOLOR="#8F8FFF"><FONT COLOR="#000000">N<BR>7 SB</FONT></TD>
 <TD BGCOLOR="#F00000"><FONT COLOR="#00FFFF">O<BR>8 R</FONT></TD>
 <TD BGCOLOR="#DAA520"><FONT COLOR="#000000">F<BR>9 Go</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Ne<BR>10 DP</FONT></TD>
</TR>

<TR ALIGN=center>
 <TD BGCOLOR="#0000FF"><FONT COLOR="#FFFF00">Na<BR>11 Bl</FONT></TD>
 <TD BGCOLOR="#228B22"><FONT COLOR="#00FFFF">Mg<BR>12 FG</FONT></TD>
 <TD COLSPAN=10></TD>
 <TD BGCOLOR="#808090"><FONT COLOR="#00FFFF">Al<BR>13 DG</FONT></TD>
 <TD BGCOLOR="#DAA520"><FONT COLOR="#000000">Si<BR>14 Go</FONT></TD>
 <TD BGCOLOR="#FFA500"><FONT COLOR="#000000">P<BR>15 Or</FONT></TD>
 <TD BGCOLOR="#FFC832"><FONT COLOR="#000000">S<BR>16 Y</FONT></TD>
 <TD BGCOLOR="#00FF00"><FONT COLOR="#000000">Cl<BR>17 G</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Ar<BR>18 DP</FONT></TD>
</TR>

<TR ALIGN=center>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">K<BR>19 DP</FONT></TD>
 <TD BGCOLOR="#808090"><FONT COLOR="#00FFFF">Ca<BR>20 DG</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Sc<BR>21 DP</FONT></TD>
 <TD BGCOLOR="#808090"><FONT COLOR="#00FFFF">Ti<BR>22 DG</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">V<BR>23 DP</FONT></TD>
 <TD BGCOLOR="#808090"><FONT COLOR="#00FFFF">Cr<BR>24 DG</FONT></TD>
 <TD BGCOLOR="#808090"><FONT COLOR="#00FFFF">Mn<BR>25 DG</FONT></TD>
 <TD BGCOLOR="#FFA500"><FONT COLOR="#000000">Fe<BR>26 Or</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Co<BR>27 DP</FONT></TD>
 <TD BGCOLOR="#A52A2A"><FONT COLOR="#00FFFF">Ni<BR>28 Br</FONT></TD>
 <TD BGCOLOR="#A52A2A"><FONT COLOR="#00FFFF">Cu<BR>29 Br</FONT></TD>
 <TD BGCOLOR="#A52A2A"><FONT COLOR="#00FFFF">Zn<BR>30 Br</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Ga<BR>31 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Ge<BR>32 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">As<BR>33 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Se<BR>34 DP</FONT></TD>
 <TD BGCOLOR="#A52A2A"><FONT COLOR="#00FFFF">Br<BR>35 Br</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Kr<BR>36 DP</FONT></TD>
</TR>

<TR ALIGN=center>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Rb<BR>37 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Sr<BR>38 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Y<BR>39 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Zr<BR>40 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Nb<BR>41 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Mo<BR>42 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Tc<BR>43 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Ru<BR>44 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Rh<BR>45 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Pd<BR>46 DP</FONT></TD>
 <TD BGCOLOR="#808090"><FONT COLOR="#00FFFF">Ag<BR>47 DG</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Cd<BR>48 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">In<BR>49 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Sn<BR>50 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Sb<BR>51 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Te<BR>52 DP</FONT></TD>
 <TD BGCOLOR="#A020F0"><FONT COLOR="#00FFFF">I<BR>53 P</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Xe<BR>54 DP</FONT></TD>
</TR>

<TR ALIGN=center>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Cs<BR>55 DP</FONT></TD>
 <TD BGCOLOR="#FFA500"><FONT COLOR="#000000">Ba<BR>56 Or</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">La<BR>57 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Hf<BR>72 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Ta<BR>73 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">W<BR>74 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Re<BR>75 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Os<BR>76 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Ir<BR>77 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Pt<BR>78 DP</FONT></TD>
 <TD BGCOLOR="#DAA520"><FONT COLOR="#000000">Au<BR>79 Go</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Hg<BR>80 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Tl<BR>81 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Pb<BR>82 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Bi<BR>83 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Po<BR>84 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">At<BR>85 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Rn<BR>86 DP</FONT></TD>
</TR>

<TR ALIGN=center>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Fr<BR>87 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Ra<BR>88 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Ac<BR>89 DP</FONT></TD>
 <TD><COLSPAN=15></TD></TR>
</TR>

<TR><TD COLSPAN=18></TD></TR>
<TR><TD COLSPAN=18></TD></TR>

<TR ALIGN=center>
 <TD COLSPAN=2><FONT SIZE=-1>Lanthinide<BR>Series</FONT></FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Ce<BR>58 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Pr<BR>59 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Nd<BR>60 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Pm<BR>61 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Sm<BR>62 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Eu<BR>63 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Gd<BR>64 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Tb<BR>65 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Dy<BR>66 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Ho<BR>67 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Er<BR>68 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Tm<BR>69 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Yb<BR>70 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Lu<BR>71 DP</FONT></TD>
 <TD COLSPAN=2></TD>
</TR>

<TR ALIGN=center>
 <TD COLSPAN=2><FONT SIZE=-1>Actinide<BR>Series</FONT></FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Th<BR>90 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Pa<BR>91 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">U<BR>92 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Np<BR>93 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Pu<BR>94 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Am<BR>95 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Cm<BR>96 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Bk<BR>97 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Cf<BR>98 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Es<BR>99 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Fm<BR>100 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Md<BR>101 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">No<BR>102 DP</FONT></TD>
 <TD BGCOLOR="#FF1493"><FONT COLOR="#00FFFF">Lr<BR>103 DP</FONT></TD>
 <TD COLSPAN=2></TD>
</TR>
</TABLE>

</CENTER>
<P>
<!-- David Hackney (CMU) suggests: -->
For X-ray crystallographic models of proteins and nucleic acids
(<I>i.e.</I> without hydrogens) the display can be 'brightened' by converting
the O, C, and N atoms from the RasMol default <TT><B>cpk</B></TT> colors to
&quot;true red, white and blue&quot; using RasMol's predefined color scheme.
Use the following sequence of
commands to try it:
<PRE><B>
     select all
     select oxygen
     color red
     select carbon
     color white
     select nitrogen
     color blue
     select all
</B>
</PRE>
Extension of this idea to other atoms and color schemes is straightforward.
<P>
<p><hr>
<a name="groupcolours"><h3>Group Colours</h3></a><p>
The RasMol
'<tt><b>group</b></tt>'
colour scheme colour codes residues by their position in a macromolecular
chain. Each chain is drawn as a smooth spectrum from blue through green,
yellow and orange to red. Hence the N terminus of proteins and 5' terminus
of nucleic acids are coloured red and the C terminus of proteins and 3'
terminus of nucleic acids are drawn in blue. If a chain has a large number
of heterogeneous molecules associated with it, the macromolecule may not be
drawn in the full 'range' of the spectrum.
'<tt><b>Group</b></tt>'
can be selected from the RasMol
'<tt><b>Colours</b></tt>'
menu.
<p>
If a chain has a large number of heterogeneous molecules associated with it,
the macromolecule may not be drawn in the full range of the spectrum. When
RasMol performs group coloring it decides the range of colors it uses from
the residue numbering given in the PDB file. Hence the lowest residue
number is displayed in blue and the highest residue number is displayed as
red. Unfortunately, if a PDB file contains a large number of heteroatoms,
such as water molecules, that occupy the high residue numbers, the protein
is displayed in the blue-green end of the spectrum and the waters in the
yellow-red end of the spectrum. This is aggravated by there typically being
many more water molecules than amino acid residues. The solution to this
problem is to use the command
'<a href="#sethetero"><tt><b>set hetero off</b></tt></a>'
before applying the group
color scheme. This can also be achieved by toggling
'<tt><b>Hetero Atoms</b></tt>'
on the
'<tt><b>Options</b></tt>'
menu before selecting
'<tt><b>Group</b></tt>'
on the
'<tt><b>Colour</b></tt>'
menu. This command
instructs RasMol to only use non-hetero residues in the group color scaling.
<p><hr>
<a name="nmrmodelcolours"><h3>NMR Model Colours</h3></a><p>
The RasMol
'<tt><b>model</b></tt>'
colour scheme codes each NMR model with a distinct
colour.  The NMR model number is taken as a numeric value.  High values
are coloured in blue and lower values coloured in red. Rather than use a
fixed scale this scheme determines the maximum value of the
NMR model number and interpolates from red to blue appropriately.
<p><hr>
<a name="shapelycolours"><h3>Shapely Colours</h3></a><p>
The RasMol
'<tt><b>shapely</b></tt>'
colour scheme colour codes residues by amino acid property. This scheme
is based upon Bob Fletterick's &quot;Shapely Models&quot;. Each amino acid and
nucleic acid residue is given a unique colour. The
'<tt><b>shapely</b></tt>'
colour scheme is used by David Bacon's Raster3D program. This colour
scheme is similar to the
'<a href="#aminocolours"><tt><b>amino</b></tt></a>'
colour scheme.
<p>
<CENTER>
<TABLE BORDER CELLPADDING=2>

<TR><TH HEIGHT="20">Residues</TH>
<TH HEIGHT="20">Color Name</TH>
<TH HEIGHT="20">Sample</TH>
<TH HEIGHT="20">RGB Values</TH>
<TH HEIGHT="20">Hexadecimal</TH></TR>

<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>ALA</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Medium Green</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="8CFF8C"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[140,255,140]</TT></TD>
<TD HEIGHT="20"><TT>8CFF8C</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>GLY</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>White</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="FFFFFF"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,255,255]</TT></TD>
<TD HEIGHT="20"><TT>FFFFFF</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>LEU</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Olive Green</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="455E45"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[ 69, 94, 69]</TT></TD>
<TD HEIGHT="20"><TT>455E45</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>SER</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Medium Orange</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="FF7042"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,112, 66]</TT></TD>
<TD HEIGHT="20"><TT>FF7042</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>VAL</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Light Purple</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="FF8CFF"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,140,255]</TT></TD>
<TD HEIGHT="20"><TT>FF8CFF</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>THR</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Dark Orange</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="B84C00"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[184, 76, 0]</TT></TD>
<TD HEIGHT="20"><TT>B84C00</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>LYS</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Royal Blue</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="4747B8"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[ 71, 71,184]</TT></TD>
<TD HEIGHT="20"><TT>4747B8</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>ASP</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Dark Rose</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="A00042"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[160,0,66]</TT></TD>
<TD HEIGHT="20"><TT>A00042</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>ILE</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Dark Green</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="004C00"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[  0, 76,  0]</TT></TD>
<TD HEIGHT="20"><TT>004C00</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>ASN</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Light Salmon</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="FF7C70"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,124,112]</TT></TD>
<TD HEIGHT="20"><TT>FF7C70</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>GLU</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Dark Brown</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="660000"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[102,  0,  0]</TT></TD>
<TD HEIGHT="20"><TT>660000</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>PRO</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Dark Grey</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="525252"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[ 82, 82, 82]</TT></TD>
<TD HEIGHT="20"><TT>525252</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>ARG</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Dark Blue</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="00007C"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[  0,  0,124]</TT></TD>
<TD HEIGHT="20"><TT>00007C</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>PHE</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Olive Grey</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="534C42"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[ 83, 76, 66]</TT></TD>
<TD HEIGHT="20"><TT>534C42</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>GLN</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Dark Salmon</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="FF4C4C"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[255, 76, 76]</TT></TD>
<TD HEIGHT="20"><TT>FF4C4C</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>TYR</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Medium Brown</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="8C704C"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[140,112,76]</TT></TD>
<TD HEIGHT="20"><TT>8C704C</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>HIS</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Medium Blue</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="7070FF"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[112,112,255]</TT></TD>
<TD HEIGHT="20"><TT>7070FF</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>CYS</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Medium Yellow</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="FFFF70"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,255,112]</TT></TD>
<TD HEIGHT="20"><TT>FFFF70</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>MET</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Light Brown</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="B8A042"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[184,160, 66]</TT></TD>
<TD HEIGHT="20"><TT>B8A042</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>TRP</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Olive Brown</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="4F4600"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[ 79, 70,  0]</TT></TD>
<TD HEIGHT="20"><TT>4F4600</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>ASX,GLX,PCA,HYP</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Medium Purple</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="FF00FF"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,  0,255]</TT></TD>
<TD HEIGHT="20"><TT>FF00FF</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>A</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Light Blue</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="A0A0FF"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[160,160,255]</TT></TD>
<TD HEIGHT="20"><TT>A0A0FF</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>C</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Light Orange</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="FF8C4B"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,140,75]</TT></TD>
<TD HEIGHT="20"><TT>FF8C4B</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>G</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Medium Salmon</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="FF7070"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,112,112]</TT></TD>
<TD HEIGHT="20"><TT>FF7070</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>T</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Light Green</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="A0FFA0"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[160,255,160]</TT></TD>
<TD HEIGHT="20"><TT>A0FFA0</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Backbone</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Light Grey</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="B8B8B8"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[184,184,184]</TT></TD>
<TD HEIGHT="20"><TT>B8B8B8</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Special</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Dark Purple</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="5E005E"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[ 94,  0, 94]</TT></TD>
<TD HEIGHT="20"><TT>5E005E</TT></TD></TR>


<TR VALIGN=MIDDLE ALIGN=LEFT>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000">
<TT><B>Default</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="#FFFFFF"><FONT COLOR="#000000"><TT><B>Medium Purple</B></TT></FONT></TD>
<TD HEIGHT="20" BGCOLOR="FF00FF"><TT>&nbsp;</TD>
<TD HEIGHT="20"><TT>[255,  0,255]</TT></TD>
<TD HEIGHT="20"><TT>FF00FF</TT></TD></TR>
</TABLE>
</CENTER>
<P>

<p><hr>
<a name="structurecolours"><h3>Structure Colours</h3></a><p>
The RasMol
'<tt><b>structure</b></tt>'
colour scheme colours the molecule by protein secondary structure.
Alpha helices are coloured magenta, [240,0,128], beta sheets are
coloured yellow, [255,255,0], turns are coloured pale blue, [96,128,255]
and all other residues are coloured white. The secondary structure
is either read from the PDB file (HELIX, SHEET and TURN records), if available,
or determined using Kabsch and Sander's DSSP algorithm. The RasMol
'<a href="#structure"><tt><b>structure</b></tt></a>'
command may be used to force DSSP's structure assignment to be used.
<p><hr>
<a name="temperaturecolours"><h3>Temperature Colours</h3></a><p>
The RasMol
'<tt><b>temperature</b></tt>'
colour scheme colour codes each atom according to the anisotropic
temperature (beta) value stored in the PDB file. Typically this gives
a measure of the mobility/uncertainty of a given atom's position. High
values are coloured in warmer (red) colours and lower values in colder
(blue) colours. This feature is often used to associate a &quot;scale&quot; value
[such as amino acid variability in viral mutants] with each atom in a
PDB file, and colour the molecule appropriately.
<p>
The difference between the
'<tt><b>temperature</b></tt>'
and
'<a href="#chargecolours"><tt><b>charge</b></tt></a>'
colour schemes is that increasing temperature values proceed from blue
to red, whereas increasing charge values go from red to blue.
<p><hr>
<a name="usercolours"><h3>User Colours</h3></a><p>
The RasMol
'<tt><b>user</b></tt>'
colour scheme allows RasMol to use the colour scheme stored in the
PDB file. The colours for each atom are stored in COLO records placed
in the PDB data file. This convention was introduced by David Bacon's
Raster3D program.
<p><hr>
<a name="hbondtypecolours"><h3>HBond Type Colours</h3></a><p>
The RasMol
'<tt><b>type</b></tt>'
colour scheme applies only to hydrogen bonds, hence is used in the command
'<tt><b>colour hbonds type</b></tt>'.
This scheme colour codes each hydrogen bond according to the
distance along a protein chain between hydrogen bond donor and acceptor.
This schematic representation was introduced by Belhadj-Mostefa and
Milner-White. This representation gives a good insight into protein
secondary structure (hbonds forming alpha helices appear red, those
forming sheets appear yellow and those forming turns appear magenta).
<p>
<pre>
      Offset    Colour    Triple
        +2      white     [255,255,255]
        +3      magenta   [255,0,255]
        +4      red       [255,0,0]
        +5      orange    [255,165,0]
        -3      cyan      [0,255,255]
        -4      green     [0,255,0]
      default   yellow    [255,255,0]
</pre>
<p><hr>
<a name="potentialcolours"><h3>Potential Colours</h3></a><p>
The RasMol
'<tt><b>potential</b></tt>'
colour scheme applies only to dot surfaces, hence is used in the command
'<tt><b>colour dots potential</b></tt>'.
This scheme colours each currently displayed dot by the electrostatic
potential at that point in space. This potential is calculated using
Coulomb's law taking the temperature/charge field of the input file to
be the charge assocated with that atom. This is the same interpretation
used by the
'<a href="#chargecolours"><tt><b>colour charge</b></tt></a>'
command. Like the
'<a href="#chargecolours"><tt><b>charge</b></tt></a>'
colour scheme low values are blue/white and high values are red.
The table below shows the static assignment of colours using a
dielectric constant value of 10.
<p>
<pre>
     25 < V          red       [255,0,0]
     10 < V <  25    orange    [255,165,0]
      3 < V <  10    yellow    [255,255,0]
      0 < V <   3    green     [0,255,0]
     -3 < V <   0    cyan      [0,255,255]
    -10 < V <   3    blue      [0,0,255]
    -25 < V < -10    purple    [160,32,240]
          V < -25    white     [255,255,255]
</pre>
<p><hr>
<a name="aminoacidcodes"><h3>Amino Acid Codes</h3></a><p>
The following table lists the names, single letter and three letter
codes of each of the amino acids.
<p>
<center><table BORDER CELLPADDING=2>
<TR><TD>Alanine         <TD>A  <TD>ALA         <TD>Arginine        <TD>R  <TD>ARG
<TR><TD>Asparagine      <TD>N  <TD>ASN         <TD>Aspartic acid   <TD>D  <TD>ASP
<TR><TD>Cysteine        <TD>C  <TD>CYS         <TD>Glutamic acid   <TD>E  <TD>GLU
<TR><TD>Glutamine       <TD>Q  <TD>GLN         <TD>Glycine         <TD>G  <TD>GLY
<TR><TD>Histidine       <TD>H  <TD>HIS         <TD>Isoleucine      <TD>I  <TD>ILE
<TR><TD>Leucine         <TD>L  <TD>LEU         <TD>Lysine          <TD>K  <TD>LYS
<TR><TD>Methionine      <TD>M  <TD>MET         <TD>Phenylalanine   <TD>F  <TD>PHE
<TR><TD>Proline         <TD>P  <TD>PRO         <TD>Serine          <TD>S  <TD>SER
<TR><TD>Threonine       <TD>T  <TD>THR         <TD>Tryptophan      <TD>W  <TD>TRP
<TR><TD>Tyrosine        <TD>Y  <TD>TYR         <TD>Valine          <TD>V  <TD>VAL
</table></center>
<p><hr>
<a name="booleans"><h3>Booleans</h3></a><p>
A boolean parameter is a truth value. Valid boolean values are 'true' and
'false', and their synonyms 'on' and 'off'. Boolean parameters are commonly
used by RasMol to either enable or disable a representation or option.
<p><hr>
<a name="chfile"><h2>File Formats</h2></a>
<p>
<a name="proteindatabankfiles"><h3>Protein Data Bank Files</h3></a><p>
If you do not have the PDB documentation, you may find the
following summary of the PDB file format useful. The Protein Data Bank
is a computer-based archival database for macromolecular structures.
The database was established in 1971 by Brookhaven National
Laboratory, Upton, New York, as a public domain repository for resolved
crystallographic structures. The Bank uses a uniform format to store
atomic coordinates and partial bond connectivities as derived from
crystallographic studies.  In 1999 the Protein Data Bank moved
to the Research Collaboratory for Structural Biology.
<p>
PDB file entries consist of records of 80 characters each. Using the
punched card analogy, columns 1 to 6 contain a record-type identifier,
the columns 7 to 70 contain data. In older entries, columns 71 to 80
are normally blank,
but may contain sequence information added by library management
programs.  In new entries conforming to the 1996 PDB format, there
is other information in those columns. The first four characters of
the record identifier are
sufficient to identify the type of record uniquely, and the syntax of
each record is independent of the order of records within any entry for
a particular macromolecule.
<p>
The only record types that are of major interest to the RasMol program
are the ATOM and HETATM records which describe the position of each
atom. ATOM/HETATM records contain standard atom names and residue
abbreviations, along with sequence identifiers, coordinates in
&Aring;ngstrom units, occupancies and thermal motion factors. The exact
details are given below as a FORTRAN format statement.  The &quot;fmt&quot;
column indicates use of the field in all PDB formats,
in the 1992 and earlier formats or in the 1996 and later formats.
<pre>
FORMAT(6A1,I5,1X,A4,A1,A3,1X,A1,I4,A1,3X,3F8.3,2F6.2,1X,I3,2X,A4,2A2)
</pre>
<CENTER><TABLE BORDER CELLPADDING=2>

<TR><TH>Column<TH ALIGN=left>Content<TH>fmt
<TR><TD>1-6<TD>'ATOM' or 'HETATM'<TD>all
<TR><TD>7-11<TD>Atom serial number (may have gaps)<TD>all
<TR><TD>13-16<TD>Atom name, in IUPAC standard format<TD>all
<TR><TD>17<TD>Alternate location indicator indicated by A, B or C<TD>all
<TR><TD>18-20<TD>Residue name, in IUPAC standard format<TD>all
<TR><TD>23-26<TD>Residue sequence number<TD>all
<TR><TD>27<TD>Code for insertions of residues (i.e. 66A & 66B)<TD>all
<TR><TD>31-38<TD>X coordinate<TD>all
<TR><TD>39-46<TD>Y coordinate<TD>all
<TR><TD>47-54<TD>Z coordinate<TD>all
<TR><TD>55-60<TD>Occupancy<TD>all
<TR><TD>61-66<TD>Temperature factor<TD>all
<TR><TD>68-70<TD>Footnote number<TD>92
<TR><TD>73-76<TD>Segment Identifier (left-justified)<TD>96
<TR><TD>77-78<TD>Element Symbol (right-justified)<TD>96
<TR><TD>79-80<TD>Charge on the Atom<TD>96
</TABLE></CENTER>
<p>
Residues occur in order starting from the N-terminal residue
for proteins and
5'-terminus for nucleic acids. If the residue sequence is known,
certain atom serial numbers may be omitted to allow for future insertion
of any missing atoms. Within each residue, atoms are ordered in a
standard manner, starting with the backbone (N-C-C-O for proteins) and
proceeding in increasing remoteness from the alpha carbon, along the
side chain.
<p>
HETATM records are used to define post-translational modifications and
cofactors associated with the main molecule. TER records are
interpreted as breaks in the main molecule's backbone.
<p>
If present, RasMol also inspects HEADER, COMPND, HELIX, SHEET, TURN,
CONECT, CRYST1, SCALE, MODEL, ENDMDL, EXPDTA and END records. Information such as the
name, database code, revision date and classification of the molecule
are extracted from HEADER and COMPND records, initial secondary
structure assignments are taken from HELIX, SHEET and TURN records, and
the end of the file may be indicated by an END record.
<p><hr>
<a name="rasmolinterpretationofpdbfields"><h3>RasMol Interpretation of PDB fields</h3></a><p>
Atoms located at 9999.000, 9999.000, 9999.000 are assumed to be Insight
pseudo atoms and are ignored by RasMol. Atom names beginning ' Q' are
also assumed to be pseudo atoms or position markers.
<p>
When a data file contains an NMR structure, multiple conformations may
be placed in a single PDB file delimited by pairs of MODEL and ENDMDL
records. RasMol displays all the NMR models contained in the file.
<p>
Residue names &quot;CSH&quot;, &quot;CYH&quot; and &quot;CSM&quot; are considered pseudonyms for
cysteine &quot;CYS&quot;. Residue names &quot;WAT&quot;, &quot;H20&quot;, &quot;SOL&quot; and &quot;TIP&quot; are
considered pseudonyms for water &quot;HOH&quot;. The residue name &quot;D20&quot; is
consider heavy water &quot;DOD&quot;. The residue name &quot;SUL&quot; is considered a
sulphate ion &quot;SO4&quot;. The residue name &quot;CPR&quot; is considered to be
cis-proline and is translated as &quot;PRO&quot;. The residue name &quot;TRY&quot; is
considered a pseudonym for tryptophan &quot;TRP&quot;.
<p>
RasMol uses the HETATM fields to define the sets hetero, water, solvent
and ligand. Any group with the name &quot;HOH&quot;, &quot;DOD&quot;, &quot;SO4&quot; or &quot;PO4&quot; (or
aliased to one of these names by the preceding rules) is considered a
solvent and is considered to be defined by a HETATM field.
<p>
RasMol only respects CONECT connectivity records in PDB files containing
fewer than 256 atoms. This is explained in more detail in the section on
determining molecule connectivity. CONECT records that define a bond
more than once are interpreted as specifying the bond order of that
bond, <i>i.e.</i> a bond specified twice is a double bond and a bond specified
three (or more) times is a triple bond.  This is not a standard PDB feature.
<p><hr>
<a name="pdbcolourschemespecification"><h3>PDB Colour Scheme Specification</h3></a><p>
RasMol also accepts the supplementary COLO record type in the PDB
files. This record format was introduced by David Bacon's Raster3D
program for specifying the colour scheme to be used when rendering the
molecule. This extension is not currently supported by the PDB. The
COLO record has the same basic record type as the ATOM and HETATM
records described above.
<p>
Colours are assigned to atoms using a matching process. The Mask field
is used in the matching process as follows. First RasMol reads in and
remembers all the ATOM, HETATM and COLO records in input order. When the
user-defined ('User') colour scheme is selected, RasMol goes through
each remembered ATOM/HETATM record in turn, and searches for a COLO
record that matches in all of columns 7 through 30. The first such COLO
record to be found determines the colour and radius of the atom.
<p>
<CENTER><TABLE BORDER CELLPADDING=2>

<TR><TH>Column<TH ALIGN=left>Content
<TR><TD>1-6<TD>'COLOR' or 'COLOUR'
<TR><TD>7-30<TD>Mask (described below)
<TR><TD>31-38<TD>Red component
<TR><TD>39-46<TD>Green component
<TR><TD>47-54<TD>Blue component
<TR><TD>55-60<TD>Sphere radius in &Aring;ngstroms
<TR><TD>61-70<TD>Comments
</TABLE></CENTER>
<p>
Note that the Red, Green and Blue components are in the same positions
as the X, Y, and Z components of an ATOM or HETA record, and the van
der Waals radius goes in the place of the Occupancy. The Red, Green and
Blue components must all be in the range 0 to 1.
<p>
In order that one COLO record can provide colour and radius
specifications for more than one atom (e.g. based on residue, atom
type, or any other criterion for which labels can be given somewhere in
columns 7 through 30), a 'don't-care' character, the hash mark &quot;#&quot;
(number or sharp sign) is used. This character, when found in a COLO
record, matches any character in the corresponding column in a
ATOM/HETATM record. All other characters must match identically to count
as a match. As an extension to the specification, any atom that fails
to match a COLO record is displayed in white.
<p><hr>
<a name="multiplenmrmodels"><h3>Multiple NMR Models</h3></a><p>
RasMol loads all of the NMR models from a PDB file no matter which
command is used:
'<a href="#load"><tt><b>load pdb &lt;filename&gt;</b></tt></a>'
or
'<a href="#load"><tt><b>load nmrpdb &lt;filename&gt;</b></tt></a>'
<p>
Once multiple NMR conformations have been loaded they may be
manipulated with the atom expression extensions described in
'<a href="#primitiveexpressions"><tt><b>Primitive Expressions</b></tt></a>'.
In particular, the command
'<a href="#restrict"><tt><b>restrict */1</b></tt></a>'
will restrict the display to the first model only.
<p><hr>
<a name="cifandmmcifformatfiles"><h3>CIF and mmCIF Format Files</h3></a><p>
CIF is the IUCr standard for presentation of small molecules and mmCIF
is intended as the replacement for the fixed-field PDB format for
presentation of macromolecular structures. RasMol can accept data sets
in either format.
<p>
There are many useful sites on the World Wide Web where information
tools and software related to CIF, mmCIF and the PDB can be found. The
following are good starting points for exploration:
<p>
The International Union of Crystallography (IUCr) provides access to
software, dictionaries, policy statements and documentation relating to
CIF and mmCIF at: IUCr, Chester, England (www.iucr.org/iucr-top/cif/)
with many mirror sites.
<p>
The Nucleic Acid Database Project provides access to its entries,
software and documentation, with an mmCIF page giving access to the
dictionary and mmCIF software tools at Rutgers University, New Jersey,
USA (http://ndbserver.rutgers.edu/NDB/mmcif) with many mirror sites.
<p>
This version of RasMol restricts CIF or mmCIF tag values to essentially
the same conventions as are used for the fixed-field PDB format. Thus
chain identifiers and alternate conformation identifiers are limited to
a single character, atom names are limited to 4 characters, etc. RasMol
interprets the following CIF and mmCIF tags:
<CENTER><TABLE BORDER CELLPADDING=2>
<TR><TH ALIGN=LEFT>mmCIF tag<TH ALIGN=LEFT>CIF tag<TH ALIGN=LEFT>Used for
<TR><TD>&nbsp;<TD>&nbsp;<TD>&nbsp;
<TR><TD>_struct_biol.details<TD>&nbsp;<TD>Info.classification
<TR><TD>_database_2.database_code<TD>&nbsp;<TD>Info.identcode
<TR><TD>_entry.id <TD>&nbsp;<TD>&nbsp;
<TR><TD>_struct_biol.id<TD>&nbsp;<TD>&nbsp;
<TR><TD>_struct.title<TD>&nbsp;Info.moleculename
<TR><TD>_chemical_name_common <TD>&nbsp;<TD>&nbsp;
<TR><TD>_chemical_name_systematic<TD>&nbsp;<TD>&nbsp;
<TR><TD>_chemical_name_mineral<TD>&nbsp;<TD>&nbsp;
<TR><TD> &nbsp;<TD>&nbsp;<TD>&nbsp;
<TR><TD>_symmetry.space_group_name_H-M<TD>_symmetry_space_group_name_H-M<TD> Info.spacegroup
<TR><TD>_cell.length_a<TD>_cell_length_a<TD>Info.cell
<TR><TD>_cell.length_b<TD>_cell_length_b<TD>&nbsp;
<TR><TD>_cell.length_c<TD>_cell_length_c<TD>&nbsp;
<TR><TD>_cell.angle_alpha<TD>_cell_angle_alpha<TD>&nbsp;
<TR><TD>_cell.angle_alpha<TD>_cell_angle_alpha<TD>&nbsp;
<TR><TD>_cell.angle_beta<TD>_cell_angle_beta<TD>&nbsp;
<TR><TD>_cell.angle_gamma<TD>_cell_angle_gamma<TD>&nbsp;
<TR><TD>&nbsp;<TD>&nbsp;<TD>&nbsp;
<TR><TD>_atom_sites.fract_transf_matrix[1][1]<TD> _atom_sites_fract_tran_matrix_11<TD>Used to compute orthogonal coords
<TR><TD>...<TD>...<TD>&nbsp;
<TR><TD>_atom_sites.fract_transf_vector[1]<TD> _atom_sites_fract_tran_vector_1<TD>&nbsp;
<TR><TD>...<TD>...<TD>&nbsp;
<TR><TD>_atom_sites.cartn_transf_matrix[1][1]<TD>_atom_sites_cartn_tran_matrix_11<TD> Alternative to compute orth. coords
<TR><TD>...<TD>...<TD>&nbsp;
<TR><TD>_atom_sites.cartn_transf_vector[1]<TD> _atom_sites_cartn_tran_vector_1<TD>&nbsp;
<TR><TD>...<TD>...<TD>&nbsp;
<TR><TD>_atom_site.cartn_x<TD>_atom_site_cartn_x<TD>atomic coordinates
<TR><TD>...<TD>...<TD>&nbsp;
<TR><TD>or<TD>&nbsp;<TD>&nbsp;
<TR><TD>_atom_site.fract_x<TD>_atom_site_fract_x<TD>&nbsp;
<TR><TD>...<TD>...<TD>&nbsp;
<TR><TD>&nbsp;<TD>&nbsp;<TD>&nbsp;
<TR><TD>_struct_conn.id<TD>&nbsp;<TD>bonds
<TR><TD>...<TD>&nbsp<TD>&nbsp;
<TR><TD>_geom_bond.atom_site_id_1<TD>_geom_bond_atom_site_label_1<TD>&nbsp;
<TR><TD>...<TD>...<TD>&nbsp;
<TR><TD>&nbsp;<TD>&nbsp;<TD>&nbsp;
<TR><TD>_struct_conf.id<TD>&nbsp;<TD>helices, sheets, turns
<TR><TD>_struct_sheet_range.id<TD>&nbsp;<TD>&nbsp;
<TR><TD>...<TD>&nbsp;<TD>&nbsp;
<TR><TD>
</table></CENTER>
A search is made through multiple data blocks for the desired tags, so
a single dataset may be composed from multiple data blocks, but
multiple data sets may not be stacked in the same file.
<p><hr>
<a name="chmacspec"><h2>Machine-Specific Support</h2></a>
<p>
In the following sections, support for
'<a href="#monochromexwindowssupport"><tt><b>Monochrome X-Windows</b></tt></a>',
'<a href="#tcltkipcsupport"><tt><b>Tcl/Tk IPC</b></tt></a>',
'<a href="#unixsocketsbasedipc"><tt><b>UNIX sockets based IPC</b></tt></a>',
'<a href="#compilingraswinwithborlandandmetrowerks"><tt><b>Compiling RasWin with Borland and MetroWerks</b></tt></a>'
are described.
<p><hr>
<a name="monochromexwindowssupport"><h3>Monochrome X-Windows Support</h3></a><p>
RasMol supports the many
monochrome UNIX workstations typically found in academia, such as low-end
SUN workstations and NCD X-terminals. The X11 version of RasMol (when
compiled in 8 bit mode) now detects black and white X-Windows displays and
enables dithering automatically. The use of run-time error diffusion dithering
means that all display modes of RasMol are available when in monochrome
mode. For best results, users should experiment with the set ambient
command to ensure the maximum contrast in resulting images.
<p><hr>
<a name="tcltkipcsupport"><h3>Tcl/Tk IPC support</h3></a><p>
Version 4 of Tk
graphics library changed the protocol used to communicate between Tk
applications. RasMol version 2.6 was modified such that it could
communicate with both this new protocol and the previous version 3 protocol
supported by RasMol v2.5. Although Tcl/Tk 3.x applications may only
communicate with other 3.x applications and Tcl/Tk 4.x applications with other
4.x applications, these changes allow RasMol to communicate between
processes with both protocols (potentially concurrently).
<p><hr>
<a name="unixsocketsbasedipc"><h3>UNIX sockets based IPC</h3></a><p>
The UNIX implementation of RasMol
supports BSD-style socket communication. An identical socket mechanism is
also being developed for VMS, Apple Macintosh and Microsoft Windows
systems. This should allow RasMol to interactively display results of a
computation on a remote host. The current protocol acts as a TCP/IP server on
port 21069 that executes command lines until either the command
'<tt><b>exit</b></tt>'
or the
command
'<tt><b>quit</b></tt>'
is typed. The command
exit
from the RasMol server, the command
'<tt><b>quit</b></tt>'
both disconnects the current
session and terminates RasMol. This functionality may be tested using the
UNIX command
'<tt><b>telnet &lt;hostname&gt; 21069</b></tt>'.
<p><hr>
<a name="compilingraswinwithborlandandmetrowerks"><h3>Compiling RasWin with Borland and MetroWerks</h3></a><p>
A number of changes were made to the
source code in the transition from version 2.5 to 2.6 to allow
the Microsoft Windows version of RasMol to compile
using the Borland C/C++ compiler. These fixes include name changes for the
standard library and special code to avoid a bug in _fmemset.
Additional changes were made in the transition from 2.6 to 2.7 to
allow compilation with the MetroWerks compilers.
<p><hr>
<a name="chbib"><h2>Bibliography</h2></a>
<p>
<a name="moleculargraphics"><h3>Molecular Graphics</h3></a><p>
[1] Nelson Max, &quot;Computer Representation of Molecular Surfaces&quot;, IEEE
Computer Graphics and Applications, pp.21-29, August 1983.
<p>
[2] Arthur M. Lesk, &quot;Protein Architecture: A Practical Approach&quot;, IRL
Press Publishers, 1991.
<p>
<a name="moleculargraphicsprograms"><h3>Molecular Graphics Programs</h3></a><p>
[3] Per J. Kraulis, &quot;MOLSCRIPT: A Program to Produce both Detailed and
Schematic Plots of Protein Structures&quot;, Journal of Applied
Crystallography, Vol.24, pp.946-950, 1991.
<p>
[4] David Bacon and Wayne F. Anderson, &quot;A Fast Algorithm for Rendering
Space-Filling Molecule Pictures&quot;, Journal of Molecular Graphics, Vol.6,
No.4, pp.219-220, December 1988.
<p>
[5] David C. Richardson and Jane S. Richardson, &quot;The Kinemage: A tool
for Scientific Communication&quot;, Protein Science, Vol.1, No.1,pp.3-9,
January 1992.
<p>
[6] Mike Carson, &quot;RIBBONS 2.0&quot;, Journal of Applied Crystallography,
Vol.24, pp.958-961, 1991.
<p>
[7] Conrad C. Huang, Eric F. Pettersen, Teri E. Klein, Thomas E.
Ferrin and Robert Langridge, &quot;Conic: A Fast Renderer for
Space-Filling Molecules with Shadows&quot;, Journal of Molecular Graphics,
Vol.9, No.4, pp.230-236, December 1991.
<p>
<a name="molecularbiologyalgorithms"><h3>Molecular Biology Algorithms</h3></a><p>
[8] Wolfgang Kabsch and Christian Sander, &quot;Dictionary of Protein
Secondary Structure: Pattern Recognition of Hydrogen-Bonded and
Geometrical Features&quot;, Biopolymers, Vol.22, pp.2577-2637, 1983.
<p>
[9] Michael L. Connolly, &quot;Solvent-Accessible Surfaces of Proteins and Nucleic
Acids&quot;, Science, Vol.221, No.4612, pp.709-713, August 1983.
<p>
[10] Khaled Belhadj-Mostefa, Ron Poet and E. James Milner-White,
&quot;Displaying Inter-Main Chain Hydrogen Bond Patterns in Proteins&quot;,
Journal of Molecular Graphics, Vol.9, No.3, pp.194-197, September 1991.
<p>
[11] Mike Carson, &quot;Ribbon Models of Macromolecules&quot;, Journal of
Molecular Graphics, Vol.5, No.2, pp.103-106, June 1987.
<p>
[12] Mike Carson and Charles E. Bugg, &quot;Algorithm for Ribbon Models of
Proteins&quot;, Journal of Molecular Graphics, Vol.4, No.2, pp.121-122, June
1986.
<p>
[13] H. Iijima, J. B. Dunbar Jr. and G. Marshall, &quot;Calibration of
Effective van der Waals Atomic Contact Radii for Proteins and
Peptides&quot;, Proteins: Structure, Functions and Genetics, Vol.2,
pp.330-339,1987.
<p>
<a name="graphicsalgorithms"><h3>Graphics Algorithms</h3></a><p>
[14] J. Foley, A. van Dam, S. Feiner and J. Hughes, &quot;Computer Graphics:
Principles and Practice&quot;, 2nd Edition, Addison Wesley Publishers, 1990.
<p>
[15] J. Cleary and G. Wyvill, &quot;Analysis of an Algorithm for Fast Ray
Tracing using Uniform Space Subdivision&quot;, The Visual Computer, Vol.4,
pp.65-83, 1988.
<p>
[16] Thomas Porter,&quot;Spherical Shading&quot;, Computer Graphics Vol.12, ACM
SIGGRAPH, pp.282-285, 1978.
<p>
[17] Jean-Michel Cense, &quot;Exact Visibility Calculation for Space-Filling
Molecular Models&quot;, Journal of Molecular Graphics, Vol.9, No.3,
pp.191-193, September 1991.
<p>
[18] Chris Schafmeister, &quot;Fast Algorithm for Generating CPK Images on
Graphics Workstations&quot;, Journal of Molecular Graphics, Vol.8, No.4,
pp.201-206, December 1990.
<p>
[19] Bruce A. Johnson, &quot;MSURF: A Rapid and General Program for the
Representation of Molecular Surfaces&quot;, Journal of Molecular Graphics,
Vol.5, No.3, pp.167-169, September 1987.
<p>
<a name="fileformats"><h3>File Formats</h3></a><p>
[20] Frances C. Bernstein et al., &quot;The Protein Data Bank: A
Computer-Based Archival File for Macromolecular Structures&quot;, Journal of
Molecular Biology, Vol.112, pp.535-542, 1977.
<p>
[21] Arthur Dalby, James G. Nourse, W. Douglas Hounshell, Ann K. I.
Gushurst, David L. Grier, Burton A. Leland and John Laufer, &quot;Description
of Several Chemical File Formats Used by Computer Programs Developed at
Molecular Design Limited&quot;, Journal of Chemical Information and Computer
Sciences, Vol.32, No.3, pp.244-255, 1992.
<p>
[22] Adobe Systems Inc., &quot;PostScript Language Reference Manual&quot;,
Addison-Wesley Publishers, Reading, Mass., 1985.
<p>
[23] Philip E. Bourne et al., &quot;The Macromolecular Crystallographic
Information File (mmCIF)&quot;, Meth. Enzymol. (1997) 277, 571-590.
<p>
[24] Sydney R. Hall, &quot;The STAR File: a New Format for Electronic Data
Transfer and Archiving&quot;, Journal of Chemical Information and Computer
Sciences, Vol. 31, 326-333, 1991.
<p><hr>
<CENTER>
| <a href="http://www.OpenrasMol.org">OpenRasMol</a> |
<A href="../README.html#Copying">Copying and Distribution</A> |
<A href="../README.html#Contents">Contents</A> |
<A href="../INSTALL.html">Installation Instructions</A> |<BR>
| <A href="../ChangeLog.html">Changes</A> |
<A href="../TODO.html">Things To Do</A> |
<A href="../README.html#Introduction">Introduction</A> |
<A href="../README.html#CodeAndBinaries">Source Code and Binaries</A> |<BR>
| <A href="rasmol.html">RasMol Manual</A> |
<a href="esrasmol27.html">Spanish Translation of RasMol Manual</a>|
<a href="../README.html">Release README</A> |<BR>
</CENTER>
<HR>
Updated 21 January 2001.<BR>
Herbert J. Bernstein<BR>
Bernstein + Sons, 5 Brewster Lane, Bellport, NY 11713-2803, USA<BR>
<A HREF="mailto:yaya@bernstein-plus-sons.com">
yaya@bernstein-plus-sons.com</A><BR>
+1-631-286-1339<BR>
</font>

</BODY>
</HTML>
